<!DOCTYPE html>
<html lang="en" >
	<meta charset="utf-8"/>
	<head>
		<title>MoBPS </title>
		<!-- Javascripte Online -->
		<script src="https://cdn.jsdelivr.net/npm/vue@2.6.10/dist/vue.js"></script>
		<script src="https://unpkg.com/mathjs@5.8.0/dist/math.js"></script>
		<script src="https://cdnjs.cloudflare.com/ajax/libs/numeric/1.2.6/numeric.min.js"></script>
		
		<script src="https://unpkg.com/vue-slider"></script>
			
		<script src="https://cdn.jsdelivr.net/npm/@riophae/vue-treeselect@0.0.38/dist/vue-treeselect.min.js"></script>
		<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@riophae/vue-treeselect@0.0.38/dist/vue-treeselect.min.css">
 
		
		<!-- for plotting graphs: -->
		<script src="https://cdn.plot.ly/plotly-latest.min.js"></script>

		<script type="text/javascript" src="https://visjs.github.io/vis-network/dist/vis-network.min.js" ></script>
		<script type="text/javascript" src="https://visjs.github.io/vis-network/examples/network/exampleUtil.js"></script>
		<link src="https://visjs.github.io/vis-network/dist/vis-network.min.css" rel="stylesheet" type="text/css" />
		
		<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/vis/4.21.0/vis.min.js"></script>
		<link href="https://cdnjs.cloudflare.com/ajax/libs/vis/4.21.0/vis.min.css" rel="stylesheet" type="text/css" />

		<link rel="stylesheet" type="text/css" href="./styles/network.css">
		
		<!--  for javascript and ajax call -->		
		<script type="text/javascript" src="javascript/jQ1.js"></script>
		<script type="text/javascript" src="javascript/jQ2.js"></script>
		<script type="text/javascript" src="javascript/jQ2_2.js"></script>
		<script type="text/javascript" src="javascript/jQ3.js"></script>
		
		<!-- excel to json related files -->
		<script lang="javascript" src="javascript/jszip_excelToArray.js"></script>
		<script lang="javascript" src="javascript/xlsx_excelToArray.js"></script>
		<script lang="javascript" src="javascript/excelToArray.js"></script>
		
		<!-- for drag and drop for Phenotype table -->	
		<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/sortablejs@1.7.0/Sortable.min.js"></script>
		<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/Vue.Draggable/2.16.0/vuedraggable.min.js"></script>	

	</head>

	<body onload="init();">

		
		<div id="runningDog">
			<button style="color: red" class='bar_button' onclick="stopR();">Stop R Simulation</button> 
			<h3 id="runningDogTitle">
			</h3>
			<div id="runningDogLog">
			</div>
		</div>
		
		<div id='mainDiv'>

			<div id="Project_Bar">
				<div style="overflow:auto;">
					&nbsp;&nbsp;
					<< {{geninfo['Project Name']}} >>
					&nbsp;&nbsp;
					<button class='bar_button' id="new_button" onclick="window.location.reload();">New Project</button> 
					&nbsp;&nbsp;
					<button class='bar_button' id="save_button" onclick="saveProject();">Save Project</button> 
					&nbsp;&nbsp;
					<button class='bar_button' onclick="saveProjectAs();">Copy Project</button> 
					&nbsp;&nbsp;
					<button class='bar_button' id="download_button" onclick="importNetwork()" >Import Project</button>
					&nbsp;&nbsp;
					<button class='bar_button' id="download_button" onclick="downloadNetwork()" >Export Project</button>
					&nbsp;&nbsp;
					<button class='bar_button' id="delete_button" onclick="deleteProject()" >Delete Project</button>
					&nbsp;&nbsp;
					<button class='bar_button' v-on:click='show_geninfo = !show_geninfo; show_warnings = false;' >Show/Hide Info</button>
					&nbsp;&nbsp;
					<button class='bar_button' v-on:click='show_warnings = !show_warnings; show_geninfo=false;' >Show/Hide Warnings</button>
				</div>
			</div>
			<div id='SaveAs_Div'>
				New Project Name: <input id="saveAs-name" placeholder="Type a name"/> <br><br>
				<input type="button" value="save" id="SaveAs-saveButton" onclick="saveProject(document.getElementById('saveAs-name').value)" />
				<input type="button" value="cancel" id="SaveAs-cancelButton" onclick="document.getElementById('SaveAs_Div').style.display = 'none'"/>
			</div>


			<div id="warnings_sum">
				<h4 v-if="warnings.length > 0" style="color: red; font-style:italic;"> Attention, there are {{warnings.length}} warnings! R Simulation most likely cannot be run unless they are fixed.</h4>
			</div>
			<transition name="slide-generalInfo">
				<div id ="warnings" v-show='show_warnings' >
					<br>
					<br>
					<br>
					<br>
					<br>
					<br>
					<span style="font-weight:bold;">Warnings  </span>
					<br><br>
					<span v-for='(war, index) in warnings'> Warning {{index}} : <span style="color: red">{{war}} <br></span></span>
				</div>
			</transition>
			<transition name="slide-generalInfo">
				<div id ="generalInfo" v-show='show_geninfo' >
					<br>
					<br>
					<br>
					<br>
					<br>
					<br>
					<span v-if="show_info =='general'">
						<span style="font-weight:bold;">General Info  </span>
						<br><br>
						<span v-if="key != 'Ensembl Filter' & key != 'Ensembl Filter Values' & value != ''" v-for='(value,key) in geninfo'> {{key}} : <span style="color: red">{{value}}  <br></span></span> 
					</span>
					
					<span v-if="show_info =='node'">
						<span style="font-weight:bold;">Node Info  </span>
						<br><br>
						<span v-if="['x','y', 'individualsVar', 'label','color','title'].indexOf(key) == -1 && value != '' && value != []" v-for='(value,key) in active_node'> {{key}} : <span style="color: red">{{value}} <br></span></span>
					</span>
					<span v-if="show_info =='edge'">
						<span style="font-weight:bold;">Edge Info  </span>
						<br><br>
						<span v-if="['useVar','id','color','title'].indexOf(key) == -1 && value != '' && value != []" v-for='(value,key) in active_edge'> {{key}} : 
							<span style="color: red">{{value}} 
								<span v-if="active_edge.useVar.indexOf(key)>-1"> ({{individualsVar_options[active_edge.useVar.indexOf(key)].name}})</span><br>
							</span>
						</span>
					</span>
				</div>
			</transition>
			
			<div id="mySidebar" class="sidebar">
			<div id="closeDiv">
				<button onclick="closeNav()" class="closebtn"> Navigation Close </button>
			</div>  
			<!-- <a href="javascript:void(0)" class="closebtn" onclick="closeNav()">Close</a> -->
				<a>&nbsp;</a>
					<a href="#General_Info">General information</a>
					<a href="#Phenotype_Info_Div">Phenotype information</a>
					<a href="#Selectionindex_Div">Optional information</a>
					<a href="#Breeding_Scheme">Breeding Scheme</a>
					<a href="#Results_Div">R Simulation/ Results</a>
					<a href="#Export_Data_H3">Export/ Import</a>
					<a href="/logout"> User: {{user}} <br> Logout</a>
					<a v-if="geninfo['curUserGroup'] > 1" href="/Change_Password"> Change Password</a>
			</div>

		<div id="main">
			<div id="openDiv">
		  		<button onclick="openNav()" class="openbtn"> Navigation Open </button>  
			</div>
			
			<div id="LoadJSON_Div" >
			<br>
			<br>
			<h2>MoBPS</h2>
			<h3>Load a new/existent project from your own database:</h3>
			Project: 
			<select id="Project_Name" onchange="loadData(value)" v-model="geninfo['Project Name']">
				<option value="Create_New_123456YYYY">New Project</option>
				<option v-for="option in database" v-bind:value="option">
					{{ option }}
				</option>
			</select>
			&nbsp;&nbsp; Version: 
			<select id="Version" onchange="loadVersion(value)">
				<option value="recent">(most recent)</option>
				<template v-if="versions.length >0">
				<option v-for="(option, index) in versions" v-bind:value="index">
					{{ option.date }}
				</option>
				</template>
			</select>
			<br>
			Exemplary Templates:
			<select onchange="loadTemplate(value)" >
				<option value="Create_New_12345XXX"></option>
				<option v-for="(option,index) in template_database" v-bind:value="index">
					{{ option.name }}
				</option>
			</select>
			
			<br>
			<br>			
			
			<h3 v-if="curUserGroup =='4'"> You are assigned to User Class Admin. 
				This enables you to use 20 Cores and maximum run time of 120 hours 
			</h3>
			<h3 v-if="curUserGroup =='3'"> You are assigned to User Class Professional. 
				This enables you to use 10 Cores and maximum run time of 48 hours 
			</h3>
			<h3 v-if="curUserGroup =='2'"> You are assigned to User Class Student. 
				This enables you to use 5 Cores and maximum run time of 4 hour 
			</h3>
			<h3 v-if="curUserGroup =='1'"> You are assigned to User Class Test. 
				It is not possible to use our backend-server for simulation. To simulate download the json-file run via json.simulation() in R.	</h3>
			
			</div>
			
			
			<div id="General_Info" v-on:click="show_info ='general'" onchange="checkEverything(this.id)">
				<br>
				<h3> General Information</h3> 
				<table id="geninfo_table">
					<colgroup>
						<col width="40%">
						<col width="60%">
					</colgroup>
					<tr>
						<td>Project Name
							<span class="tooltip">&#9432 <span class="tooltiptext">Enter the name of your project. You can save/copy/delete your project via the action bar and load different version to return to.</span></span>
						</td>
						<td>
							<input placeholder="Type a name" v-model="geninfo['Project Name']">
						</td>
					</tr>
					<tr>
						<td>Advanced settings </td> 
						<td>
							<input type="checkbox" v-model="geninfo['advanced']">
						</td>
					</tr>

					<tr v-if="geninfo['advanced']">
						<td>Test-Mode
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock the scaling of node sizes and ignore imported genotypes.</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_test']">
						</td>
					</tr>
					
					<tr v-if="geninfo['advanced']">
						<td>miraculix-active
							<span class="tooltip">&#9432 <span class="tooltiptext">Deactivating this will increase run-times but lead to RData population that can be analyzed without miraculix/RandomFieldsUtils</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_miraculix']">
						</td>
					</tr>
					
					<tr v-if="geninfo['advanced']">
						<td>Parallel Computing + Multiple Simulation
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will allow to run a simulation multiple times.</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_parallel']">
						</td>
					</tr>
					
					<tr v-if="geninfo['advanced']">
						<td>Complex Trait architecture
							<span class="tooltip">&#9432 <span class="tooltiptext"> Activating this will unlock use of dominate and epistatic trait architectures.</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_trait']">
						</td>
					</tr>
				
				

					<tr v-if="geninfo['advanced']">
						<td>Culling Module 
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock the culling module to simulate the death of individuals. Death individuals can still be used for BVE but not for Reproduction</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_culling']">
						</td>
					</tr>
						
					<tr v-if="geninfo['advanced']">
						<td>Subpopulation Module 
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock the specification of subpopulation with different allele frequencies and trait architectures.</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_sub']">
						</td>
					</tr>
					
					<tr v-if="geninfo['advanced']">
						<td>Economic Module 
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock the use of economic parameters.</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_eco']">
						</td>
					</tr>
					
					<tr v-if="geninfo['advanced']">
						<td>Population History 
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating provide you with additional features for the simulation of thousands of generation</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_history']">
						</td>
					</tr>
					
					<tr v-if="geninfo['advanced']">
						<td>Advanced Edge/Node options
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges (share genotyped, selection options etc.)</span></span>
						</td>
						<td> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced']">
						</td>
					</tr>
					<tr>
						<td>Species
							<span class="tooltip">&#9432 <span class="tooltiptext">Select the species. This will unlock the correct Ensembl Datasets etc. Use a template to start with already entered somewhat realistic values.</span></span>
						</td>
						<td> 
							<select v-model="geninfo.Species">  <!-- onchange="updateSpeciesData(value)">  -->
								<option v-for="option in Species_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr>
						<td>Time Unit
							<span class="tooltip">&#9432 <span class="tooltiptext">Select the time unit used. Time of generation for each cohort will be displayed.</span></span>
						</td>
						<td>
							<select v-model="geninfo['Time Unit']">
								<option v-for="option in Time_Unit_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					
					<tr>
						<td>Genetic Data 
							<span class="tooltip">&#9432 <span class="tooltiptext">Select the genome you want to work with.</span></span>
						</td>
						<td>
						<input type="radio" id="ens" value="Ens" v-model="genetic_data">
						<label for="ens">Use Ensembl Map</label> <br>
						<input type="radio" id="own" value="Own" v-model="genetic_data">
						<label for="own">Upload Own Map (vcf/plink)</label> <br>
						<input type="radio" id="cus" value="Cus" v-model="genetic_data">
						<label for="cus">Create customized Map</label>
						</td>
					</tr>	
					<tr v-if="genetic_data == 'Ens'">
						<td>Ensembl Dataset 
							<span class="tooltip">&#9432 <span class="tooltiptext">Currently only the genomic maps available in MoBPSmaps are provided. Alternatively upload your own map or provide us with the map you want us to add in (torsten.pook@uni-goettingen.de).</span></span>
						</td>
						<td v-if="geninfo.Species != ''">
							<select v-model="geninfo['Ensembl Dataset']">
								<option v-for="option in ensembl_options[geninfo.Species]" v-bind:value="option.Value" >
								{{ option.Value }}
								</option>
							</select>
						</td>
						<td v-else>
							Please choose a species first.
						</td>
					</tr>	
					<tr v-if="genetic_data == 'Ens'">
						<td>Max. Number of SNPs 	
							<span class="tooltip">&#9432 <span class="tooltiptext">Only a subsample of the Ensembl dataset is used to save computing time. If nothing is enter use the full dataset.</span></span>
						</td>
						<td v-if="geninfo.Species != ''">
							<input placeholder="Type a number" v-model="geninfo['Max Number of SNPs']">
						</td>
						<td v-else>
						</td>
					</tr>	
<!--					<tr v-if="geninfo['Use Ensembl Map'] =='No'">
						<td>Use Own Map (vcf or plink-map file)</td>
						<td>
						<input type="checkbox" false-value="No" true-value="Yes" v-model="geninfo['Use Own Map']">
						<template v-if="geninfo['Use Own Map'] == 'Yes'">
								hier kommt upload-modul!
						</template>
						</td>
					</tr>  -->
					<tr v-if="geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='Yes'">
						<td>Upload Map File
							<span class="tooltip">&#9432 <span class="tooltiptext">Currently only .vcf .map and .RData-files are supported. Adding more is no problem. Contact Torsten if required.</span></span>
						</td>
						<td>
							<form id='MapUpload' target="print_popup" action='/MapUpload' method='post' encType='multipart/form-data' onsubmit="window.open('about:blank','print_popup','width=1000,height=800'); setMapPath();">
								<input type='file' name='mapFile' id='mapFile'>
								<input type='submit' value='Upload Map'>
							</form>
						</td>
					</tr>
					<tr v-if="geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='Yes'">
						<td>Own Map Path </td>
						<td>
							{{geninfo["Own Map Path"]}}
						</td>
					</tr>
					<tr v-if="geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='No'">
						<td>Number of Chromosomes</td>
						<td><input placeholder="Type a number" v-model="geninfo['Number of Chromosomes']" v-on:change='createChrom();'></td>
					</tr>			
					<tr v-if="geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='No'">
						<td>Chromosomes of Equal Length</td>
						<td> 
							<input type="checkbox" false-value="No" true-value="Yes" v-model="geninfo['Chromosomes of Equal Length']">
						</td>
					</tr>			
					<tr v-show="geninfo['Chromosomes of Equal Length'] == 'Yes' & geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='No'">
						<td colspan=2>
							<button v-on:click='show_chromo_info = !show_chromo_info' >Show/Hide Chromosome Information</button>
							<br>
							<br>
								<table id="chrom_table" v-show="show_chromo_info">
									<thead>
										<tr>
											<td>Chromosome</td>
											<td>Physical Length (in MbP)
												<span class="tooltip">&#9432 <span class="tooltiptext">The human genome has a size of 3200 Mb. Most livestock/crops are smaller!</span></span>
											</td>
											<td>Marker Density (SNP/Mbp)
												<span class="tooltip">&#9432 <span class="tooltiptext">Do not overdo it with markers! They oftentimes mostly add memory requirement and computation time.</span></span>
											</td>
											<td>Recombination (cM/Mbp)
												<span class="tooltip">&#9432 <span class="tooltiptext">1cM/Mbp is commonly assumed. E.g. chicken have higher recombination rates (~3-4 cM/Mbp).</span></span>
											</td>
										</tr>
									</thead>
									<tbody>
										<tr v-for='chrom in chromo_display.slice(0,1)' >
											<td>Chromosome 1 - {{geninfo['Number of Chromosomes']}}</td>
											<td v-for="key in Object.keys(chrom)">
												<input v-model="chrom[key]"> 
											</td>
										</tr>
									</tbody>
								</table>
						</td>
					</tr>
					<tr v-show="geninfo['Chromosomes of Equal Length'] == 'No' & geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='No'">
						<td colspan=2>
							<button v-on:click='show_chromo_info = !show_chromo_info' >Show/Hide Chromosome Information</button>
							<br>
							<br>
							<table id="chrom_table" v-show="show_chromo_info">
								<thead>
									<tr>
										<td>Chromosome</td>
										<td>Physical Length (in MbP)</td>
										<td>Marker Density (SNP/Mbp)</td>
										<td>Recombination (cM/Mbp)</td>
									</tr>
								</thead>
								<tbody>
									<tr v-for='chrom in chromo_display' >
										<td>Chrom {{chromo_display.indexOf(chrom)+1}}</td>
										<td v-for="key in Object.keys(chrom)">
											<input v-model="chrom[key]"> 
										</td>
									</tr>
								</tbody>
							</table>
						</td>
					</tr>
					
					<tr v-if="geninfo['advanced'] && geninfo['advanced_test']">
						<td>Advanced Test-mode:
						</td>

					</tr>
					
					<tr v-if="geninfo['advanced'] && geninfo['advanced_test']">
						<td>Test/Speed Mode
							<span class="tooltip">&#9432 <span class="tooltiptext">Activate this to set the number of repeats to 1, reduce the genome length to 10M and 200 SNPs and deactivate data import. This should heavily reduce computation times for testing.</span></span>
						</td>
						<td> 
							<input type="checkbox" false-value="No" true-value="Yes" v-model="geninfo['speed-mode']">
						</td>
					</tr>
					
					<tr v-if="geninfo['advanced'] && geninfo['advanced_test']">
						<td>Individual numbers scaling 
							<span class="tooltip">&#9432 <span class="tooltiptext">Scale the number of individuals in each node by this factor. This can massively reduce computing times! Point numbers are rounded.</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['individual-scaling']">
						</td>
					</tr>
					

					<tr v-show="geninfo['advanced'] && geninfo['advanced_parallel']">
						<td>Advanced parallel computing: 
						</td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_parallel']">
						<td>Number of Simulations 
							<span class="tooltip">&#9432 <span class="tooltiptext">Number of simulations of the breeding program you what to perform. This will massively increase run-time but also reduce variance of results.</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['number-simulations']">
						</td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_parallel']">
						<td>Number of Simulations run in parallel
							<span class="tooltip">&#9432 <span class="tooltiptext">Number of simulations of the breeding program to run simultaneously. If taking more cores than allowed this will automatically be reduced internally.</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['number-simulations-parallel']">
						</td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_parallel']">
						<td>Number of cores used per simulation
							<span class="tooltip">&#9432 <span class="tooltiptext">Number of cores used per simulation. If taking more cores than allowed this will automatically be reduced internally.</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['number-simulations-core']">
						</td>
					</tr>
					
					<tr v-show="geninfo['advanced'] && geninfo['advanced_history']">
						<td> Advanced population history:
						</td>

					</tr>
					
					<tr v-show="geninfo['advanced'] && geninfo['advanced_history']">
						<td> Set a new base-line population every X - generations
							<span class="tooltip">&#9432 <span class="tooltiptext">Genotypes for a baseline/founder population are stored leading to less points of recombination/mutation.</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['history_baseline']">
						</td>
					</tr>
					
					<tr v-show="geninfo['advanced'] && geninfo['advanced_history']">
						<td> Delete generations before base-line
							<span class="tooltip">&#9432 <span class="tooltiptext">Genotypes for a baseline/founder population are stored leading to less points of recombination/mutation.</span></span>
						</td>
						<td>
							<input type="checkbox" v-model="geninfo['history_delete']">
						</td>
					</tr>

					<tr v-show="geninfo['advanced'] && geninfo['advanced_trait']">
						<td> Advanced trait rescaling:
						</td>

					</tr>
					
					<tr v-show="geninfo['advanced'] && geninfo['advanced_trait']">
						<td> Time point to apply trait rescaling
							<span class="tooltip">&#9432 <span class="tooltiptext">Semicolon separated list of time point to apply rescaling of trait mean/variance (e.g. 50;100)</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['trait_rescaling']">
						</td>
					</tr>
					

				</table>
				<br>

				<br>
			</div>
			<div id="Phenotype_Info_Div" onchange="checkEverything(this.id)">
				<h3 id="Phenotype_Info"> Phenotype Information
					<span class="tooltip">&#9432 <span class="tooltiptext">Here you can design the traits you want to work with.</span></span>
				</h3> 
				
				<button class="accordion"> Click here to read guildelines for Traits table, Upload Excel File, Correlation tables, SI, PC: </button>
				<div class="guideline">
					<p> Trait Table : 1. Drag and Drop facility is available - select a trait and drag your mouse pad Up/Down and drop it to your desired place. </p>
					<p> Trait Table : 2. After Drag and Drop of Traits - dynamically the Correlation matrices values, SI values , PC will be moved according to the Traits order. </p>
					<p> Upload Excel : Please select the type of Excel file from the drop down box. For Genetic correlation - above the diagonal. For Residual correlation - below the diagonal. For both - the above and the below diagonal values should be filled in the selected file. Also set "1" to the diagonal elements. </p>
					<p> Upload Excel : Please choose the Excel File according to the type of Correlation, then click "Import Table" to display the excel values to the related Correlation.</p> 
				</div>
				
				<button v-on:click="addPheno();"> Add new phenotype </button>
				<button v-if="traitsinfo.length > 0" v-on:click='show_pheno_info = !show_pheno_info' >Show/Hide {{traitsinfo.length}} phenotypes</button>
				<button v-if="traitsinfo.length > 0" v-on:click="show_QTL = !show_QTL"> Show/Hide QTLs</button>
				<button v-if="traitsinfo.length > 1" v-on:click="show_matrix = !show_matrix"> Show/Hide residual correlation</button>
				<button v-if="traitsinfo.length > 1" v-on:click="show_matrix2 = !show_matrix2"> Show/Hide genetic correlation</button>
				<br>
				
				
				<table class="drag_table" v-if="traitsinfo.length > 0 && show_pheno_info">
					<thead>
						<tr>
							<td style="width:80px;">Phenotype
								<span class="tooltip">&#9432 <span class="tooltiptext">Name of the trait.</span></span>
							</td>
							<td>Unit
								<span class="tooltip">&#9432 <span class="tooltiptext">Optional. Unit of measurement for the trait.</span></span>
							</td>
							<td>Pheno. Mean
								<span class="tooltip">&#9432 <span class="tooltiptext">The avg. Founder individual will have this mean. To have subpopulations with different mean use the Subpopulation module.</span></span>
							</td>
							<td>Pheno. SD
								<span class="tooltip">&#9432 <span class="tooltiptext">Trait variance for the group of all founders will be scaled accordingly. Residual variance is assumed to be fixed over all generations.</span></span>
							</td>
							<td>Heritability
								<span class="tooltip">&#9432 <span class="tooltiptext">Heritability of the trait.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==false || geninfo['advanced_trait']==false"># Polygenic Loci
								<span class="tooltip">&#9432 <span class="tooltiptext">Number of QTL. To simulate non-additive QTL effect activate the advanced setting Complex Trait architecture.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true"># additive QTL
								<span class="tooltip">&#9432 <span class="tooltiptext">Number of QTL. To simulate non-additive QTL effect activate the advanced setting Complex Trait architecture.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true" ># dominant QTL
								<span class="tooltip">&#9432 <span class="tooltiptext">Number of purely dominant QTL.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true" ># qualitative epistatic QTL
								<span class="tooltip">&#9432 <span class="tooltiptext">Number of QTL with qualitative epistatic effect between two markers.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true" ># quantitative epistatic QTL
								<span class="tooltip">&#9432 <span class="tooltiptext">Number of QTL with quantitative epistatic effect between two markers.</span></span>
							</td>
							<td>Major QTL
								<span class="tooltip">&#9432 <span class="tooltiptext">Use this if you want to manually add single marker effects to SNPs with known effects.</span></span>
							</td>
							<td>Value per unit (&#8364;)
								<span class="tooltip">&#9432 <span class="tooltiptext">Economic gain by increased unit of the trait. Only relevant for economic calculations. Currently only linear effect and no interactions - Contact Torsten if you need more!</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true">Apply Transformation
								<span class="tooltip">&#9432 <span class="tooltiptext">The only allowed input variable is the genomic value (x). Potential input: function(x){y = x^2; return(y)}</span></span>
							</td>
							<td>Show Cor</td>
							<td></td>
							
						</tr>
					</thead>
				</table>
				
								<table>			
					<draggable :list="traitsinfo" 
					class="dragArea"
					:move="moveMatrix"
					@change="seeChange"
					@end="onEnd">
					<template v-for='(trait, index) in traitsinfo' >
						<tr>
							<td style="width:1%;padding-right:6px;">
								<input style="width:80px;" v-model="trait['Trait Name']" > 
							</td>
							<td style="width:1%;padding-right:20px;">
								<input style="width:50px;" v-model="trait['Trait Unit']" > 
							</td>
							<td style="width:1%;padding-right:20px;">
								<input style="width:50px;" v-model="trait['Trait Mean']" class="no-spinners" step=0.001 v-on:change="trait['Trait Mean'] = trait['Trait Mean'].replace(/,/g, '.')"> 
							</td>
							<td style="width:1%;padding-right:20px;">
								<input style="width:50px;" v-model="trait['Trait Std Deviation']" class="no-spinners" step=0.001 v-on:change="trait['Trait Std Deviation'] = trait['Trait Std Deviation'].replace(/,/g, '.')">  
							</td>
							<td style="width:1%;padding-right:20px;">
								<input style="width:50px;" v-model="trait['Trait Heritability']" max=1 min=0 class="no-spinners" step=0.0001 v-on:change="trait['Trait Heritability'] = trait['Trait Heritability'].replace(/,/g, '.')"> 
							</td>
							<td style="width:1%;padding-right:17px;">
								<input style="width:50px;" v-model="trait['Trait Number of Polygenic Loci']" class="no-spinners" step=1> 
							</td>
							<td style="width:1%;padding-right:10px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true">
								<input style="width:50px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true" v-model="trait['dominant_qtl']"  class="no-spinners" step=1> 
							</td>
							<td style="width:1%;padding-right:12px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true">
								<input style="width:50px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true" v-model="trait['qualitative_qtl']" class="no-spinners" step=1> 
							</td>
							<td style="width:1%;padding-right:17px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true">
								<input style="width:50px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true" v-model="trait['quantitative_qtl']"  class="no-spinners" step=1> 
							</td>
							<td style="width:1%;padding-right:17px;">
								<input style="width:50px;" v-model="trait['Trait Major QTL']"  type="Number" class="no-spinners" step=1 v-on:change="createQTL(index)"> 
							</td>
							<td style="width:1%;padding-right:15px;">
								<input style="width:50px;" v-model="trait['Trait Value per Unit']"  type="Number" class="no-spinners" step=0.001 v-on:change="trait['Trait Value per Unit'] = trait['Trait Value per Unit'].replace(/,/g, '.')">  
							</td>
							<td style="width:10%" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true">
								<input v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true" v-model="trait['trafo']" style="width:200px;"> 
							</td>
							<td style="width:2%">
								<input style="width:50px;" type="checkbox" v-model="show_matrix_element[index].show">
							</td>
							<td>
								<button v-on:click="removePheno(index)">X</button>
							</td>
						</tr>
					</template>
					</draggable>
				</table>
				
				<br>
				<template v-for="trait in traitsinfo" v-if="show_QTL && trait['Trait Major QTL'] > 0">
					<table class="trait_table">
						<thead>
							<tr>
								<td>SNP for {{trait['Trait Name']}}</td>
								<td>SNP ID
									<span class="tooltip">&#9432 <span class="tooltiptext">If SNP ID is given in the dataset it is used. If not the closest marker to bp/chromo combination is used. If not given, use SNP/chromosome combination.</span></span>
								</td>
								<td>bp</td>
								<td>chromo</td>
								<td>Effect AA</td>
								<td>Effect AB</td>
								<td>Effect BB</td>
								<td>Allele Freq. (B)
									<span class="tooltip">&#9432 <span class="tooltiptext">Simulated allele frequency of the marker is set to this. This is only active if no genotype data is imported.</span></span>
								</td>
								<td>Optional Info</td>
								<td></td>
							</tr>
						</thead>
						<tbody>
							<tr v-for='snp in trait["Trait QTL Info"]' >
								<td v-for="key in Object.keys(snp).slice(0,1)">
									<input v-model="snp[key]" style="width:100px;"> 
								</td>
								<td v-for="key in Object.keys(snp).slice(1,9)">
									<input v-model="snp[key]" style="width:60px;" v-on:change="snp[key] = snp[key].replace(/,/g, '.')"> 
								</td>
								<td>
									<button v-on:click="removeQTL(trait,snp)">X</button>
								</td>
							</tr>
						</tbody>
					</table>
				</template>


			<div id="corrDiv" style="display:none">
				<button class="accordion" style="font-style:normal;font-size: 14px;"> Click here to Upload Excel file for Correlation:  <span class="tooltip">&#9432 <span class="tooltiptext">"Here you can import an Excel file with genetic correlations above the diagonal and residual correlations below the diagonal. Please set "1" to the diagonal elements. The first row and first column should be name of the Phenotype."</span></span> </button>
				<div class="guideline">
 				<table>
					<tr>
					<td>Select the type of Excel File:
						<span class="tooltip">&#9432 <span class="tooltiptext">"Please select an Excel file of genetic correlations or residual correlations or both."</span></span>
					</td>
					 <td> 
						<select v-model="geninfo.Excel_File">  
							<option v-for="option in Excel_File_options" v-bind:value="option">
							{{ option }}
							</option>
						</select>
					</td>
					</tr>
					<tr> <td>Choose Excel File from Desktop:
						<span class="tooltip">&#9432 <span class="tooltiptext">"Please choose an Excel file from desktop."</span></span>
					</td> 
					<td> <input id="excelToArray" type=file style="width:178px;" >
					<button id="save_button" onclick="saveProject(geninfo['Project Name']);">Import Table</button></td>
					</tr>
				</table>
				</div>	
				</div>

				<br>
				<table class="corr_table" v-if="traitsinfo.length > 1 && show_matrix">
					<tr>
						<td colspan="5" style="font-weight:bold;width:60px;padding:5px;text-align:left">Residual Correlation
							<span class="tooltip">&#9432 <span class="tooltiptext">Make sure this matrix is positive definit. Otherwise we will project the matrix to the space of positive definit matrices.</span></span>
						</td>
					</tr>
					<tr>
						<td></td>
						<td style="width:60px;padding:5px;" v-for="(row, rind) in matrix" v-if="show_matrix_element[rind].show">
							{{traitsinfo[rind]['Trait Name']}}
						</td>
					</tr>
					<tr v-for='(row, rind) in matrix' v-if="show_matrix_element[rind].show">
						<td style="padding:2px;">
							{{traitsinfo[rind]['Trait Name']}}
						</td>
						<td v-for="(cl, cind) in matrix" v-if="show_matrix_element[rind].show & show_matrix_element[cind].show">
							<input v-model="matrix[rind].row[cind].val" style="width:40px;" v-if="rind > cind" class="no-spinners" step=0.000001 min=-1 max=1 v-on:change="matrix[rind].row[cind].val = matrix[rind].row[cind].val.replace(/,/g, '.')">  
							<template v-if="rind <= cind">
								{{ matrix[cind].row[rind].val }}
							</template>
						</td>
					</tr>
				</table>

				<table v-if="traitsinfo.length > 1 && show_matrix">
				<tr>
				<td>
					Enter Phenotypic correlation instead of residual correlation
					</td>
					<td>
						<input type="checkbox" v-model="use_phenotypic_cor">
					</td>
					</tr>
					<br>
				</table>

				
				<table class="corr_table" v-if="traitsinfo.length > 1 && show_matrix2">
					<tr>
						<td colspan="5" style="font-weight:bold;width:60px;padding:5px;text-align:left">Genetic Correlation
							<span class="tooltip">&#9432 <span class="tooltiptext">Make sure this matrix is positive definit. Otherwise we will project it so the space of positive definit matrices</span></span>
						</td>
					</tr>

					<tr>
						<td></td>
						
						<td style="width:60px;padding:5px;" v-for="(row, rind) in matrix2" v-if="show_matrix_element[rind].show">
							{{traitsinfo[rind]['Trait Name']}}
						</td>
					</tr>
					<tr v-for='(row, rind) in matrix2' v-if="show_matrix_element[rind].show">
						<td style="padding:2px;">
							{{traitsinfo[rind]['Trait Name']}}
						</td>
						<td v-for="(cl, cind) in matrix2" v-if="show_matrix_element[rind].show & show_matrix_element[cind].show">
							<input v-model="matrix2[rind].row[cind].val" style="width:40px;" v-if="rind > cind" class="no-spinners" step=0.000001 min=-1 max=1 v-on:change="matrix2[rind].row[cind].val = matrix2[rind].row[cind].val.replace(/,/g, '.')">
							<template v-if="rind <= cind">
								{{ matrix2[cind].row[rind].val }}
							</template>
						</td>
					</tr>
				</table>
				<br>
				<br>
			</div>
			
			<div id="Selectionindex_Div" onchange="checkEverything(this.id)">
				<h3 v-if="traitsinfo.length > 0">Creating own Selection Indexes (Optional)
					<span class="tooltip">&#9432 <span class="tooltiptext">Create your own selection indexes in order to use it in the breeding scheme.</span></span>
				</h3> 
				<input id="newSI" placeholder="Type SI name" v-show="traitsinfo.length > 0">
				<button v-on:click="addSI()" v-if="traitsinfo.length > 0"> Add new SI</button>
				<br>
				<br>
				<template v-if="traitsinfo.length > 0">
					<table class="trait_table">
						<thead>
							<tr>
								<td>SI
									<span class="tooltip">&#9432 <span class="tooltiptext">Name of the selection index</span></span>
								</td>

								<td v-for="trait in traitsinfo"> {{trait['Trait Name']}}</td>
								
								<td></td>
							</tr>
						</thead>
						<tbody>
							<tr v-for='(si,index) in selection_index' >
								<td>
									<!-- <input v-model="si['Name']" style="width:80px;"> -->
									{{si['Name']}}
								</td>
								
								<td v-for="(key, ind) in Object.keys(si)" v-if="ind >0">
									<input v-model="si[key]" style="width:30px;" v-on:change="si[key] = si[key].replace(/,/g, '.')">
								</td>


								<td>
									<button v-on:click="removeSI(si, index)" v-if="index > 1">X</button>
								</td>
							</tr>

						</tbody>
					</table>
					<br>
					<br>
					<table class="trait_table">
						<thead>
							<tr>
								<td>SI
									<span class="tooltip">&#9432 <span class="tooltiptext">Name of the selection index</span></span>
								</td>
								
								<!--td>Selection Index via Miesenberger 1997
									<span class="tooltip">&#9432 <span class="tooltiptext">Index weights are taken as w in the Miesenberger 1997 formula (extension of Hazel and Lush). This will account for correlations between genomic/phenotypic values. THIS IS STILL WORK IN PROGRESS!</span></span>
								</td-->
								
								<td>Standardization 
									<span class="tooltip">&#9432 <span class="tooltiptext">Scaling of breeding values before application of index weights.</span></span>
								</td>
								<td></td>
							</tr>
						</thead>
						<tbody>

							<tr v-for='(si,index) in selection_index_scaling' >
								<td>
									<!-- <input v-model="si['Name']" style="width:80px;"> -->
									{{si['Name']}}
								</td>
								<!--td>
									<input type="checkbox" v-model="si.miesenberger">
								</td-->
								<td>
									<select v-model="si.w_scaling">
									<option v-for="option in w_options" v-bind:value="option">
									{{ option }}
									</option>
									</select>
								</td>

							</tr>

						</tbody>
					</table>
					<br>
					<br>
				</template>			
				<h3 v-if="traitsinfo.length > 0">Creating own Phenotyping Classes (Optional)
					<span class="tooltip">&#9432 <span class="tooltiptext">Phenotyping Classes can be attached to each cohort in the breeding scheme, indicating which phenotype is available in the respective cohort. </span></span>
				</h3>
				<input id="newPhenoC" placeholder="Type PhenoC name" v-show="traitsinfo.length > 0">
				<button v-on:click="addPC()" v-if="traitsinfo.length > 0"> Add new PhenoC</button>
				<br>
				<br>
				<template v-if="traitsinfo.length > 0">
					<table class="trait_table">
						<thead>
							<tr>
								<td>PhenoClass
									<span class="tooltip">&#9432 <span class="tooltiptext">A number higher than 1 means multiple INDEPENDET phenotypes are observed. This reduces residual variance and increases heritability.</span></span>
								</td>
								<td>Phenotyping Cost</td>
								<td v-for="trait in traitsinfo"> {{trait['Trait Name']}}</td>
								<td></td>
							</tr>
						</thead>
						<tbody>
							<tr v-for='(si,index) in phenotyping_class' >
								<td>
									<!-- <input v-model="si['Name']" style="width:90px;"> -->
									{{si['Name']}}
								</td>
								<td v-for="(key, ind) in Object.keys(si)" v-if="ind >0" >
									<input v-model="si[key]" style="width:30px;" v-on:change="si[key] = si[key].replace(/,/g, '.')">								</td>
								<td>
									<button v-on:click="removePC(si, index)" v-if="index > 0">X</button>
								</td>
							</tr>
						</tbody>
					</table>
					<br>
				</template>	
				
				<h3 v-if="geninfo['advanced'] && geninfo['advanced_culling']"> Culling
					<span class="tooltip">&#9432 <span class="tooltiptext">Define reasons for individuals to exit the breeding program for reproduction. Usage for BVE is still possible. 
					Individuals with BV1 (in the index selected) will die with probability 'Share death at BV1'. In case of no underlying genomics all individuals will die with this 
					probability. In case of a genetic reason provide BV2. Values for each genomic value will be derived based on linear extension.</span></span>
				</h3>
				<table class="trait_table" v-if="geninfo['advanced'] && geninfo['advanced_culling']">
					<tr>	
						<td style="font-weight:bold">Culling reason	
						</td>
						<td style="padding:5px;">At age (in time unit)
							<span class="tooltip">&#9432 <span class="tooltiptext">At what age does the culling potentially occur. Usage for reproduction at the time point itself is still possible.</span></span>
						</td>
						<td style="padding:5px;">Relevant for sex
							<span class="tooltip">&#9432 <span class="tooltiptext">Is this relevant for Male/Female/Both?</span></span>
						</td>
						<td style="padding:5px;">Genetic Index for survival
							<span class="tooltip">&#9432 <span class="tooltiptext">Select a selection index linked to the culling</span></span>
						</td>
						<td style="padding:5px;">Share death at BV1
							<span class="tooltip">&#9432 <span class="tooltiptext">What share of individuals are culled when they have BV1 (genomic value * selected index).</span></span>
						</td>
						<td style="padding:5px;">BV1
						</td>
						<td style="padding:5px;">Share death at BV2
							<span class="tooltip">&#9432 <span class="tooltiptext">What share of individuals are culled when they have BV2 (genomic value * selected index).</span></span>
						</td>
						<td style="padding:5px;">BV2
						</td>
						<td>
						</td>

					</tr>
					<tr v-for="(vv, ind) in culling['culling_reasons']">
						<td>
							{{vv.Name}}  
						</td>
						<td>
							<input v-model="vv.At_age" style="width:100px;"> 
						</td>
						<td>
							<select v-model="vv.Relevant_Sex">
								<option v-for="option in Sex_options2" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
						<td>
							<select v-model="vv.Genetic_index_for_survival">
								<option v-for="option in selection_index" v-bind:value="option.Name">
								{{ option.Name }}
								</option>
							</select>
						</td>
						<td>
							<input v-model="vv.Share_death_at_BV1" style="width:100px;"> 
						</td>
						<td>
							<input v-model="vv.BV1" style="width:100px;"> 
						</td>
						<td>
							<input v-model="vv.Share_death_at_BV2" style="width:100px;"> 
						</td>
						<td>
							<input v-model="vv.BV2" style="width:100px;"> 
						</td>
						<td>
							<button v-on:click="removeCulling(vv, ind)">X</button>
						</td>
					</tr>
					<tr>
						<td>
							<input id="newCullingClass" placeholder="Type a Culling Reason">
						</td>

						<td colspan="2" style="text-align:left;">
							<button for="newCullingClass" v-on:click="createCulling()"> Creating new Culling Reason </button>
						</td>
					</tr>	
				</table>
				
				<br>
				<br>
				
				<h3 v-if="geninfo['advanced'] && geninfo['advanced_sub']"> Multiple Subpopulations
					<span class="tooltip">&#9432 <span class="tooltiptext">Define allele frequencies for multiple subpopulations. 
					Allele frequencies are drawn from a Beta(p,q) distribution.
					To model different levels of diversity use Share of fixated markers.
					A population with less diversity should have more fixated markers.
					More similar population should be fixated in the same direction for often. 
					This cannot replace the use of REAL genomic data or simulated data with population structure (this module is mostly intended for baseline testing).</span></span>
				</h3>
				<table v-if="geninfo['advanced'] && geninfo['advanced_sub']" class="trait_table">
					<tr>	
						<td style="font-weight:bold">Subpopulation	
						</td>
						<td style="padding:5px;">Sampled allele frequency (p)
							<span class="tooltip">&#9432 <span class="tooltiptext">Allele frequencies are drawn from a Beta (p,q)</span></span>
						</td>
						<td style="padding:5px;">Sampled allele frequency (q)
							<span class="tooltip">&#9432 <span class="tooltiptext">Allele frequencies are drawn from a Beta (p,q)</span></span>
						</td>
						<td style="padding:5px;">Share of fixated markers in A
											<span class="tooltip">&#9432 <span class="tooltiptext">Define allele frequencies for multiple subpopulations. 
					Allele frequencies are drawn from a Beta(p,q) distribution.
					To model different levels of diversity use Share of fixated markers.
					A population with less diversity should have more fixated markers.
					More similar population should be fixated in the same direction for often. 
					This cannot replace the use of REAL genomic data or simulated data with population structure (this module is mostly intended for baseline testing).</span></span>
						</td>
						<td style="padding:5px;">Share of fixated markers in B
											<span class="tooltip">&#9432 <span class="tooltiptext">Define allele frequencies for multiple subpopulations. 
					Allele frequencies are drawn from a Beta(p,q) distribution.
					To model different levels of diversity use Share of fixated markers.
					A population with less diversity should have more fixated markers.
					More similar population should be fixated in the same direction for often. 
					This cannot replace the use of REAL genomic data or simulated data with population structure (this module is mostly intended for baseline testing).</span></span>
						</td>
						<td style="padding:5px;">Nr. of Markers with manual chosen allele frequency
						</td>
						<td v-for="trait in traitsinfo"> Deviation from Mean for Trait {{trait['Trait Name']}}</td>
						<td>
						</td>

					</tr>
					<tr v-for="(vv, ind) in subpopulation['subpopulation_list']">
						<td>
							{{vv.Name}}  
						</td>
						<td>
							<input v-model="vv.beta1" style="width:100px;"> 
						</td>
						<td>
							<input v-model="vv.beta2" style="width:100px;"> 
						</td>
						<td>
							<input v-model="vv.share0" style="width:100px;"> 
						</td>
						<td>
							<input v-model="vv.share1" style="width:100px;"> 
						</td>
						<td>
							<input v-model="vv.fixed_freq" style="width:100px;" type="Number" class="no-spinners" step=1 v-on:change="createQTLsub(ind)"> 
						</td>
						
						<td v-for="(trait, ind2) in traitsinfo">
							<input v-model="vv['P' + (ind2+1)]" style = "width:30px;" v-if="ind>0" v-on:change="vv['P' + (ind2+1)] = vv['P' + (ind2+1)].replace(/,/g, '.')">
						</td> 

						
						<td>
							<button v-on:click="removeSub(vv, ind)">X</button>
						</td>
					</tr>
					<tr>
						<td colspan="2" style="text-align:left;">
							<input id="newSubpopulationClass" placeholder="Type a Subpopulation name">
						</td>
						<td>
							<button for="newSubpopulationClass" v-on:click="createSub()"> Creating new Subpopulation </button>
						</td>
					</tr>	
				</table>
				
				<template v-for="subpop in subpopulation['subpopulation_list']" v-if="subpop['fixed_freq'] > 0">
					<table class="trait_table">
						<thead>
							<tr>
								<td>SNP for {{subpop['Name']}}</td>
								<td>SNP ID
									<span class="tooltip">&#9432 <span class="tooltiptext">If SNP ID is present in the dataset it is used. If not the closed marker to bp/chromo combination is used. If not given use SNP/chromo combination.</span></span>
								</td>
								<td>bp</td>
								<td>Chromosome</td>
								<td>Allele Freq. (B)
									<span class="tooltip">&#9432 <span class="tooltiptext">Simulated allele frequency of the marker is set to this. Only active if no genotype data is imported.</span></span>
								</td>
								<td>Optional Info</td>
								<td></td>
							</tr>
						</thead>
						<tbody>
							<tr v-for='snp in subpop["QTL Info"]' >
								<td v-for="key in Object.keys(snp).slice(0,1)">
									<input v-model="snp[key]" style="width:100px;"> 
								</td>
								<td v-for="key in Object.keys(snp).slice(1,9)">
									<input v-model="snp[key]" style="width:60px;" v-on:change="snp[key] = snp[key].replace(/,/g, '.')"> 
								</td>
<!--							
								<td>
									<button v-on:click='removeQTLsub(subpop, subpop["QTL Info"], snp)'>X</button>
								</td>
-->								
							</tr>
						</tbody>
					</table>
					<br>
				</template>
				
				<br>
				<br>
				
				<h3 v-if="geninfo['advanced'] && geninfo['advanced_eco']">General Economy Parameters (Optional)
					<span class="tooltip">&#9432 <span class="tooltiptext">Define your own economic parameters for the calculation of economic values in your breeding scheme. </span></span>
				</h3>
				<table v-if="geninfo['advanced'] && geninfo['advanced_eco']" style="text-align:left;background:white; border: 1px solid lightgray" class="trait_table">
					<colgroup>	
						<col width="40%">
						<col width="60%">
					</colgroup>
					<tr>
						<td>Fixed cost (&#8364;)
							<span class="tooltip">&#9432 <span class="tooltiptext">Fixed costs of the breeding program</span></span>
						</td>
						<td><input placeholder="Type a number" v-model="economy['Fixed Cost']"></td>
					</tr>
					<tr>
						<td>Interest rate (%)
							<span class="tooltip">&#9432 <span class="tooltiptext">Applied interest rate for all costs occurring.</span></span>
						</td>
						<td><input placeholder="Type a number" v-model="economy['Interest Rate']"></td>
					</tr>
					<tr>
						<td>Genotyping cost per individual (&#8364;)
							<span class="tooltip">&#9432 <span class="tooltiptext">Genotyping costs per individual. Currently one of genotyping with a single array supported possible. Modes with different chips and marker densities are coming.</span></span>
						</td>
						<td><input placeholder="Type a number" v-model="economy['Genotyping Cost']"></td>
					</tr>
					<tr style="height:15px"><td colspan=2></td></tr>
					<tr>	
						<td style="font-weight:bold">Housing/Field cost class	
							<span class="tooltip">&#9432 <span class="tooltiptext">Type a name and then click on "Creating new HFC Class" to create your own classes of housing/field costs. These classes can then be assigned to each cohort in the breeding scheme. </span></span>
						</td>
						<td style="font-weight:bold">Housing/Field cost (&#8364;)</td>
					</tr>
					<tr v-for="(vv, ind) in economy['Animal Housing Costs']">
						<td>
							{{vv.Name}}  
						</td>
						<td>
							<input v-model="vv.Cost" style="width:100px;"> 
							<button v-if="ind > 0" v-on:click="removeHCostClass(vv, ind)">X</button>
						</td>
					</tr>

					<tr>
						<td>
							<input id="newHCostClass" placeholder="Type a HFC name">
						</td>
						<td colspan="2">
							<button for="newHCostClass" v-on:click="createHCostClass()"> Creating new HFC Class </button>
						</td>
					</tr>					
				</table>
			<br>
					<div id="Variables_Info" onchange="checkEverything(this.id)">
					<h3> Declaring own variables (Optional)	
						<span class="tooltip">&#9432 <span class="tooltiptext">Variables generated here can be used as values for numerics in the graphical enviroment. This allows to directly change numeric values for multiple nodes. </span></span>
					</h3> 
					<input id="newVariable" placeholder="Type variable name">
					<button for="newVariable" v-on:click="createVariable()"> Creating new Variable </button>
					<br>
						<table class="trait_table" v-if="individualsVar_options.length > 0" style="width:500px">  
						<thead>
							<tr>
								<td>Variable Name</td>
								<td>Variable Value</td>
								<td></td>
							</tr>
						</thead>
						<tbody>
							<tr v-for="vv in individualsVar_options">
								<td>
									{{vv.name}}  
								</td>
								<td>
									<input v-model="vv.value" style="width:100px;" v-on:input="updateNodesVar(vv); updateEdgesVar(vv);"> 
								</td>
								<td>
									<button v-on:click="removeVariable(vv)">X</button>
								</td>
							</tr>
						</tbody>
					</table>
					<br>
					</div>
			</div>
			<div id="Breeding_Scheme">
				<br>
				<br>
				<h3> Breeding Scheme
					<span class="tooltip">&#9432 <span class="tooltiptext">Draw your breeding scheme in the following interactive enviroment.</span></span>
				</h3> 
				
				<button class="accordion"> Click here to read the guildelines for Breeding Scheme: </button>
				<div class="guideline">
					<p> Add a Node : Click Edit -> Add Node  -> Click in an empty space to place a new node.  </p>
					<p> Edit a Node : Select a Node from the diagram -> Click Edit Node </p>
					<p> Drag a Node : Add/Edit a Node or Edge -> Use grey area to drag the Node/Edge display box. </p>
					<p> Doublic Click a Node/Edge : Display only box appears for double click a node or edge. Need to click Edit Node to update a node/edge.</p>
					<p> Copy Node : Select a Node -> Right Click.</p>					
				</div> 
				
				<h4 style="color: red">Important: Please AVOID node names with : and _ . In particular avoid trailing numbers like ABC_1. Repeated nodes will use this syntax!</h4>
				<div id="mynetwork_container" >
					<div id="mynetwork_label">
		<!--				<button onclick="saveProject();">Save to Database</button>
						<br>
						<br>
						<input style="width:100px" type="text" value = "" name="saveAs_name" id="saveAs_name" placeholder="Enter a file name">
						<button id = "save_as" onclick="saveAsProject();">Save As</button>
						<br> -->
						<h4>Legends</h4>
						Nodes:
						<p v-for="key in Object.keys(node_colors)" v-bind:style="{color: node_colors[key]}"> &#9645;  {{key}}</p>
						Edges:
						<p v-for="key in Object.keys(edge_colors)" v-bind:style="{color: edge_colors[key]}"> &#8599;  {{key}}</p>
					</div>
					<div id="mynetwork"></div>

				</div>
			</div>

			<div id="node-popUp" onchange="checkEverything(this.id)">
				<div id="node-popUpheader"> 
					<h3> {{node_operation}} </h3> <h6> Drag Node in this Area! </h6> 
					<h4> Node {{ active_node.id }}</h4> <h6> </h6>
				</div> 
				<table style="margin:auto; width:100%">
					<colgroup>
						<col width="40%">
						<col width="60%">
					</colgroup>
					<tr>
						<td>Name
							<span class="tooltip">&#9432 <span class="tooltiptext">Name of the cohort. Please avoid names with : or _ . In particular avoid trailing numbers like ABC_1. Repeated nodes will use this syntax!</span></span>
						</td><td><input placeholder="Type a name" v-model="active_node.id" ></td>
					</tr>
					<tr>
						<td>Number of individuals
							<span class="tooltip">&#9432 <span class="tooltiptext">Number of individuals in this cohort</span></span>
						</td>
						<td><input placeholder="Type number/Select variable" id="individuals" list="indVar" v-model="active_node.individualsVar" onchange="updateIndividuals(value)">
							<datalist id="indVar">
								<option v-for="ivar in individualsVar_options" v-bind:value="ivar.name">
							</datalist>
						</td>
	
					</tr>
					
					<tr>
						<td>Founder
							<span class="tooltip">&#9432 <span class="tooltiptext">Are the individuals in this cohort founders or are they generated?</span></span>
						</td>
						<td> 
							<input type="checkbox" false-value="No" true-value="Yes" v-model="active_node.Founder">
						</td>
					</tr>
					<tr v-show="active_node.Founder == 'Yes'">
						<td> Genotype generation type 
							<span class="tooltip">&#9432 <span class="tooltiptext">How to generate genotypes/haplotypes for founder individuals</span></span>
						</td>
						<td>
							<select v-model="active_node['Genotype generation']">
								<option v-for="option in Genotype_generation_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-show="active_node.Founder == 'Yes' & active_node['Genotype generation'] == 'Upload Genotypes'">
						<td>Upload genotypes (plink/vcf)
							<span class="tooltip">&#9432 <span class="tooltiptext">You can provide a dataset in plink/vcf format. Only used markers that are included in the map. Make sure your dataset is phased or activate phasing via checking Phasing required! </span></span>
						</td>
						<td> 
							<form id='GenoUpload' target="print_popup" action='/GenoUpload' method='post' encType='multipart/form-data' onsubmit="window.open('about:blank','print_popup','width=1000,height=800'); setGenoPath();">
								<input type='file' name='genoFile' id='genoFile'>
								<input type='submit' value='Upload'>
							</form>
						</td>

					</tr>

					<tr v-show="active_node.Founder == 'Yes' & active_node['Genotype generation'] == 'Upload Genotypes' ">
						<td> Path genotypes </td><td>{{active_node.Path}}</td>
					</tr>	
					<tr v-show="active_node.Founder == 'Yes' & active_node['Genotype generation'] == 'Upload Genotypes' ">
						<td>Phasing required
							<span class="tooltip">&#9432 <span class="tooltiptext">If imputation via BEAGLE 5.0 default will be automatically processed. This is using the MoBPS function pedmap.to.phasebeaglevcf(). It is not supported when running MoBPS on your own system as paths will usally not match! Contact Torsten if interested to use this locally.</span></span>
						</td>
						<td>
							<input type="checkbox" false-value="No" true-value="Yes" v-model="active_node.phasing_required">
						</td>
					</tr>
					<tr v-show="active_node.Founder == 'Yes' & geninfo['advanced'] & geninfo['advanced_sub']">
						<td>Population allele frequencies
							<span class="tooltip">&#9432 <span class="tooltiptext">Pick the originating subpopulation. Allele frequencies for nodes with no genotype data from the same subpopulation will have similar allele frequencies.</span></span>
						</td>
						<td>
							<select v-model="active_node['Genotype generation subpopulation']">
								<option v-for="option in subpopulation['subpopulation_list']" v-bind:value="option.Name">
								{{ option.Name }}
								</option>
							</select>
						</td>
					</tr>					
					<tr>
						<td>Sex
							<span class="tooltip">&#9432 <span class="tooltiptext">Sex of the individuals in this cohort</span></span>
						</td>
						<td> 
							<select v-model="active_node.Sex">
								<option v-for="option in Sex_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr>
						<td>Phenotyping Class
							<span class="tooltip">&#9432 <span class="tooltiptext">Choose one of the phenotyping classes generated prior. Default is observation of all traits.</span></span>
						</td>
						<td> 
							<select v-model="active_node['Phenotyping Class']">
								<option v-for="option in phenotyping_class" v-bind:value="option.Name">
								{{ option.Name }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] && geninfo['advanced_eco']">
						<td>Housing Cost Class
							<span class="tooltip">&#9432 <span class="tooltiptext">Choose the housing class for the individuals in this cohort</span></span>
						</td>
						<td> 
							<select v-model="active_node['Housing Cost Class']">
								<option v-for="option in economy['Animal Housing Costs']" v-bind:value="option.Name">
								{{ option.Name }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>Proportion of genotyped individuals
							<span class="tooltip">&#9432 <span class="tooltiptext">Select the share of genotyped individuals. Mostly relevant for the use of single-step and economic calculations.</span></span>
						</td>
						<td> <input placeholder="Type a number" type="number" max=1 min=0 step=0.001 v-model="active_node['Proportion of genotyped individuals']"> </td>
					</tr>
					<!--tr>
						<td>Include in Plot </td> 
						<td> <input type="checkbox" false-value="No" true-value="Yes" v-model="active_node['BV Plot']"> 
						</td>
					</tr-->	
					<tr v-show="active_node.Sex == 'Both'">
						<td>Proportion of Male</td>
						<td> <input placeholder="Type a number" type="number" max=1 min=0 step=0.001 v-model="active_node['Proportion of Male']"> </td>
					</tr>
				</table>
				<br>
				<input class='button_type' type="button" value="save" id="node-saveButton" />
				<input class='button_type' type="button" value="cancel" id="node-cancelButton" />
			</div>
			<div id="edge-popUp" onchange="checkEverything(this.id)">
				
				<div id="edge-popUpheader"> 
					<h3> {{edge_operation}} </h3><h6> Drag Edge in this Area! </h6><br>
					<h4>Edge {{ active_edge.id }}</h4>
				</div>
				
				<table style="margin:auto; width: 100%;">
					<colgroup>
						<col width="40%">
						<col width="60%">
					</colgroup>
					<br>
					<tr>
						<td>Breeding Type
							<span class="tooltip">&#9432 <span class="tooltiptext">Selected the breeding action you want to use to generate new cohorts and/or link them. </span></span>
						</td>
						<td> 
							<select v-model="active_edge['Breeding Type']">
								<option v-for="option in Breedingtype_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-if="active_edge['Breeding Type'] == 'Split' && edge_operation == 'Add Edge'">
						<td>ID of 2.Sub-Group
							<span class="tooltip">&#9432 <span class="tooltiptext">Some breeding action require multiple cohorts</span></span>
						</td>
						<td> 
							<input id="split_input" list="splits" placeholder="Create/Choose ID">
							<datalist id="splits">
								<option v-for="par in nodes.get()" v-if="par.Sex == nodes.get(active_edge.from).Sex & par.id != nodes.get(active_edge.to).id & par.id != nodes.get(active_edge.from).id" v-bind:value="par.id">
							</datalist>
						</td>
					</tr>
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production','Aging', 'Selection', 'Split','Combine'].indexOf(active_edge['Breeding Type']) >= 0 ">
						<td>Time needed
							<span class="tooltip">&#9432 <span class="tooltiptext">Provide the time this breeding action takes. Reproduction currently needs at least one time unit.</span></span>
						</td>
						<td><input id="time_n" list="indVar" onchange="updateVar('Time Needed', value)" placeholder="Type a number" min="0" max="10000" step="1" v-model="active_edge['Time Needed']"> {{geninfo['Time Unit']}}</td>
					</tr>	
					<tr v-show="geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>Use last available
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will use the last repeat available from that node - not necessary the node from the same repeat</span></span>
						</td>		
						<td><input type="checkbox" v-model="active_edge['last_available']"> </td>
					</tr>
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>OGC
							<span class="tooltip">&#9432 <span class="tooltiptext">Use optimum genetic contribution according to Meuwissen 1996 to determine how often each individual is used for reproduction.</span></span>
						</td>
						<td><input type="checkbox" false-value="No" true-value="Yes"  v-model="active_edge['OGC']"></td>
					</tr>	
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['OGC']=='Yes'">
						<td>Maximum increase of inbreeding </td>
						<td><input placeholder="Type in the target increase in inbreeding. Use 0 to minimize." v-model="active_edge['ogc_aCa']"></td>
					</tr>	

					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>Maximal # offspring
							<span class="tooltip">&#9432 <span class="tooltiptext">Maximum number of offspring each individual from the parent node is allowed to have.</span></span>
						</td>
						<td><input placeholder="Type a number or Inf" v-model="active_edge['Max Offspring']"></td>
					</tr>	
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>Selection ratio
							<span class="tooltip">&#9432 <span class="tooltiptext">Use this if you want to use some individuals more frequently for reproduction. A value of 2 will lead to the individual will the highest genomic value to be used twice as often as the worst individual.</span></span>
						</td>
						<td><input placeholder="Type a number (if nothing enter all individuals are picked equally)" v-model="active_edge['selection_ratio']"></td>
					</tr>	
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>Selection ratio type
							<span class="tooltip">&#9432 <span class="tooltiptext">Use this if you want to use some individuals more frequently for reproduction. A value of 2 will lead to the individual will the highest genomic value to be used twice as often as the worst individual.</span></span>
						</td>
						<td> 
							<select v-model="active_edge['selection_ratio_type']">
								<option v-for="option in selection_type" v-bind:value="option">
								{{ option}}
								</option>
							</select>
						</td>					</tr>	
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>Index for Selection ratio / OGC
							<span class="tooltip">&#9432 <span class="tooltiptext">Select the index used for the selection ratio or OGC. </span></span>
						</td>
						<td> 
							<select v-model="active_edge['selection_ratio_index']">
								<option v-for="option in selection_index" v-bind:value="option">
								{{ option.Name }}
								</option>
							</select>
						</td>
					</tr>	
					<tr v-if="active_edge['Breeding Type'] == 'Reproduction' && edge_operation == 'Add Edge'">
						<td>ID of 2.parent </td> 
						<td> 
							<input id="parents_input" list="parents" placeholder="Create/Choose ID">
							<datalist id="parents">
								<option v-for="par in nodes.get()" v-if="par.Sex != nodes.get(active_edge.from).Sex & par.id != nodes.get(active_edge.to).id" v-bind:value="par.id">
							</datalist>
						</td>
					</tr>
					<tr v-show="active_edge['Breeding Type'] == 'Repeat'">
						<td>Number of Repeat 
							<span class="tooltip">&#9432 <span class="tooltiptext">Number of times the Repeat is executed. The parent node will be the new input of the child node and everything in between is rerun. Nodes will be named the same + _1, _2, _3 etc.</span></span>
						</td> 
						<td> 
							<input list="indVar" placeholder="Type a number" v-model="active_edge['Number of Repeat']" onchange="updateVar('Number of Repeat', value)">
						</td>
					</tr>
					<!--tr v-show="active_edge['Breeding Type'] == 'Recombination'">
						<td>New Mutation Rate </td> 
						<td> <input onchange="updateVar('New Mutation Rate', value)" placeholder="Type a number" v-model="active_edge['New Mutation Rate']" list="indVar"></td>
					</tr>
					<tr v-show="active_edge['Breeding Type'] == 'Recombination'">
						<td>New Remutation Rate </td> 
						<td> <input onchange="updateVar('New Remutation Rate', value)" placeholder="Type a number" v-model="active_edge['New Remutation Rate']" list="indVar"></td>
					</tr>
					<tr v-show="active_edge['Breeding Type'] == 'Recombination'">
						<td>Number of Recombination per M </td> 
						<td> <input onchange="updateVar('Number of Rec per M', value)" placeholder="Type a number" v-model="active_edge['Number of Rec per M']" list="indVar"></td>
					</tr -->
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 ">
						<td>Selection Type
							<span class="tooltip">&#9432 <span class="tooltiptext">Random: Each individual is taken with same probability, BVE/Pheno selected based on BVE/Phenotypes</span></span>
						</td>
						<td> 
							<select v-model="active_edge['Selection Type']">
								<option v-for="option in selectionType_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE'">
						<td>BVE Method
							<span class="tooltip">&#9432 <span class="tooltiptext">Direct-Mixed-Model assumes known heritability and thereby is fastest. EMMREML/sommer use the respective R-package. Sommer also support multi-variable models but takes much longer! Bayesian models used are using BGLR. MAS is using lm() on randomly sampled effect markers.</span></span>
						</td>
						<td> 
							<select v-model="active_edge['BVE Method']">
								<option v-for="option in BVEMethod_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE' & active_edge['BVE Method']=='Marker assisted selection (lm)'">
						<td>Number of markers used for MAS
							<span class="tooltip">&#9432 <span class="tooltiptext">This is a very simple implementation of MAS using a linear model of randomly sampled effect markers</span></span>
						</td>
						<td><input placeholder="Type a number or Inf" v-model="active_edge['MAS_marker']"></td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 & active_edge['Selection Type']=='Pseudo-BVE'" v-for='(trait, index) in traitsinfo' >
						<td>Accuracy for Trait {{ trait[["Trait Name"]] }} 
							<span class="tooltip">&#9432 <span class="tooltiptext">If nothing is entered here the accuracy will be chosen according to the starting heritability</span></span>
						</td>
						<td>
							<input v-model="active_edge[index]" style="width:80px;"> 
						</td>
					</tr>					
					
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 & traitsinfo.length > 1">
						<td>Selection Index
						<span class="tooltip">&#9432 <span class="tooltiptext">Select which index to use for BVE/Phenotypic selection</span></span>
						</td>
						<td> 
							<select v-model="active_edge['Selection Index']">
								<option v-for="option in selection_index" v-bind:value="option.Name">
								{{ option.Name }}
								</option>
							</select>
						</td>
					</tr>						
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0">
						<td>Selection Proportion
							<span class="tooltip">&#9432 <span class="tooltiptext">Selection intensity - the tool will automatically calculate this for you based on the size of the parent/child node.</span></span>
						</td>
						<td>
							<!-- <input list="indVar" onchange="updateVar('Selection Proportion', value)" placeholder="Type a number" min="0" max="1" step="0.001" v-model="selection_proportion"> -->
							{{ selection_proportion }} (calculated based on # Individuals)
						</td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>Threshold for selection
							<span class="tooltip">&#9432 <span class="tooltiptext">Calculate reliability of BVE according to vanRaden 2008</span></span>
						</td>		
						<td><input placeholder="Type a number if wanted" v-model="active_edge['threshold']"></td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>Threshold sign
							<span class="tooltip">&#9432 <span class="tooltiptext"> Only keep individuals with higher/lower/eval estimated breeding value</span></span>
						</td>		
						<td> 
							<select v-model="active_edge['threshold_sign']">
								<option v-for="option in threshold_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type']=='BVE' & geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>Calculate reliability
							<span class="tooltip">&#9432 <span class="tooltiptext">Calculate reliability of BVE according to vanRaden 2008</span></span>
						</td>		
						<td><input type="checkbox" false-value="No" true-value="Yes" v-model="active_edge['reliability']"> </td>
					</tr>
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced']">
						<td>Input Phenotype
							<span class="tooltip">&#9432 <span class="tooltiptext">Use the avg. phenotype of the offspring as the phenotype of the individuals in the parent node. Make sure there are offspring! (use of time needed!)</span></span>
						</td>
						<td>
						<select v-model="active_edge['phenotype_used']">
						
								<option v-for="option in phenotype_options" v-bind:value="option">
								{{ option }}
								</option>
								
							</select>
						</td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE'">
						<td>Relationship Matrix
							<span class="tooltip">&#9432 <span class="tooltiptext">According to vanRaden 2008, pedigree-based or singleStep (H based on Legarra 2014)</span></span>
						</td>
						<td> 
							<select v-model="active_edge['Relationship Matrix']">
								<option v-for="option in RelationshipMatrix_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE' && ( active_edge['Relationship Matrix']== 'Single Step'|active_edge['Relationship Matrix']== 'Pedigree')">
						<td>Depth of Pedigree (# generation back)
							<span class="tooltip">&#9432 <span class="tooltiptext">Depth of the pedigree. All individuals before are assumed to be unrelated! If you need something else contact Torsten or use the R-package.</span></span>
						</td> 
						<td> <input onchange="updateVar('Depth of Pedigree', value)" placeholder="Type a number" v-model="active_edge['Depth of Pedigree']" list="indVar"></td>
					</tr>

					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE'">
						<td>Cohorts used in BVE
							<span class="tooltip">&#9432 <span class="tooltiptext">Manual select provides you with maximum flexibility to select which individuals to use.</span></span>
						</td>
						<td> 
							<select v-model="active_edge['Cohorts used in BVE']">
								<option v-for="option in Cohorts_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE' && active_edge['Cohorts used in BVE'] == 'Manual select'">
						<td>Select cohorts for manuel select:
							<span class="tooltip">&#9432 <span class="tooltiptext">Cohorts available for selection are only refresh on reloading or when EDITING an edge. We are working on that.</span></span>
						</td>
						<td> 
							<treeselect :value-consists-of="'LEAF_PRIORITY'" :multiple="true" :options="node_manualselect_options" v-model="active_edge['Manuel selected cohorts']" :searchable="true"/>
							<!--select v-model="active_edge['Manuel selected cohorts']" size=6 multiple>
								<option v-for="option in nodes.getIds()" v-bind:value="option">
								{{ option }}
								</option>
							</select-->
						</td>
					</tr>
				</table>
				<br>
				<input class='button_type' type="button" value="save" id="edge-saveButton" />
				<input class='button_type' type="button" value="cancel" id="edge-cancelButton" />
			</div>
			<br>
        	
        	<br>
        	<br>	
			<div v-if = "geninfo['Project Name']" id="Results_Div">
				<h3 id="RSim"></h3>
				<br>
				<br>
				<h3>Analyze Breeding Program with R</h3>
				<p>Click on button 'Start R Simulation' to run R.</p>
				<button onclick="runningR1(); " >Start R Simulation</button>
				<p>Or click here to restore the latest R result that you have already generated for the breeding program {{geninfo['Project Name']}}.</p>
				<button onclick="ReloadSim(); " >Reload Old Results</button>	
				&nbsp;&nbsp;
				
				<p> <a v-if="filename" href="/Rdownload"> Download {{geninfo['Project Name']}}.RData </a> </p>
				<p> <a v-if="filename" href="/Exceldownload"> Download {{geninfo['Project Name']}}.csv </a> </p> 

				<table>
				<tr>
					<td> Cohorts Information :
						<span class="tooltip">&#9432 <span class="tooltiptext">List of all cohorts and some basic information (time of generation, economic stuff etc.)</span></span>
					</td>
					<td>
						<textarea v-model="geninfo['listOfCohorts_withInfo']" placeholder="add multiple lines" id="listOfCohorts" style="width:450px;height:50px;"></textarea>						
					</td>
				</tr>
				</table>
				
				<p>Attention: The actual breeding program must match the uploaded R results, otherwise plot functions will not work.</p>
				<form id='ResultUpload' target="print_popup" action='/ResultUpload' method='post' encType='multipart/form-data' onsubmit="window.open('about:blank','print_popup','width=1000,height=800')">
					<input type='file' name='ResultFile' id='ResultFile'>
					<!--input type='submit' value='Load Results'-->
					<button onclick="UploadSim();">Load Results</button>
				</form>
				<p v-show="runned">Click here to clear all results and plots.</p>
				<button v-show="runned" onclick="clearResult()" >Clear Results</button>
				<button v-show="runned" v-on:click="rlog = !rlog" >Show/Hide R Log</button>
				<br>
				<br>
				<div v-show="runned & rlog" id="Rout_Div">

				</div>
				<br>
				<br>
				<div v-show="runned">
					<h4 style="color: 'red'">Results: Summary</h4>
					<div id="Result_Summary">
					</div>
					<br>
					<br>
					
					
					<h4> Download population list </h4>	
					<a href="/Rdownload"> Download RData</a>
					<br>
					<br>
					<h4>Download: Population data</h4>
					<td> Selected datatype: </td>
					<select v-model="plottingPar.download1" >
						<option v-for="option in download_data" v-bind:value="option">
						{{ option }}
						</option>
					</select>
					<br>
					<td> Selected cohort: </td>
					<select v-model="plottingPar.download2" >
						<option v-for="option in nodes.get()" v-bind:value="option">
						{{ option.id }}
						</option>
					</select>
					<br>
					<td> Selected repeat: </td>
						<input placeholder="Type a number (Default: 0)" v-model="plottingPar.download3">
					<br>
					
					<button onclick="RunDataPreparation()">Prepare data</button>
					<a v-if="plottingPar.download1=='Ped'" href="/peddownload"> Download Ped</a>
					<a v-if="plottingPar.download1=='Map'" href="/mapdownload"> Download Map</a>
					<a v-if="plottingPar.download1=='VCF'" href="/vcfdownload"> Download VCF</a>
					<a v-if="plottingPar.download1!='Ped' && plottingPar.download1!='Map' && plottingPar.download1!='VCF'" href="/txtdownload"> Download txt</a>					
					
					<br>
					<br>
					
					<h4>Results: Observed Phenotypes</h4>
					<button v-if="plottingData.RespMean == ''" onclick="RunResultpMean()" >Run Analysis</button>
					<template v-if="plottingData.RespMean != ''">
						Select plotting type:
						<select v-model="plottingPar.RespMean_pType" >
							<option v-for="option in plottingType" v-bind:value="option">
							{{ option }}
							</option>
						</select>
						<br>
						Select cohorts (multiple selection possible): 
						<button onclick="plottingResultpMean()" >Plot Results</button>
						<treeselect :multiple="true" :options="Summary" v-model="plottingPar.RespMean_cohorts" :searchable="true" style="width:800px"/>
						<!--select v-model="plottingPar.RespMean_cohorts" size="6" multiple>
							<option v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
							{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
							</option>
						</select-->
					</template>
					<br>
					<br>
					<div v-show="plottingData.RespMean != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
						<div v-for="(trait, index) in traitsinfo" v-bind:id="'RespMean_Div'+(index+1)">
						</div>
					</div>
					<br>
					<br>
					<h4>Results: True Breeding Values</h4>
					<button v-if="plottingData.ResgMean == ''" onclick="RunResultgMean()" >Run Analysis</button>
					<template v-if="plottingData.ResgMean != ''">
						Select plotting type:
						<select v-model="plottingPar.ResgMean_pType" >
							<option v-for="option in plottingType" v-bind:value="option">
							{{ option }}
							</option>
						</select>
						<br>
						Select cohorts (multiple selection possible): 
						<!--select v-model="plottingPar.ResgMean_cohorts" size="6" multiple>
							<option v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
							{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
							</option>
						</select-->
						<button onclick="plottingResultgMean()" >Plot Results</button>
						<treeselect :multiple="true" :options="Summary" v-model="plottingPar.ResgMean_cohorts" :searchable="true" style="width:800px"/>
					</template>
					<br>
					<br>
					<div v-show="plottingData.ResgMean != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
						<div v-for="(trait, index) in traitsinfo" v-bind:id="'ResgMean_Div'+(index+1)">
						</div>
					</div>
					<br>
					<br>
<!--
					<h4>CURRENTLY INACTIV - Results: Accuracy of Breeding Value Estimation</h4>
					<button v-if="plottingData.ResAccBVE == ''" onclick="RunResultAccBVE()" >Run Analysis</button>
-->
					<template v-if="plottingData.ResAccBVE != ''">
						Select plotting type:
						<select v-model="plottingPar.ResAccBVE_pType" >
							<option v-for="option in plottingType" v-bind:value="option">
							{{ option }}
							</option>
						</select>
						<br>
						Select cohorts (multiple selection possible):
						<button onclick="plottingResultAccBVE()" >Plot Results</button>
						<treeselect :multiple="true" :options="Summary_AccBVE" v-model="plottingPar.ResAccBVE_cohorts" :searchable="true" style="width:800px"/>
					</template>

					<div v-show="plottingData.ResAccBVE != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
						<div v-for="(trait, index) in selection_index" v-bind:id="'ResAccBVE_Div'+(index+1)">
						</div>
					</div>
					<br>
					<br>

					<h4> Results: Relationship and Inbreeding within Cohorts </h4>
					<button v-if="plottingData.ResRel == ''" onclick="RunResultRel()" >Run Analysis</button>
					<template v-if="plottingData.ResRel != ''">
						Select plotting type:
						<select v-model="plottingPar.ResRel_pType" >
							<option v-for="option in plottingType" v-bind:value="option">
							{{ option }} 
							</option>
						</select>
						<br>
						Select cohorts (multiple selection possible):
						<!--select v-model="plottingPar.ResRel_cohorts" size="6" multiple>
							<option v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
							{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
							</option>
						</select-->
						<button onclick="plottingResultRel()" >Plot Results</button>
						<treeselect :multiple="true" :options="Summary" v-model="plottingPar.ResRel_cohorts" :searchable="true" style="width:800px"/>
					</template>
					<br>
					<br>
					<div v-show="plottingData.ResRel != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
						<div id="ResRel_Div1">
						</div>
						<div id="ResRel_Div2">
						</div>
					</div>
					<br>
					<br>
<!--
					<h4> CURRENTLY INACTIV - Results: Relationship between Cohorts </h4>
					<button v-if="plottingData.ResRelbetweenC == ''" onclick="RunResultRelbetweenC()" >Run Analysis</button>
-->
					<template v-if="plottingData.ResRelbetweenC != ''">
						Select cohort 1 :
						<select v-model="plottingPar.ResRelbetweenC_cohort1">
							<option v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
							{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
							</option>
						</select>
						Select cohort 2 :
						<select v-model="plottingPar.ResRelbetweenC_cohort2">
							<option v-if="option != plottingPar.ResRelbetweenC_cohort1" v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
							{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
							</option>
						</select>
						<button onclick="plottingResultRelbetweenC()" >Plot Results</button>
					</template>


					<div v-show="plottingData.ResRelbetweenC != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
						<div id="ResRelbetweenC_Div1">
						</div>
					</div>
					<br>
					<br>

					<h4> Results: Major QTLs (Allele Frequency, exp./obs. Heterozygosity)</h4>
					<button v-if="plottingData.ResQTL == ''" onclick="RunResultQTL()" >Run Analysis</button>
					<template v-if="plottingData.ResQTL != ''">
						Select Trait :
						<select v-model="plottingPar.ResQTL_trait">
							<option v-if="option['Trait Major QTL'] > 0" v-for="(option, index) in traitsinfo" v-bind:value="index+1">
							{{ option["Trait Name"] }} 
							</option>
						</select>
						Select QTL :
						<select v-if="plottingPar.ResQTL_trait != ''" v-model="plottingPar.ResQTL_qtl">
							<option v-for="(option, index) in traitsinfo[plottingPar.ResQTL_trait -1]['Trait QTL Info']" v-bind:value="index">
							{{ option['QTL ID'] }}  
							</option>
						</select>
						<br>
						Select Cohorts (multiple selection possible):
						<!--select v-model="plottingPar.ResQTL_cohorts" size="6" multiple>
							<option v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
							{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
							</option>
						</select-->
						<button onclick="plottingResultQTL()" >Plot Results</button>
						<treeselect :multiple="true" :options="Summary" v-model="plottingPar.ResQTL_cohorts" :searchable="true" style="width:800px"/>
					</template>
					<br>
					<br>
					<div v-show="plottingData.ResQTL != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
						<div id="ResQTL_Div1">
						</div>
						<div id="ResQTL_Div2">
						</div>				
						<div id="ResQTL_Div3">
						</div>
					</div>
				</div>
				<br>
				<br>
			</div>
			<div>
			</div>
			<div id="Export_Data_H3">
				<span class="tooltip">&#9432 <span class="tooltiptext"><p> Download JSON file to Desktop. </p>
				<p> Display JSON data into this text area. </p></span></span>
				<button id="export_button" onclick="downloadNetwork()" >Download Project</button>
				<span class="tooltip">&#9432 <span class="tooltiptext"><p> This option can be used to manually export the JSON File in order to run the MoBPS R-package on one's own computer.</p>
				<p> You might also paste a JSON File into this text area and import it for further processing. </p></span></span>
				<button id="import_button" onclick="importNetwork()" >Import Data</button>
				<br>
				<br>
				<div>
					<textarea id='OutputArea' style="width:700px; height:200px;"></textarea>
				</div>
				<br>
				<br>
			</div>
			<!-- <div id="Export_Data_H3">
				<br>
				<br>
				<h3> Import JSON Data </h3>
				<p> This option can be used to manually export the JSON File in order to run the MoBPS R-package on one's own computer.</p>
				<p> You might also paste a JSON File into this text area and import it for further processing. </p>
				<button id="export_button" onclick="downloadNetwork()" >Download Project</button>
				<button id="import_button" onclick="importNetwork()" >Import Data</button>
				<br>
				<br>
				<div>
					<textarea id='OutputArea' style="width:700px; height:200px;"></textarea>
				</div>
				<br>
				<br>
			 </div> -->
		</div>
	</div>
	</body>
		
	<!-- Eigene Javascripte Species Data -->
	<script type="text/javascript" src="javascript/GenomeData.js"></script>
	
	<!-- Eigene Javascripte Main Code -->
	<script src="javascript/socket.io.js"></script>
	<script type="text/javascript" src="javascript/network.js"></script>
	<script type="text/javascript" src="javascript/Rmodule.js"></script>
	
	
	<!-- Eigene Javascripte Checking and additional Services -->
	<script type="text/javascript" src="javascript/Check_Services.js"></script>

	<!-- Node and Edge PopUp draggable -->
	<script type="text/javascript" src="javascript/node_edge_draggable.js"></script>
	
	<!-- Navigation Menu related -->
	<script type="text/javascript">
		function openNav() {
		  document.getElementById("mySidebar").style.width = "250px";
		  document.getElementById("main").style.marginLeft = "250px";
		  document.getElementById("openDiv").style.display = "none";
		}
		
		function closeNav() {
		  document.getElementById("mySidebar").style.width = "0";
		  document.getElementById("main").style.marginLeft= "0";
		  document.getElementById("openDiv").style.display = "block";
		}
				
	function action(hidden) {
	        if(hidden==true) {
	            document.getElementById('node-saveButton').style.visibility = 'hidden';
				document.getElementById('edge-saveButton').style.visibility = 'hidden';
	        } else {
	            document.getElementById('node-saveButton').style.visibility = 'visible';
				document.getElementById('edge-saveButton').style.visibility = 'visible';
	        }
	    }

		var acc = document.getElementsByClassName("accordion");
		var z;
		
		for (z = 0; z < acc.length; z++) {
		  acc[z].addEventListener("click", function() {
		    this.classList.toggle("active");
		    var panel = this.nextElementSibling;
		    if (panel.style.maxHeight) {
		      panel.style.maxHeight = null;
		    } else {
		      panel.style.maxHeight = panel.scrollHeight + "px";
		    } 
		  });
		}

		function isTrait_dragdrop(isDragDrop) {
			if(isDragDrop==true) {
				document.getElementById("save_matrix").style.display = "none";
		        } else {
				document.getElementById("save_matrix").style.display = "block";
		        }
		    }


		function showCorrDiv(val) {
		if (val==true) {
		    document.getElementById("corrDiv").style.display = "none";
		  } else {
		    document.getElementById("corrDiv").style.display = "block";
		  }
		}
		
		function textAreaAdjust(o) 
		    {
		        o.style.height = "1px";
		        o.style.height = (25+o.scrollHeight)+"px";
		    }

	</script>		

</html>

