<!DOCTYPE html>
<html lang="en" >
	<meta charset="utf-8"/>
	
	<script type="text/javascript">
		var isBrowserSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
	</script>
		
	<head>
		<title>MoBPS (v1.6.31)</title>
		<!-- Javascripte Online -->
		<script src="https://cdn.jsdelivr.net/npm/vue@2.6.12"></script> 

		<script src="https://unpkg.com/mathjs@5.8.0/dist/math.js"></script>
		<script src="https://cdnjs.cloudflare.com/ajax/libs/numeric/1.2.6/numeric.min.js"></script>
		
		<script src="https://cdn.jsdelivr.net/npm/@riophae/vue-treeselect@^0.4.0/dist/vue-treeselect.umd.min.js"></script>
    	<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@riophae/vue-treeselect@^0.4.0/dist/vue-treeselect.min.css">
		
		<link rel="shortcut icon" href="#">
		
		<!-- for plotting graphs: -->
		<script src="https://cdn.plot.ly/plotly-latest.min.js"></script>

		<script type="text/javascript" src="https://unpkg.com/alea@1.0.0/alea.js"></script>
		<link src="https://visjs.github.io/vis-network/dist/vis-network.min.css" rel="stylesheet" type="text/css" />
		
		<script type="text/javascript" src="https://almende.github.io/vis/dist/vis.js"></script>
		<link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/vis/4.21.0/vis.min.css" rel="stylesheet" type="text/css" />
		<link rel="stylesheet" type="text/css" href="./styles/mobps.css">
	
		<!--  for javascript and ajax call	-->
		<script type="text/javascript" src="https://code.jquery.com/jquery-2.2.0.min.js"></script> 
		<script type="text/javascript" src="javascript/jQ3.js"></script>

		<!-- for drag and drop for Phenotype table -->	
		<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/sortablejs@1.7.0/Sortable.min.js"></script>
		<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/Vue.Draggable/2.16.0/vuedraggable.min.js"></script>	
		<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.0/axios.min.js"></script>
	
		<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/liquor-tree/dist/liquor-tree.umd.js"></script>
		<link href="https://code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css" media="all" rel="stylesheet" type="text/css" />
	
	</head>
		
	<body onload="init();">
		
		<div id="runningDog">
			<button style="color: red" class='bar_button' onclick="stopR();">Stop R Simulation</button> 
			<h3 id="runningDogTitle">
			</h3>
			<div id="runningDogLog">
			</div>
		</div>
		
		<div id='mainDiv'>

			<div id="mySidebar" class="sidebar">
				<div id="closeDiv">
					<button onclick="closeNav()" id="closeB" class="closebtn"> Navigation Close </button>
				</div>  
				<a>&nbsp;</a>
					<a href="#General_Info">General information</a>
					<a href="#Phenotype_Info_Div">Phenotype information</a>
					<a href="#Selectionindex_Div">Optional information</a>
					<a href="#Breeding_Scheme">Breeding Scheme</a>
					<a href="#Results_Div">R Simulation/ Results</a>
					<a href="#Export_Data_H3">Export/ Import</a>
					<a v-if="geninfo['curUserGroup'] != 1" href="/Change_Password"> Change Password</a>
					<a href="/logout"> User: {{user}} <br> <br> Logout</a>
					<br>
					<a v-if="geninfo['curUserGroup'] >= 4" href="/users"> Access to Database</a>
			</div>			

			<div class="navlinks" >
				<button onclick="openNav()" id="openB" class="openbtn"> Navigation Open </button> 
				<button class="invisible_button"></button>
				<button class='bar_button' id="new_button" onclick="window.location.reload();">NEW</button> 
				<button v-if="geninfo['sharedWith'] === '' " class='bar_button' id="save_button" onclick="saveProject();">SAVE</button> 
				<button v-if="geninfo['sharedWith'] !== '' " class='bar_button' id="save_button" onclick="share_SaveProject();">SAVE</button>
				<button class='bar_button' onclick="saveProjectAs();">SAVE AS</button> 
				<a v-if="geninfo['isPrShared'] === false && geninfo['isSharedExist'] === '' " class='bar_button' onclick="shareProjectWith(geninfo['Project Name']);"> SHARE </a>
				<a v-if="geninfo['isPrShared'] === true && geninfo['isSharedExist'] === 'Yes'  " class='bar_button' onclick="isShareUserExists(data_Vue.geninfo['sharedWith']);"> STOP SHARE </a>				<input type="button" class="fileType" id="importProject" value="IMPORT" class='bar_button' onclick="document.getElementById('input-file').click();importNetwork()" />

				<input type="file" style="display:none;" id="input-file" name="file"/>
				<button class='bar_button' id="download_button" onclick="downloadNetwork()" >EXPORT</button>
				<button class='bar_button' id="delete_button" v-if="editProjectForTree !=='' " onclick="deleteProject();deleteProject_FromTree();" >DELETE</button>
				<button class='showhide_button' v-on:click='show_geninfo = !show_geninfo; show_warnings = false;' >Show/Hide Info</button>
				<button class='showhide_button' v-on:click='show_warnings = !show_warnings; show_geninfo=false;' >Show/Hide Warnings</button>
				&nbsp;
				<a :href="'https://mobps.de/compareProjects?cpuser=' + geninfo['user']+'&cpusergroup='+geninfo['curUserGroup']" class ="additional_link" > Compare Projects</a>
			</div>
			
			<div id='shareProject_Div'>
			    <p>
			      <input type="button" class="button_type" id="addRow" value="Add User" onclick="addRow()" />
			    </p>
											<br>
			    <div id="cont"></div>
				<br>
			      <p>
			        <input type="button" class="button_type" value="Submit" onclick="shareUsersubmit()" />
					<input type="button" value="Cancel" class="button_type" onclick="document.getElementById('shareProject_Div').style.display = 'none'"/>
			      </p>

			</div>
			<div id='SaveAs_Div'>
				New Project Name: <input id="saveAs-name" type="text" placeholder="Type a name"/> <br><br>
				<input type="button" value="Save" id="SaveAs-saveButton" class="button_type" onclick="saveProject(document.getElementById('saveAs-name').value)" />
				<input type="button" value="Cancel" id="SaveAs-cancelButton" class="button_type" onclick="document.getElementById('SaveAs_Div').style.display = 'none'"/>
			</div>
			<div id="warnings_sum">
				<h4 v-if="warnings.length > 0" style="color: red; font-style:italic;"> Attention, there are {{warnings.length}} warnings! R Simulation most likely cannot be run unless they are fixed.</h4>
			</div>
			<transition name="slide-generalInfo">
				<div id ="warnings" v-show='show_warnings' >
					<br>
					<span style="font-weight:bold;">Warnings  </span>
					<br><br>
					<span v-for='(war, index) in warnings'> Warning {{index}} : <span style="color: red">{{war}} <br></span></span>
				</div>
			</transition>
			<transition name="slide-generalInfo">
				<div id ="generalInfo" v-show='show_geninfo' >
					<br>
					<br>
					<br>
					<br>
					<br>
					<br>
					<span v-if="show_info =='general'">
						<span style="font-weight:bold;">General Info  </span>
						<br><br>
						<span v-if="key != 'Ensembl Filter' & key != 'Ensembl Filter Values' & value != ''" v-for='(value,key) in geninfo'> {{key}} : <span style="color: red">{{value}}  <br></span></span> 
					</span>
					
					<span v-if="show_info =='node'">
						<span style="font-weight:bold;">Node Info  </span>
						<br><br>
						<span v-if="['x','y', 'individualsVar', 'label','color','title'].indexOf(key) == -1 && value != '' && value != []" v-for='(value,key) in active_node'> {{key}} : <span style="color: red">{{value}} <br></span></span>
					</span>
					<span v-if="show_info =='edge'">
						<span style="font-weight:bold;">Edge Info  </span>
						<br><br>
						<span v-if="['useVar','id','color','title'].indexOf(key) == -1 && value != '' && value != []" v-for='(value,key) in active_edge'> {{key}} : 
							<span style="color: red">{{value}} 
								<span v-if="active_edge.useVar.indexOf(key)>-1"> ({{individualsVar_options[active_edge.useVar.indexOf(key)].name}})</span><br>
							</span>
						</span>
					</span>
				</div>
			</transition>			

		<div id="main">		
		
			<div id="LoadJSON_Div" >
				<h2 v-if="geninfo['Project Name'] == '' " > MoBPS </h2>
				<h2 v-if="geninfo['Project Name'] !== '' " > MoBPS &nbsp;&nbsp;  {{geninfo['Project Name']}} </h2>
				<h4 style="margin-left:125px" v-if="(geninfo['sharedWith'].length > 0 )" > {{geninfo['sharedWith'] }} </h4>	
			
			<h3>Load a new/existent project from your own database:</h3>
			
			<h3 v-if="database.length == 0"> Welcome to MoBPS. Please create a project!</h3>
				
			<table id="load_table">
				<tr>
					<td>
					Exemplary Templates: 						
					<select onchange="loadTemplate(value)" >
						<option value="Create_New_12345XXX"></option>
						<option v-for="(option,index) in template_database" v-bind:value="index">
							{{ option.name }}
						</option>
					</select>
					</td>
				</tr>					
				<tr v-if="database.length >0">
					<td>
					&nbsp;&nbsp; Version: 
					<select id="Version" onchange="loadVersion(value)">
						<option value="recent">(most recent)</option>
						<template v-if="versions.length >0">
						<option v-for="(option, index) in versions" v-bind:value="index">
							{{ option.date }}
						</option>
						</template>
					</select>
					</td>
				</tr>
			</table> 
			
			<button class="accordion"> Click here to show guidelines for grouping the MoBPS Projects. </button>

			<div class="guideline">
				<p class="guidelines"> 
					-- If you are a new MoBPS user, there will not be any projects to display in the grouping projects section. <br>
					&nbsp; &nbsp; &#8226 You could scroll down to General Information section to enter the columns and Save.  <br>
					-- If you are an existing user and already have projects on your own, All existing projects would be under the Common_Projects folder. <br>
					&nbsp; &nbsp; &#8226 You can create a folder and group your projects<br>
					-- Edit Group : Click Edit and change your own group name and enter to update.<br>
					-- Remove Group: Click Remove to remove projects under this folder.<br>
				 	-- Add Sub Folder: Click Add Sub Folder, the sub-folder will be added just below to the group.<br> 
					&nbsp; &nbsp; &#8226 You can drag and drop projects to the newly created folder (or) <br>
					&nbsp; &nbsp; &#8226 you can drag the folder to keep it as a main group too!  <br>
					-- Edit/display the project - Click your project from your folder then the project is in Edit mode.
				</p>
			</div>

			<div v-if="(groupProjectCnt>0 && groupProjectCnt < 10) || groupProjectCnt == '' " id="projectListDiv" >				
				<div class="examples">
			        <div class="example">
	 					<div class="example-description">
	            			<input type="text" placeholder="Enter Project to filter..." v-model="filter" class="filter-field">
	          			</div>
			         <div class="example-tree">
						<tree :v-if="treeData" :data="treeData" :options="treeOptions"  :filter="filter" v-model="treeModel" ref="tree" @node:checked=""
							@node:dragging:start="logDragStart" @node:clicked="nodeClicked"  @node:dragging:finish="logDragFinish" >
			              	<div slot-scope="{ node }"  class="node-container" >
			 					<div class="node-text">  
									<i v-if="node['states'].dropable == true" :class="[node.expanded() ? 'ion-android-folder-open' : 'ion-android-folder']"></i>
									{{ node.text }}
								</div>
				 				<div v-if="node['states'].dropable == true" >
									<a href="#" @mouseup.stop="editNode(node)"> Edit |</a>
									<a href="#" @mouseup.stop="saveNode(node)"> Save |</a>
						        	<a href="#" @mouseup.stop="removeNode(node)">Remove |</a> 
						        	<a href="#" @mouseup.stop="addChildFolder(node)">Add Folder </a>
			               		</div>
			              	</div>
			            </tree>	
			          </div>				
					 </div>
		        </div>
			</div>	
			
			<div v-else-if="groupProjectCnt >=10 && groupProjectCnt < 35" id="projectListDiv" style="height:700px">				
				<div class="examples">
			        <div class="example">
	 					<div class="example-description">
	            			<input type="text" placeholder="Enter Project to filter..." v-model="filter" class="filter-field">
	          			</div>
			         <div class="example-tree">
						<tree :v-if="treeData" :data="treeData" :options="treeOptions"  :filter="filter" v-model="treeModel" ref="tree" @node:checked=""
							@node:dragging:start="logDragStart" @node:clicked="nodeClicked"  @node:dragging:finish="logDragFinish" >
			              	<div slot-scope="{ node }"  class="node-container" >
			 					<div class="node-text">  
									<i v-if="node['states'].dropable == true" :class="[node.expanded() ? 'ion-android-folder-open' : 'ion-android-folder']"></i>
									{{ node.text }}
								</div>
				 				<div v-if="node['states'].dropable == true" >
									<a href="#" @mouseup.stop="editNode(node)"> Edit |</a>
									<a href="#" @mouseup.stop="saveNode(node)"> Save |</a>
						        	<a href="#" @mouseup.stop="removeNode(node)">Remove |</a> 
						        	<a href="#" @mouseup.stop="addChildFolder(node)">Add Folder </a>
			               		</div>
			              	</div>
			            </tree>	
			          </div>				
					 </div>
		        </div>
			</div>	
			
			<div v-else="groupProjectCnt >=35" id="projectListDiv" style="height:900px">				
				<div class="examples">
			        <div class="example">
	 					<div class="example-description">
	            			<input type="text" placeholder="Enter Project to filter..." v-model="filter" class="filter-field">
	          			</div>
			         <div class="example-tree">
						<tree :v-if="treeData" :data="treeData" :options="treeOptions"  :filter="filter" v-model="treeModel" ref="tree" @node:checked=""
							@node:dragging:start="logDragStart" @node:clicked="nodeClicked"  @node:dragging:finish="logDragFinish" >
			              	<div slot-scope="{ node }"  class="node-container" >
			 					<div class="node-text">  
									<i v-if="node['states'].dropable == true" :class="[node.expanded() ? 'ion-android-folder-open' : 'ion-android-folder']"></i>
									{{ node.text }}
								</div>
				 				<div v-if="node['states'].dropable == true" >
									<a href="#" @mouseup.stop="editNode(node)"> Edit |</a>
									<a href="#" @mouseup.stop="saveNode(node)"> Save |</a>
						        	<a href="#" @mouseup.stop="removeNode(node)">Remove |</a> 
						        	<a href="#" @mouseup.stop="addChildFolder(node)">Add Folder </a>
			               		</div>
			              	</div>
			            </tree>	
			          </div>				
					 </div>
		        </div>
			</div>	
			
			<h3 v-if="curUserGroup =='4'"> You are assigned to User Class Admin. 
				This enables you to use 20 Cores, 40 GB Max-Memory and maximum run time of 120 hours 
			</h3>
			<h3 v-if="curUserGroup =='3'"> You are assigned to User Class Professional. 
				This enables you to use 5 Cores, 30 GB Max-Memory and maximum run time of 48 hours 
			</h3>
			<h3 v-if="curUserGroup =='2'"> You are assigned to User Class Student. 
				This enables you to use 5 Cores, 20 GB Max-Memory and maximum run time of 4 hour 
			</h3>
			<h3 v-if="curUserGroup =='1'"> You are assigned to User Class Test. 
				It is not possible to use our backend-server for simulation. To simulate download the json-file run via json.simulation() in R.	</h3>
			

			<button class="accordion"> Click here to show some basic guildelines on the use of the MoBPS web-interface </button>
				<div class="guideline">
					<p class="guidelines"> 
						&#8226 Enter your breeding program in the following modules. <br>
						&#8226 &#9432 buttons indicate additional information when hovering over.<br>
						&#8226 Use the navigation bar to jump between modules, change your password, etc.<br>
						&#8226 In case our server resources are not enough for you please press the Export button<br>
					 	&#8226 You can run simulation on your own device in R via: json.simulation("FILENAME.json")<br>
						&#8226 Script with uploaded genotype / maps will only run on our server!<br>
					 	&#8226 We are happy to help you set up your own MoBPS web-server!<br>
						&#8226 Basic tutorials are in preparation and can be found at: https://www.youtube.com/channel/UC4LDcBka39NidOF1y_65FFw<br>
						&#8226 For advanced questions, bug reports, potentially collaborations  contact me! (Torsten Pook / torsten.pook@uni-goettingen.de)
					</p>
				</div>
				
			<div id="General_Info" v-on:click="show_info ='general'" onchange="checkEverything(this.id)">
				<h3> General Information</h3> 
				<button class="accordion"> Click here to read guidelines for the General information </button>
					<div class="guideline">
						<p class="guidelines"> 
							--In this module you can enter basic information on the species you are working on <br>
							--You can select from provided maps (via Ensembl / MoBPSmaps) or upload you own .map-file: <br>
							--Columns for a customize upload are: Chromosome, SNP-name, physical position in base-pairs, position in Morgan (if NA -> bp / 100.000.000), allele frequency (optional)<br>
							--Most additional modules on default are hidden! Press Advanced settings to unlock more complicated options 
						</p>
					</div>
					
				<table id="geninfo_table">
					<colgroup><col width="40%">	<col width="60%"></colgroup>
					<tr>
						<td>Project Name
							<span class="tooltip">&#9432 <span class="tooltiptext">Enter the name of your project. You can save/copy/delete your project via the action bar and load different version to return to.</span></span>
						</td>
						<td>
							<input placeholder="Type a name" type="text" v-model="geninfo['Project Name']">
						</td>
					</tr>
					<tr>
						<td style="font-weight:bold">Advanced settings </td> 
						<td>
							<input type="checkbox" v-model="geninfo['advanced']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced']">
						<td>Test-Mode / Size Scaling
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock the scaling of node sizes and ignore imported genotypes.</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_test']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced']">
						<td style="font-weight:bold">Advanced Trait modelling
							<span class="tooltip">&#9432 <span class="tooltiptext"> Activating this will unlock use of dominant and epistatic trait architectures.</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_trait']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_trait'] ">
						<td  style="text-indent:20px;">Non-additive effects
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will enable generation of non-additive QTL effect </span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_trait_epi']" style="vertical-align: 75%;">
						</td>
					</tr>						
					<tr v-if="geninfo['advanced'] & geninfo['advanced_trait'] ">
						<td  style="text-indent:20px;">Repeatability
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will enable to set a repeatability unequal to heritability </span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_trait_repeat']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_trait'] ">
						<td  style="text-indent:20px;">Maternal / paternal effects
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will enable modelling of maternal / paternal effects </span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_trait_maternal']" style="vertical-align: 75%;">
						</td>
					</tr>						
					<tr v-if="geninfo['advanced'] & geninfo['advanced_trait'] ">
						<td  style="text-indent:20px;"> Traits as combination of other traits
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will enable modelling features so traits are modelled as linear combinations of other traits </span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_trait_combi']" style="vertical-align: 75%;">
						</td>
					</tr>						
					<tr v-if="geninfo['advanced'] & geninfo['advanced_trait'] ">
						<td  style="text-indent:20px;"> Transformation function
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will enable application of transformation functions on trait phenotypes </span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_trait_trafo']" style="vertical-align: 75%;">
						</td>
					</tr>												
					<tr v-if="geninfo['advanced'] & geninfo['advanced_trait'] ">
						<td  style="text-indent:20px;"> Trait rescaling
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will enable rescaling of traits after a couple of simulated generations </span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_trait_scaling']" style="vertical-align: 75%;">
						</td>
					</tr>				
					<tr v-if="geninfo['advanced']">
						<td> LD build-up Module
						<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock a module to simulated genomic data with baseline LD as a starting point of the simulation</span></span>
						</td>
						<td> 
						<input type="checkbox" v-model="geninfo['advanced_ld']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced']">
						<td>Culling Module 
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock the culling module to simulate the death of individuals. Death individuals can still be used for BVE but not for Reproduction</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_culling']" style="vertical-align: 75%;">
						</td>
					</tr>					
					<tr v-if="geninfo['advanced']">
						<td>Subpopulation Module 
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock the specification of subpopulation with different allele frequencies and trait architectures.</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_sub']" style="vertical-align: 75%;">
						</td>
					</tr>				
					<tr v-if="geninfo['advanced']">
						<td>Economic Module 
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock the use of economic parameters.</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_eco']" style="vertical-align: 75%;">
						</td>
					</tr>					
					<tr v-if="geninfo['advanced']">
						<td>Population History 
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating provide you with additional features for the simulation of thousands of generations</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_history']" style="vertical-align: 75%;">
						</td>
					</tr>					
					<tr v-if="geninfo['advanced']">
						<td>Litter size Module
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating provide you with additional features to control the litter size</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_litter']" style="vertical-align: 75%;">
						</td>
					</tr>					
					<tr v-if="geninfo['advanced']">
						<td>Modify multiple nodes/edges
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating provide you with additional features to modifiy multiple nodes / edges simultaneously </span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_multi']" style="vertical-align: 75%;">
						</td>
					</tr>				
					<tr v-if="geninfo['advanced']">
						<td style="font-weight:bold">Advanced Edge/Node options
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges </span></span>
						</td>
						<td> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced'] ">
						<td  style="text-indent:20px;">Share genotyped
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges </span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_geno']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">Max offspring
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_off']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">Avoid Half/Fullsib matings
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_mating']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">OGC
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_ogc']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">Selection ratio
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_ratio']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">Threshold selection
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_thres']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">Advanced input phenotype
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_pheno']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">Skip BVE
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges </span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_skip']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">Calculate reliability
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_reli']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">Use last available
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_last']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">Delete data
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_del']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;">Ignore Size scaling
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_scaling']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] & geninfo['advanced_advanced']">
						<td style="text-indent:20px;"> Copy settings from other nodes/edges
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will unlock additional options for nodes/edges</span></span>
						</td>
						<td style="text-indent:20px;"> 
							<input type="checkbox"  v-model="geninfo['advanced_advanced_copy']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced']">
						<td>miraculix-active
							<span class="tooltip">&#9432 <span class="tooltiptext">Deactivating this will increase run-times but lead to RData population that can be analyzed without miraculix/RandomFieldsUtils</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_miraculix']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced']">
						<td>Parallel Computing + Multiple Simulation
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will allow to run a simulation multiple times.</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_parallel']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr v-if="geninfo['advanced']">
						<td>Export/Import Box 
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating provide you with a text box for import/export </span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="geninfo['advanced_box']" style="vertical-align: 75%;">
						</td>
					</tr>
					<tr>										
						<td>Species
							<span class="tooltip">&#9432 <span class="tooltiptext">Select the species. This will unlock the correct Ensembl Datasets etc. Use a template to start with already entered somewhat realistic values.</span></span>
						</td>
						<td> 
							<select v-model="geninfo.Species">  <!-- onchange="updateSpeciesData(value)">  -->
								<option v-for="option in Species_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr>
						<td>Time Unit
							<span class="tooltip">&#9432 <span class="tooltiptext">Select the time unit used. Time of generation for each cohort will be displayed.</span></span>
						</td>
						<td>
							<select v-model="geninfo['Time Unit']">
								<option v-for="option in Time_Unit_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>					
					<tr>
						<td>Genetic Data 
							<span class="tooltip">&#9432 <span class="tooltiptext">Select the genome you want to work with.</span></span>
						</td>
						<td>
						<input type="radio" id="ens" value="Ens" v-model="genetic_data">
						<label for="ens">Use Ensembl Map</label> <br>
						<input type="radio" id="own" value="Own" v-model="genetic_data">
						<label for="own">Upload Own Map (vcf/plink)</label> <br>
						<input type="radio" id="cus" value="Cus" v-model="genetic_data">
						<label for="cus">Create customized Map</label>
						</td>
					</tr>	
					<tr v-if="genetic_data == 'Ens'">
						<td>Ensembl Dataset 
							<span class="tooltip">&#9432 <span class="tooltiptext">Currently only the genomic maps available in MoBPSmaps are provided. Alternatively upload your own map or provide us with the map you want us to add in (torsten.pook@uni-goettingen.de).</span></span>
						</td>
						<td v-if="geninfo.Species != ''">
							<select v-model="geninfo['Ensembl Dataset']">
								<option v-for="option in ensembl_options[geninfo.Species]" v-bind:value="option.Value" >
								{{ option.Value }}
								</option>
							</select>
						</td>
						<td v-else>
							Please choose a species first.
						</td>
					</tr>	
					<tr v-if="genetic_data == 'Ens'">
						<td>Max. Number of SNPs 	
							<span class="tooltip">&#9432 <span class="tooltiptext">Only a subsample of the Ensembl dataset is used to save computing time. If nothing / non-numeric value is entered use the full dataset.</span></span>
						</td>
						<td v-if="geninfo.Species != ''">
							<input placeholder="Type a number" type="text" v-model="geninfo['Max Number of SNPs']">
						</td>
						<td v-else>
						</td>
					</tr>	
					<tr v-if="geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='Yes'">
						<td>Upload Map File
							<span class="tooltip">&#9432 <span class="tooltiptext">Currently only .vcf .map and .RData-files are supported. Contact Torsten (torsten.pook (at) uni-goettingen.de) to add additional maps.</span></span>
						</td>
						<td>
							<form id='MapUpload' target="print_popup" action='/MapUpload' method='post' encType='multipart/form-data' onsubmit="window.open('about:blank','print_popup','width=1000,height=800'); setMapPath();">
								<input type='file' name='mapFile' id='mapFile'>
								<input type='submit' value='Upload Map'>
							</form>
						</td>
					</tr>
					<tr v-if="geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='Yes'">
						<td>Own Map Path </td>
						<td>
							{{geninfo["Own Map Path"]}}
						</td>
					</tr>
					<tr v-if="geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='No'">
						<td>Number of Chromosomes</td>
						<td><input placeholder="Type a number" type="text" v-model="geninfo['Number of Chromosomes']" v-on:change='createChrom();'></td>
					</tr>			
					<tr v-if="geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='No'">
						<td>Chromosomes of Equal Length</td>
						<td> 
							<input type="checkbox" false-value="No" true-value="Yes" v-model="geninfo['Chromosomes of Equal Length']">
						</td>
					</tr>			
					<tr v-show="geninfo['Chromosomes of Equal Length'] == 'Yes' & geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='No'">
						<td colspan=2>
							<button v-on:click='show_chromo_info = !show_chromo_info' >Show/Hide Chromosome Information</button>
							<br>
							<br>
								<table id="chrom_table" v-show="show_chromo_info">
									<thead>
										<tr>
											<td>Chromosome</td>
											<td>Physical Length (in MbP)
												<span class="tooltip">&#9432 <span class="tooltiptext">The human genome has a size of 3200 Mb. Most livestock/crops are smaller!</span></span>
											</td>
											<td>Marker Density (SNP/Mbp)
												<span class="tooltip">&#9432 <span class="tooltiptext">Do not overdo it with markers! They oftentimes mostly add memory requirement and computation time.</span></span>
											</td>
											<td>Recombination (cM/Mbp)
												<span class="tooltip">&#9432 <span class="tooltiptext">1cM/Mbp is commonly assumed. E.g. chicken have higher recombination rates (~3-4 cM/Mbp).</span></span>
											</td>
										</tr>
									</thead>
									<tbody>
										<tr v-for='chrom in chromo_display.slice(0,1)' >
											<td>Chromosome 1 - {{geninfo['Number of Chromosomes']}}</td>
											<td v-for="key in Object.keys(chrom)">
												<input v-model="chrom[key]"> 
											</td>
										</tr>
									</tbody>
								</table>
						</td>
					</tr>
					<tr v-show="geninfo['Chromosomes of Equal Length'] == 'No' & geninfo['Use Ensembl Map'] =='No' && geninfo['Use Own Map'] =='No'">
						<td colspan=2>
							<button v-on:click='show_chromo_info = !show_chromo_info' >Show/Hide Chromosome Information</button>
							<br>
							<br>
							<table id="chrom_table" v-show="show_chromo_info">
								<thead>
									<tr>
										<td>Chromosome</td>
										<td>Physical Length (in MbP)</td>
										<td>Marker Density (SNP/Mbp)</td>
										<td>Recombination (cM/Mbp)</td>
									</tr>
								</thead>
								<tbody>
									<tr v-for='chrom in chromo_display' >
										<td>Chrom {{chromo_display.indexOf(chrom)+1}}</td>
										<td v-for="key in Object.keys(chrom)">
											<input v-model="chrom[key]"> 
										</td>
									</tr>
								</tbody>
							</table>
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] && geninfo['advanced_test']">
						<td class="boldText">Advanced Test-mode:	</td>
					</tr>
					<tr v-if="geninfo['advanced'] && geninfo['advanced_test']">
						<td>Test/Speed Mode
							<span class="tooltip">&#9432 <span class="tooltiptext">Activate this to set the number of repeats to 1, reduce the genome length to 10M and 200 SNPs and deactivate data import. This should heavily reduce computation times for testing.</span></span>
						</td>
						<td> 
							<input type="checkbox" false-value="No" true-value="Yes" v-model="geninfo['speed-mode']">
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] && geninfo['advanced_test']">
						<td>Individual numbers scaling 
							<span class="tooltip">&#9432 <span class="tooltiptext">Scale the number of individuals in each node by this factor. This can massively reduce computing times! Point numbers are rounded.</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['individual-scaling']">
						</td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_parallel']">
						<td class="boldText">Advanced parallel computing: </td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_parallel']">
						<td>Number of Simulations 
							<span class="tooltip">&#9432 <span class="tooltiptext">Number of simulations of the breeding program you want to perform. This will massively increase run-time but also reduce variance of results.</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['number-simulations']">
						</td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_parallel']">
						<td>Number of Simulations run in parallel
							<span class="tooltip">&#9432 <span class="tooltiptext">Number of simulations of the breeding program to run simultaneously. If taking more cores than allowed this will automatically be reduced internally.</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['number-simulations-parallel']">
						</td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_parallel']">
						<td>Number of cores used per simulation
							<span class="tooltip">&#9432 <span class="tooltiptext">Number of cores used per simulation. If taking more cores than allowed this will automatically be reduced internally.</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['number-simulations-core']">
						</td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_history']">
						<td class="boldText"> Advanced population history:	</td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_history']">
						<td> Set a new base-line population every X - generations
							<span class="tooltip">&#9432 <span class="tooltiptext">Genotypes for a baseline/founder population are stored leading to less points of recombination/mutation.</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['history_baseline']">
						</td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_history']">
						<td> Delete generations before base-line
							<span class="tooltip">&#9432 <span class="tooltiptext">Genotypes for a baseline/founder population are stored leading to less points of recombination/mutation.</span></span>
						</td>
						<td>
							<input type="checkbox" v-model="geninfo['history_delete']">
						</td>
					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_trait'] && geninfo['advanced_trait_scaling']">
						<td class="boldText"> Advanced trait rescaling: </td>

					</tr>
					<tr v-show="geninfo['advanced'] && geninfo['advanced_trait'] && geninfo['advanced_trait_scaling']">
						<td> Time point to apply trait rescaling
							<span class="tooltip">&#9432 <span class="tooltiptext">Semicolon separated list of time point to apply rescaling of trait mean/variance (e.g. 50;100)</span></span>
						</td>
						<td>
							<input type="value" v-model="geninfo['trait_rescaling']">
						</td>
					</tr>
				</table>
				<br>
				<br>
			</div>
			
			<div id="Phenotype_Info_Div" onchange="checkEverything(this.id)">
				<h3 id="Phenotype_Info"> Phenotype Information
					<span class="tooltip">&#9432 <span class="tooltiptext">Here you can design the traits you want to work with.</span></span>
				</h3> 
				<button class="accordion"> Press here to get info on how this module works! </button>
				<div class="guideline">
					<p class="guidelines"> 
					Press "Add new phenotype" to add additional traits <br>
					&#8226 Polygenic Loci are purely additive QTL but on a randomly selected SNP of the used array <br>
					&#8226 To model more complex traits activate the "Advanced Trait modelling" in the General Information <br>
					&#8226 For correlated traits the number of QTLs for each trait will be higher than entered as QTLs from other traits will
					  also impact a given trait. Unless correlations are high these should only be small effect (see MoBPS Guidelines for details on the method) <br>
					Make sure you correlation matrices are positive definit - otherwise we will make them positive definit automatically!
					</p>
				</div>
				<button class="sub_button" v-on:click="addPheno();"> Add new phenotype </button>
				<button class="sub_button" v-if="traitsinfo.length > 0" v-on:click='show_pheno_info = !show_pheno_info' >Show/Hide {{traitsinfo.length}} phenotypes</button>
				<button class="sub_button" v-if="traitsinfo.length > 0" v-on:click="show_QTL = !show_QTL"> Show/Hide QTLs</button>
				<button class="sub_button" v-if="traitsinfo.length > 1" v-on:click="show_matrix = !show_matrix"> Show/Hide residual correlation</button>
				<button class="sub_button" v-if="traitsinfo.length > 1" v-on:click="show_matrix2 = !show_matrix2"> Show/Hide genetic correlation</button>
				<br>
				<table class="drag_table" v-if="traitsinfo.length > 0 && show_pheno_info">
					<thead>
						<tr>
							<td style="width:80px;">Phenotype
								<span class="tooltip">&#9432 <span class="tooltiptext">Name of the trait.</span></span>
							</td>
							<td>Unit
								<span class="tooltip">&#9432 <span class="tooltiptext">Optional. Unit of measurement for the trait.</span></span>
							</td>
							<td>Pheno. Mean
								<span class="tooltip">&#9432 <span class="tooltiptext">The avg. Founder individual will have this mean. To have subpopulations with different means use the Subpopulation module.</span></span>
							</td>
							<td>Pheno. SD
								<span class="tooltip">&#9432 <span class="tooltiptext">Trait variance for the group of all founders will be scaled accordingly. Residual variance is assumed to be fixed over all generations.</span></span>
							</td>
							<td style="padding-right:0px;">Heritability
								<span class="tooltip">&#9432 <span class="tooltiptext">Heritability of the trait.</span></span>
							</td>
							<td  v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_repeat']">Repeatability
								<span class="tooltip">&#9432 <span class="tooltiptext">Repeatability of the trait (If empty will be equal to heritability - independent observations).</span></span>
							</td>
							<td v-if = "geninfo['advanced']==false || geninfo['advanced_trait']==false || geninfo['advanced_trait_epi']==false"># Polygenic Loci
								<span class="tooltip">&#9432 <span class="tooltiptext">Number of QTL. To simulate non-additive QTL effect activate the advanced setting Complex Trait architecture.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_epi']==true"># additive QTL
								<span class="tooltip">&#9432 <span class="tooltiptext">Number of QTL. To simulate non-additive QTL effect activate the advanced setting Complex Trait architecture.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_epi']==true"># dominant QTL
								<span class="tooltip">&#9432 <span class="tooltiptext">Number of purely dominant QTL.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_epi']==true" ># qualitative epistatic QTL
								<span class="tooltip">&#9432 <span class="tooltiptext">Number of QTL with qualitative epistatic effect between two markers.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_epi']==true" ># quantitative epistatic QTL
								<span class="tooltip">&#9432 <span class="tooltiptext">Number of QTL with quantitative epistatic effect between two markers.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_maternal']==true" > maternal trait
								<span class="tooltip">&#9432 <span class="tooltiptext">Genomic value of the trait is caused by the mother</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_maternal']==true" > paternal trait
								<span class="tooltip">&#9432 <span class="tooltiptext">Genomic value of the trait is caused by the father</span></span>
							</td>

							<td style="padding-right:0px;">Major QTL
								<span class="tooltip">&#9432 <span class="tooltiptext">Use this if you want to manually add single marker effects to SNPs with known effects.</span></span>
							</td>
							<td v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_combi']==true" > combination of traits
								<span class="tooltip">&#9432 <span class="tooltiptext"> This trait is a linear combination of other traits</span></span>
							</td>
							<td style="padding-right:0px;">Value per unit (&#8364;)
								<span class="tooltip">&#9432 <span class="tooltiptext">Economic gain by increased unit of the trait. Only relevant for economic calculations. Currently only linear effect and no interactions - Contact Torsten if you need more!</span></span>
							</td>
							<td style="width:90px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_trafo']==true">Apply Transformation
								<span class="tooltip">&#9432 <span class="tooltiptext">The only allowed input variable is the genomic value (x). Potential input: function(x){y = x^2; return(y)}</span></span>
							</td>
							<td>Show Cor</td>
							<td></td> 							
						</tr>
					</thead>
				</table>
				
				<table v-if = "traitsinfo.length > 0 && show_pheno_info">			
					<draggable :list="traitsinfo" 
					class="dragArea"
					:move="moveMatrix"
					@change="seeChange"
					@end="onEnd">
					<template v-for='(trait, index) in traitsinfo' >
						<tr>
							<td style="width:0.06%;">
								<input class="phenoMedium" v-model="trait['Trait Name']"> 
							</td>
							<td style="width:0.05%;padding-right:12px;">
								<input class="phenoSmall" v-model="trait['Trait Unit']" > 
							</td>
							<td style="width:0.05%;padding-right:12px;"  v-if = "trait['is_combi'] == false">
								<input class="phenoSmall" v-model="trait['Trait Mean']" class="no-spinners" step=0.001 v-on:change="trait['Trait Mean'] = trait['Trait Mean'].replace(/,/g, '.')"> 
							</td>
							<td style="width:0.1%;"  v-if = "trait['is_combi'] == true">
								<input class="phenoSmallReadOnly" v-model="trait['Trait Mean']" class="no-spinners" step=0.001 v-on:change="trait['Trait Mean'] = trait['Trait Mean'].replace(/,/g, '.')" readonly> 
							</td>
							<td style="width:0.05%;padding-right:4px;"  v-if = "trait['is_combi'] == false">
								<input class="phenoSmall" v-model="trait['Trait Std Deviation']" class="no-spinners" step=0.001 v-on:change="trait['Trait Std Deviation'] = trait['Trait Std Deviation'].replace(/,/g, '.')">  
							</td>
							<td style="width:0.1%;"  v-if = "trait['is_combi'] == true">
								<input class="phenoSmallReadOnly" v-model="trait['Trait Std Deviation']" class="no-spinners" step=0.001 v-on:change="trait['Trait Std Deviation'] = trait['Trait Std Deviation'].replace(/,/g, '.')" readonly>  
							</td>							
							<td style="width:0.05%;padding-right:0px;">
								<input class="phenoSmall" v-model="trait['Trait Heritability']" max=1 min=0 class="no-spinners" step=0.0001 v-on:change="trait['Trait Heritability'] = trait['Trait Heritability'].replace(/,/g, '.')"> 
							</td>
							<td style="width:0.3%;padding-right:0.07px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_repeat']==true">
								<input class="phenoSmall"  v-model="trait['Trait Repeatability']" max=1 min=0 class="no-spinners" step=0.0001 v-on:change="trait['Trait Repeatability'] = trait['Trait Repeatability'].replace(/,/g, '.')"> 
							</td>
							<td style="width:0.15%;padding-right:0px;" v-if = "trait['is_combi'] == false">
								<input class="phenoSmall" v-model="trait['Trait Number of Polygenic Loci']" class="no-spinners" step=1> 
							</td>
							<td style="width:0.15%;padding-right:0.07px;" v-if = "trait['is_combi'] == true">
								<input class="phenoSmallReadOnly" v-model="trait['Trait Number of Polygenic Loci']" class="no-spinners" step=1 readonly> 
							</td>
							<td style="width:0.1%;padding-right:0px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_epi']==true && trait['is_combi']==false">
								<input class="phenoSmall"  v-model="trait['dominant_qtl']"  class="no-spinners" step=1> 
							</td>
							<td style="width:0.4%;padding-right:4px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_epi']==true && trait['is_combi']==true">
								<input class="phenoSmallReadOnly" v-model="trait['dominant_qtl']"  class="no-spinners" step=1 readonly> 
							</td>
							<td style="width:0.1%;padding-right:0px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_epi']==true && trait['is_combi']==false">
								<input class="phenoSmall" v-model="trait['qualitative_qtl']" class="no-spinners" step=1> 
							</td>
							<td style="width:0.4%;padding-right:4px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_epi']==true && trait['is_combi']==true" >
								<input class="phenoSmallReadOnly"  v-model="trait['qualitative_qtl']" class="no-spinners" step=1 readonly> 
							</td>
							<td style="width:0.1%;padding-right:0px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_epi']==true && trait['is_combi']==false">
								<input class="phenoSmall"  v-model="trait['quantitative_qtl']"  class="no-spinners" step=1> 
							</td>
							<td style="width:0.4%;padding-right:4px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_epi']==true && trait['is_combi']==true">
								<input class="phenoSmallReadOnly"  v-model="trait['quantitative_qtl']"  class="no-spinners" step=1 readonly> 
							</td>							<td style="width:0.3%;padding-right:5px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_maternal']==true">
								<input class="phenoCheck" type="checkbox" v-model="trait['is_maternal']" >
							</td>
							<td style="width:0.3%;padding-right:5px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_maternal']==true">
								<input class="phenoCheck" type="checkbox" v-model="trait['is_paternal']" >
							</td>							<td style="width:0.04%;padding-right:0px;" v-if = "trait['is_combi'] == false">
								<input class="phenoNumber" v-model="trait['Trait Major QTL']"  type="number" class="no-spinners" step=1 v-on:change="createQTL(index)"> 
							</td>
							<td style="width:0.1%;padding-right:15px;" v-if = "trait['is_combi'] == true">
								<input class="phenoNumberReadOnly" v-model="trait['Trait Major QTL']"  type="number" class="no-spinners" step=1 v-on:change="createQTL(index)" readonly> 
							</td>
							<td style="width:0.3%;padding-right:9px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_combi']==true">
								<input class="phenoCheck" type="checkbox" v-model="trait['is_combi']" v-on:change="createCombi(index)" >
							</td>
							<td style="width:0.08%;padding-right:4px;">
								<input class="phenoNumber" v-model="trait['Trait Value per Unit']"  type="number" class="no-spinners" step=0.001 v-on:change="trait['Trait Value per Unit'] = trait['Trait Value per Unit'].replace(/,/g, '.')">  
							</td>
							<td style="width:0.09%;padding-right:9px;" v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_trafo']==true">
								<input class="phenoTransfo"  v-model="trait['trafo']" > 
							</td>							
							<td style="width:0.1%;padding-right:10px;">
								<input class="phenoCheck" type="checkbox" v-model="show_matrix_element[index].show">
							</td>
							<td>
								<button class="sub_button" v-on:click="removePheno(index)">X</button>
							</td>
						</tr>
					</template>
					</draggable>
				</table>

				<br>
				<template v-if = "geninfo['advanced']==true && geninfo['advanced_trait']==true && geninfo['advanced_trait_combi']==true">
					<table class="trait_table">
						<thead>
							<tr>
								<td> Trait Name </td>
								<td v-for="trait in traitsinfo"> Weight {{trait['Trait Name']}} </td>
							</tr>
						</thead>
						<tbody>
							<tr v-for="(trait_row, index2) in traitsinfo">
								<td v-if="traitsinfo[index2]['is_combi'] == true"> {{trait_row['Trait Name']}} </td>
								<td v-for="(trait,index) in traitsinfo" style = "width:15px;" v-if="traitsinfo[index2]['is_combi'] == true"> <input v-model="traitsinfo[index2]['combi_weights'][index]"></td>
							</tr>
						</tbody>
					</table>
				</template>
				
				<br>
				
				<h3 v-if="show_QTL && majorvisible"> Major QTL syntax
					<span class="tooltip">&#9432 <span class="tooltiptext">Select in which way to want to enter information on the position of the major QTLs.</span></span>
				</h3>
				<table v-if="show_QTL && majorvisible">
					<tr>
						<td> 
							<select v-model="geninfo.majorQTLsyntax"> 
								<option v-for="option in major_qtl_type" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
				</table>
				
				<template v-for="trait in traitsinfo" v-if="show_QTL && trait['Trait Major QTL'] > 0">
					<table class="trait_table">
						<thead>
							<tr>
								<td v-if="geninfo.majorQTLsyntax=='SNP + Chromosome'">SNP NR for {{trait['Trait Name']}}
									<span class="tooltip">&#9432 <span class="tooltiptext">SNP NR refers to the the position on chromosome according to the order given in the map.</span></span>
								</td>
								<td v-if="geninfo.majorQTLsyntax=='SNP ID'">SNP ID
									<span class="tooltip">&#9432 <span class="tooltiptext">SNP ID refers to the ID (row 2) of the map. Make sure this SNP ID is present (e.g. when downsampling the number of markers via MAX. Number of SNPs.</span></span>
								</td>
								<td v-if="geninfo.majorQTLsyntax=='BasePair + Chromosome'">Base-pair
									<span class="tooltip">&#9432 <span class="tooltiptext">Base-pair will look to the SNP that is closest to the entered Base-pair and put the QTL effect on that marker.</span></span>
								</td>
								<td v-if="geninfo.majorQTLsyntax!='SNP ID'">Chromosome
									<span class="tooltip">&#9432 <span class="tooltiptext">When using Base-pair / SNP NR as input its also required to provide information on which chromosome to put the QTL.</span></span>
								</td>
								<td>Effect AA</td>
								<td>Effect AB</td>
								<td>Effect BB</td>
								<td>Allele Freq. (B)
									<span class="tooltip">&#9432 <span class="tooltiptext">Simulated allele frequency of the marker is set to this. This is only active if no genotype data is imported.</span></span>
								</td>
								<td>Optional Info</td>
								<td></td>
							</tr>
						</thead>
						<tbody>
							<tr v-for='snp in trait["Trait QTL Info"]' >
								<td v-for="key in Object.keys(snp).slice(0,1)" v-if="geninfo.majorQTLsyntax=='SNP + Chromosome'">
									<input v-model="snp[key]" style="width:100px;"> 
								</td>
								<td v-for="key in Object.keys(snp).slice(1,2)" v-if="geninfo.majorQTLsyntax=='SNP ID'">
									<input v-model="snp[key]" style="width:100px;"> 
								</td>
								<td v-for="key in Object.keys(snp).slice(2,3)"  v-if="geninfo.majorQTLsyntax=='BasePair + Chromosome'"> 
									<input v-model="snp[key]" style="width:100px;"> 
								</td>
								<td v-for="key in Object.keys(snp).slice(3,4)" v-if="geninfo.majorQTLsyntax!='SNP ID'">
									<input v-model="snp[key]" style="width:100px;"> 
								</td>
								<td v-for="key in Object.keys(snp).slice(4,9)">
									<input v-model="snp[key]" style="width:60px;" v-on:change="snp[key] = snp[key].replace(/,/g, '.')"> 
								</td>
								<td>
									<button class="sub_button" v-on:click="removeQTL(trait,snp)">X</button>
								</td>
							</tr>
						</tbody>
					</table>
				</template>
				
				<br>
				<table class="corr_table" v-if="traitsinfo.length > 1 && show_matrix">
					<tr>
						<td colspan="5" class="corr_title" >Residual Correlation
							<span class="tooltip">&#9432 <span class="tooltiptext">Make sure this matrix is positive definit. Otherwise we will project the matrix to the space of positive definit matrices.</span></span>
						</td>
					</tr>
					<tr>
						<td></td>
						<td class="corr_td" v-for="(row, rind) in matrix" v-if="show_matrix_element[rind].show">
							{{traitsinfo[rind]['Trait Name']}}
						</td>
					</tr>
					<tr v-for='(row, rind) in matrix' v-if="show_matrix_element[rind].show">
						<td class="corr_td">
							{{traitsinfo[rind]['Trait Name']}}
						</td>
						<td v-for="(cl, cind) in matrix" v-if="show_matrix_element[rind].show & show_matrix_element[cind].show">
							<input v-model="matrix[rind].row[cind].val" style="width:40px;font-size:10pt;font-weight:bold;height:18px;border:2px solid lightblue;" v-if="rind > cind" class="no-spinners" step=0.000001 min=-1 max=1 v-on:change="matrix[rind].row[cind].val = matrix[rind].row[cind].val.replace(/,/g, '.')">  
							<template v-if="rind <= cind">
								{{ matrix[cind].row[rind].val }}
							</template>
						</td>
					</tr>
				</table>
				<table v-if="traitsinfo.length > 1 && show_matrix">
					<tr>
						<td class="sub_text">
						Enter Phenotypic correlation instead of residual correlation
						</td>
						<td>
							<input type="checkbox" v-model="use_phenotypic_cor">
						</td>
					</tr>
					<br>
				</table>
				<table class="corr_table" v-if="traitsinfo.length > 1 && show_matrix2">
					<tr>
						<td colspan="5" class="corr_title">Genetic Correlation
							<span class="tooltip">&#9432 <span class="tooltiptext">Make sure this matrix is positive definit. Otherwise we will project it so the space of positive definit matrices</span></span>
						</td>
					</tr>
					<tr>
						<td></td>						
						<td class="corr_td" v-for="(row, rind) in matrix2" v-if="show_matrix_element[rind].show">
							{{traitsinfo[rind]['Trait Name']}}
						</td>
					</tr>
					<tr v-for='(row, rind) in matrix2' v-if="show_matrix_element[rind].show">
						<td class="corr_td">
							{{traitsinfo[rind]['Trait Name']}}
						</td>
						<td v-for="(cl, cind) in matrix2" v-if="show_matrix_element[rind].show & show_matrix_element[cind].show">
							<input v-model="matrix2[rind].row[cind].val" style="width:40px;font-size:10pt;font-weight:bold;height:18px;border:2px solid lightblue;" v-if="rind > cind" class="no-spinners" step=0.000001 min=-1 max=1 v-on:change="matrix2[rind].row[cind].val = matrix2[rind].row[cind].val.replace(/,/g, '.')">
							<template v-if="rind <= cind">
								{{ matrix2[cind].row[rind].val }}
							</template>
						</td>
					</tr>
				</table>
				<br>
			

			<div id="corrDiv" style="display:none">
				<h3> Import correlation matrices via Excel/csv  <span class="tooltip">&#9432 </h3>
				<button class="accordion"> Press here to get info on how this module works! </button>
			
				<div class="guideline">
					<p class="guidelines"> 
					<span class="tooltiptext">
					This module is intended to import large-scale correlation matrices. <br>
					&#8226 After importing this will overwrite all entries in Residual/Genetic correlation accordingly. <br>
					&#8226 The first row and first column of the csv/excel should be name after used phenotypes <br>
					&#8226 When importing both genetic and residual correlation matrix jointly make sure that: <br>
					&#8226 Genetic correlation is above the diagonal and phenotypic correlation is below the diagonal <br>
					</p>
				</div>

 				<table id="load_table">
					<tr>
					<td>Select the type of Excel File:
						<span class="tooltip">&#9432 <span class="tooltiptext">"Please select an Excel file of genetic correlations or residual correlations or both."</span></span>
					</td>
					 <td> 
						<select v-model="geninfo.Excel_File">  
							<option v-for="option in Excel_File_options" v-bind:value="option">
							{{ option }}
							</option>
						</select>
					</td>
					</tr>
					<tr> <td>Choose Excel File from Desktop:
						<span class="tooltip">&#9432 <span class="tooltiptext">"Please choose an Excel file from desktop."</span></span>
					</td> 
					<td> <input id="excelToArray" type=file style="width:178px;" >
					<button class="sub_button" id="save_button" onclick="saveProject(geninfo['Project Name']);">Import Table</button></td>
					</tr>
				</table>
				</div>
			</div>
			
			<br>
			
			<div id="Selectionindex_Div" onchange="checkEverything(this.id)">
				<h3 v-if="traitsinfo.length > 0">Creating own Selection Indexes	</h3> 
				<button class="accordion" v-show="traitsinfo.length > 0"> Press here to get info on how this module works! </button>
				<div class="guideline">
					<p class="guidelines"> 
						Create your own selection indexes.<br>
						&#8226 Selection Indexes can be used in all edges in which individuals are selected.  <br>
						&#8226 Values from which to select later (usually breeding values / phenotypes) are scaled before applying the selection index.<br>
						&#8226 Scaling can be performed per phenotypic, genomic or breeding value variance of the group of individuals to select from! <br>
					</p>
				</div>
				
				<input id="newSI" type="text" placeholder="Type SI name" v-show="traitsinfo.length > 0">
				<button class="sub_button" v-on:click="addSI()" v-if="traitsinfo.length > 0"> Add new SI</button>
				<br>
				<template v-if="traitsinfo.length > 0">
					<table class="trait_table">
						<thead>
							<tr>
								<td>SI
									<span class="tooltip">&#9432 <span class="tooltiptext">Name of the selection index</span></span>
								</td>

								<td v-for="trait in traitsinfo"> {{trait['Trait Name']}}</td>
								
								<td></td>
							</tr>
						</thead>
						<tbody>
							<tr v-for='(si,index) in selection_index' >
								<td>
									<!-- <input v-model="si['Name']" style="width:80px;"> -->
									{{si['Name']}}
								</td>
								
								<td v-for="(key, ind) in Object.keys(si)" v-if="ind >0">
									<input v-model="si[key]" style="width:30px;font-size:10pt;font-weight:bold;height:18px;border:2px solid lightblue;" v-on:change="si[key] = si[key].replace(/,/g, '.')">
								</td>


								<td>
									<button v-on:click="removeSI(si, index)" v-if="index > 1">X</button>
								</td>
							</tr>

						</tbody>
					</table>
					<br>
					<table class="trait_table">
						<thead>
							<tr>
								<td>SI
									<span class="tooltip">&#9432 <span class="tooltiptext">Name of the selection index</span></span>
								</td>								
								<td>Standardization 
									<span class="tooltip">&#9432 <span class="tooltiptext">Scaling of breeding values before application of index weights.</span></span>
								</td>
								<td></td>
							</tr>
						</thead>
						<tbody>
							<tr v-for='(si,index) in selection_index_scaling' >
								<td>
									{{si['Name']}}
								</td>
								<td>
									<select v-model="si.w_scaling">
									<option v-for="option in w_options" v-bind:value="option">
									{{ option }}
									</option>
									</select>
								</td>
							</tr>
						</tbody>
					</table>
					<br>
					<br>
				</template>			
				<h3 v-if="traitsinfo.length > 0">Creating own Phenotyping Classes 
				</h3>
				
				<button class="accordion" v-show="traitsinfo.length > 0"> Press here to get info on how this module works!</button>
				<div class="guideline">
						<p class="guidelines"> 
						Create your own phenotyping classes <br>
						&#8226 Only numeric values are allowed with number indicating the number of observations collected. <br>
						&#8226 Unless repeatability is entered in the trait generation, observations are simulated as INDEPENDENT observations. <br>
						&#8226 Phenotyping classes are attributes that can be assigned to nodes in the breeding scheme. <br>						
						</p>
				</div>
				
				<input id="newPhenoC" type="text" placeholder="Type PhenoC name" v-show="traitsinfo.length > 0">
				<button class="sub_button" v-on:click="addPC()" v-if="traitsinfo.length > 0"> Add new PhenoC</button>

				<br>
				<template v-if="traitsinfo.length > 0">
					<table class="trait_table">
						<thead>
							<tr>
								<td>PhenoClass
									<span class="tooltip">&#9432 <span class="tooltiptext">A number higher than 1 means multiple INDEPENDET phenotypes are observed. This reduces residual variance and increases heritability.</span></span>
								</td>
								<td>Phenotyping Cost</td>
								<td v-for="trait in traitsinfo"> {{trait['Trait Name']}}</td>
								<td></td>
							</tr>
						</thead>
						<tbody>
							<tr v-for='(si,index) in phenotyping_class' >
								<td>
									<!-- <input v-model="si['Name']" style="width:90px;"> -->
									{{si['Name']}}
								</td>
								<td v-for="(key, ind) in Object.keys(si)" v-if="ind >0" >
									<input v-model="si[key]" style="width:30px;" v-on:change="si[key] = si[key].replace(/,/g, '.')">								</td>
								<td>
									<button class="sub_button" v-on:click="removePC(si, index)" v-if="index > 0">X</button>
								</td>
							</tr>
						</tbody>
					</table>
					<br>
				</template>	
				
								<h3 v-if="geninfo['advanced'] && geninfo['advanced_ld']"> LD build-up
					<span class="tooltip">&#9432 <span class="tooltiptext">
					</span></span>
				</h3>
				
				<button class="accordion" v-show="geninfo['advanced'] && geninfo['advanced_ld']"> Press here to get info on how this module works!</button>
				<div class="guideline">
						<p class="guidelines"> 
						
					&#8226 Founder genotypes on default have no LD / haplotype structure. <br> 
					&#8226 This module provides options to perform generations of random mating to get a baseline population structure <br> 
					&#8226 For more complex population history (e.g. bottlenecks, systematic drift etc) you need to simulate this via additional nodes in the breeding scheme <br>
					&#8226 If different founder nodes should have different population history add multiple Subpopulations in the Subpopulation Module <br>
					&#8226 This will alter the allele frequency spectrum selected in the Subpopulation module as it will lead to drift and thus usually include more rare / fixated markers <br>
					</p>
				</div>
				
				<table v-if="geninfo['advanced'] && geninfo['advanced_ld']" class="trait_table">

					<tr>	
						<td style="font-weight:bold">Subpopulation	
						</td>
						<td style="padding:5px;"> Individuals
							<span class="tooltip">&#9432 <span class="tooltiptext"> Number of individuals to generate per generation </span></span>
						</td>
						<td style="padding:5px;"> Share female
							<span class="tooltip">&#9432 <span class="tooltiptext"> Share of female individuals in each generation </span></span>
						</td>
						<td style="padding:5px;"> Generations
											<span class="tooltip">&#9432 <span class="tooltiptext"> Number of generations to simulate. Larger populations usually require higher number of generations to build up stronger LD.</span></span>
						</td>
						<td style="padding:5px;"> Fix Major QTL frequency
							<span class="tooltip">&#9432 <span class="tooltiptext"> Due to drift the allelel frequency of major QTLs can change in the LD build-up. By activating this, this will be negated. LD build-up will be negated for Major QTLs!</span></span>
						</td>

					</tr>
					<tr v-for="(vv, ind) in subpopulation['subpopulation_list']">
						<td>
							{{vv.Name}}  
						</td>
						<td>
							<input v-model="vv.nindi" style="width:100px;"> 
						</td>
						<td>
							<input v-model="vv.sharef" style="width:100px;"> 
						</td>
						<td>
							<input v-model="vv.ngen" style="width:100px;"> 
						</td>
						<td> 
							<input type="checkbox" v-model="vv.majorfreq">
						</td>
					</tr>
				</table>				
				
				<h3 v-if="geninfo['advanced'] && geninfo['advanced_culling']"> Culling
					<span class="tooltip">&#9432 <span class="tooltiptext">
					</span></span>
				</h3>
				
				<button class="accordion" v-show="geninfo['advanced'] && geninfo['advanced_culling']"> Press here to get info on how this module works!</button>
				<div class="guideline">
					<p class="guidelines"> 
						
					&#8226 Define reasons for individuals to exit the breeding program for reproduction. <br> 
					&#8226 These individuals can still be used in downstream BVEs <br> 
					&#8226 Individuals with BV1 (in the index selected) will die with probability 'Share death at BV1'. <br>
					&#8226 In case of no underlying genomics all individuals will die with this probability. <br>
					&#8226 In case of a genetic reason provide BV2. Values for each genomic value will be derived based on linear extension. <br>
					<br> 
						Create your own culling reasons.<br> 
						&#8226 Parameters are chosen according the culling module in breeding.diploid(). <br>
						&#8226 Death individuals can still be used for BVE but not for reproduction.<br>
						&#8226 Individuals with BV1 (in the index selected) will die with probability 'Share death at BV1'.<br>
						&#8226 In case of no underlying genomics all individuals will die with this probability <br>
						&#8226 In case of a genetic reason provide BV2. Values for each genomic value will be derived based on linear extension.						
					</p>
				</div>
				
				<table class="culling_table" v-if="geninfo['advanced'] && geninfo['advanced_culling']">
					<tr>
						<td colspan="2" >
							<input id="newCullingClass" type="text" style="width:75%;" placeholder="Type a Culling Reason">
						</td>

						<td colspan="2" style="text-align:left;">
							<button class="sub_button" for="newCullingClass" v-on:click="createCulling()"> Creating new Culling Reason </button>
						</td>
					</tr>	

					<tr v-if="culling['culling_reasons'].length > 0">
						<td style="font-weight:bold">Culling reason	
						</td>
						<td style="padding:5px;">At age (in time unit)
							<span class="tooltip">&#9432 <span class="tooltiptext">At what age does the culling potentially occur. Usage for reproduction at the time point itself is still possible.</span></span>
						</td>
						<td style="padding:5px;">Relevant for sex
							<span class="tooltip">&#9432 <span class="tooltiptext">Is this relevant for Male/Female/Both?</span></span>
						</td>
						<td style="padding:5px;">Genetic Index for survival
							<span class="tooltip">&#9432 <span class="tooltiptext">Select a selection index linked to the culling</span></span>
						</td>
						<td style="padding:5px;">Share death at BV1
							<span class="tooltip">&#9432 <span class="tooltiptext">What share of individuals are culled when they have BV1 (genomic value * selected index).</span></span>
						</td>
						<td style="padding:5px;">BV1
						</td>
						<td style="padding:5px;">Share death at BV2
							<span class="tooltip">&#9432 <span class="tooltiptext">What share of individuals are culled when they have BV2 (genomic value * selected index).</span></span>
						</td>
						<td style="padding:5px;">BV2
						</td>
						<td>
						</td>
					</tr>
					<tr v-for="(vv, ind) in culling['culling_reasons']">
						<td>{{vv.Name}}</td>
						<td><input type="text" v-model="vv.At_age">	</td>
						<td>
							<select v-model="vv.Relevant_Sex">
								<option v-for="option in Sex_options2" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
						<td>
							<select v-model="vv.Genetic_index_for_survival">
								<option v-for="option in selection_index" v-bind:value="option.Name">
								{{ option.Name }}
								</option>
							</select>
						</td>
						<td><input type="text" v-model="vv.Share_death_at_BV1" ></td>
						<td><input type="text" v-model="vv.BV1" ></td>
						<td><input type="text" v-model="vv.Share_death_at_BV2" ></td>
						<td><input type="text" v-model="vv.BV2" ></td>
						<td><button class="sub_button" v-on:click="removeCulling(vv, ind)">X</button></td>
					</tr>
				</table>				
				<br>
				
				<h3 v-if="geninfo['advanced'] && geninfo['advanced_sub']"> Multiple Subpopulations	</h3>				
				<button class="accordion" v-show="geninfo['advanced'] && geninfo['advanced_sub']"> Press here to get info on how this module works!</button>
				<div class="guideline">
						<p class="guidelines"> 
						Create your own subpopulation.  <br>
						&#8226 Each founder node of the breeding scheme can be assign to a subpopulation. <br>
						&#8226 Allele frequencies will be calculated based on all cohorts of the given subpopulation with imported genotypes.<br>	
						&#8226 Otherwise: Each subpopulation has independently sampled allele frequencies from a Beta(p,q)-distribution.<br>		
						&#8226 To model different level of diversity controll the share of fixated markers.<br>	
						&#8226 In case of different mean trait values additional QTL effects are put on fixated markers.<br>
				</div>
				<table v-if="geninfo['advanced'] && geninfo['advanced_sub']" class="trait_table">
					<tr>
						<td colspan="2" style="text-align:left;">
							<input id="newSubpopulationClass" type="text" placeholder="Type a Subpopulation name">
						</td>
						<td colspan="2" style="text-align:left;">
							<button class="sub_button" for="newSubpopulationClass" v-on:click="createSub()"> Creating new Subpopulation </button>
						</td>
					</tr>
					<tr>	
						<td style="font-weight:bold">Subpopulation	
						</td>
						<td style="padding:5px;">Sampled allele frequency (p)
							<span class="tooltip">&#9432 <span class="tooltiptext">Allele frequencies are drawn from a Beta (p,q) - distribution </span></span>
						</td>
						<td style="padding:5px;">Sampled allele frequency (q)
							<span class="tooltip">&#9432 <span class="tooltiptext">Allele frequencies are drawn from a Beta (p,q) - distribution </span></span>
						</td>
						<td style="padding:5px;">Share of fixated markers in A
							<span class="tooltip">&#9432 <span class="tooltiptext">Define allele frequencies for multiple subpopulations. 
							Allele frequencies are drawn from a Beta(p,q) distribution.
							To model different levels of diversity use Share of fixated markers.
							A population with less diversity should have more fixated markers.
							More similar population should be fixated in the same direction for often. 
							This cannot replace the use of REAL genomic data or simulated data with population structure (this module is mostly intended for baseline testing).</span></span>
						</td>
						<td style="padding:5px;">Share of fixated markers in B
							<span class="tooltip">&#9432 <span class="tooltiptext">Define allele frequencies for multiple subpopulations. 
							Allele frequencies are drawn from a Beta(p,q) distribution.
							To model different levels of diversity use Share of fixated markers.
							A population with less diversity should have more fixated markers.
							More similar population should be fixated in the same direction for often. 
							This cannot replace the use of REAL genomic data or simulated data with population structure (this module is mostly intended for baseline testing).</span></span>
						</td>
						<td style="padding:5px;">Nr. of Markers with manual chosen allele frequency
						</td>
						<td v-for="trait in traitsinfo"> Deviation from Mean for Trait {{trait['Trait Name']}}</td>
						<td>
						</td>
					</tr>
					<tr v-for="(vv, ind) in subpopulation['subpopulation_list']">
						<td>{{vv.Name}}</td>
						<td><input v-model="vv.beta1" style="width:100px;"></td>
						<td><input v-model="vv.beta2" style="width:100px;"></td>
						<td><input v-model="vv.share0" style="width:100px;"></td>
						<td><input v-model="vv.share1" style="width:100px;"></td>
						<td><input v-model="vv.fixed_freq" style="width:100px;" type="Number" class="no-spinners" step=1 v-on:change="createQTLsub(ind)"></td>
						<td v-for="(trait, ind2) in traitsinfo">
							<input v-model="vv['P' + (ind2+1)]" style = "width:30px;" v-if="ind>0" v-on:change="vv['P' + (ind2+1)] = vv['P' + (ind2+1)].replace(/,/g, '.')">
						</td> 
						<td><button v-on:click="removeSub(vv, ind)">X</button></td>
					</tr>
				</table>
				
				<template v-for="subpop in subpopulation['subpopulation_list']" v-if="subpop['fixed_freq'] > 0">
					<table class="trait_table">
						<thead>
							<tr>
								<td>SNP for {{subpop['Name']}}</td>
								<td>SNP ID
									<span class="tooltip">&#9432 <span class="tooltiptext">If SNP ID is present in the dataset it is used. If not the closed marker to bp/chromo combination is used. If not given use SNP/chromo combination.</span></span>
								</td>
								<td>bp</td>
								<td>Chromosome</td>
								<td>Allele Freq. (B)
									<span class="tooltip">&#9432 <span class="tooltiptext">Simulated allele frequency of the marker is set to this. Only active if no genotype data is imported.</span></span>
								</td>
								<td>Optional Info</td>
								<td></td>
							</tr>
						</thead>
						<tbody>
							<tr v-for='snp in subpop["QTL Info"]' >
								<td v-for="key in Object.keys(snp).slice(0,1)">
									<input v-model="snp[key]" style="width:100px;"> 
								</td>
								<td v-for="key in Object.keys(snp).slice(1,9)">
									<input v-model="snp[key]" style="width:60px;" v-on:change="snp[key] = snp[key].replace(/,/g, '.')"> 
								</td>
							</tr>
						</tbody>
					</table>
					<br>
				</template>
				
				<h3 v-if="geninfo['advanced'] && geninfo['advanced_litter']"> Litter size	</h3>
				
				<button class="accordion" v-show="geninfo['advanced'] && geninfo['advanced_litter']"> Press here to get info on how this module works!</button>
				<div class="guideline">
					<p class="guidelines"> 
					&#8226 The entered litter size will be applied on all newly generated individuals. <br> 
					&#8226 This does NOT include new cohorts generated via selection/combine/aging etc.<br> 
					&#8226 In case probabilites do not add up to 1 this will automatically be scaled accordingly. <br
					</p>
				</div>
				
				<td v-if="geninfo['advanced'] && geninfo['advanced_litter']" colspan="2" style="text-align:left;">
					<button class="sub_button" v-on:click="createSize()" v-show="geninfo['advanced'] && geninfo['advanced_litter']" > Add new litter size </button>
				</td>
				
				<table v-if="geninfo['advanced'] && geninfo['advanced_litter']" id="general_Litter"> 			
					<thead>
					<tr> 
						<td>Litter size </td>
						<td>Probability </td>
						<td> </td>
					</tr>
					</thead>
					<tbody>
						<tr v-for='(litter, index) in litter_size["litter_info"]'>
							<td><input v-model="litter.name" style = "width: 50px;"></td>
							 <td><input v-model="litter.prob" style = "width: 50px;"></td>
							<td><button v-on:click="removeSize(index)">X</button></td>
						</tr>
					</tbody>
				</table>
				<br>
				<br>

				<h3 v-if="geninfo['advanced'] && geninfo['advanced_eco']">General Economy Parameters</h3>
				<button class="accordion" v-show="geninfo['advanced'] && geninfo['advanced_eco']"> Press here to get info on how this module works!</button>
				<div class="guideline">
						<p class="guidelines"> 
						&#8226 Assign costs to basic breeding actions.  <br>
						&#8226 Fixed costs are assumed to happen at time 0. <br>
						&#8226 Interest is applied on all cost types.<br>
						&#8226 Housing classes can be selected for all nodes of the breeding scheme.<br>
						</p>
				</div>			
		
				<table v-if="geninfo['advanced'] && geninfo['advanced_eco']" id="general_Economy">
					<tr>
						<td>Fixed cost (&#8364;)
							<span class="tooltip">&#9432 <span class="tooltiptext">Fixed costs of the breeding program</span></span>
						</td>
						<td><input type="value" placeholder="Type a number" v-model="economy['Fixed Cost']"></td>
					</tr>
					<tr>
						<td>Interest rate (%)
							<span class="tooltip">&#9432 <span class="tooltiptext">Applied interest rate for all costs occurring.</span></span>
						</td>
						<td><input type="value" placeholder="Type a number" v-model="economy['Interest Rate']"></td>
					</tr>
					<tr>
						<td>Genotyping cost per individual (&#8364;)
							<span class="tooltip">&#9432 <span class="tooltiptext">Genotyping costs per individual. Currently only genotyping with a single array  is supported. Modes with different chips and marker densities are coming.</span></span>
						</td>
						<td><input type="value" placeholder="Type a number" v-model="economy['Genotyping Cost']"></td>
					</tr>
					<tr style="height:15px"><td colspan=2></td></tr>
					<tr>	
						<td class="corr_td">Housing/Field cost class	
							<span class="tooltip">&#9432 <span class="tooltiptext">Type a name and then click on "Creating new HFC Class" to create your own classes of housing/field costs. These classes can then be assigned to each cohort in the breeding scheme. </span></span>
						</td>
						<td class="corr_td">Housing/Field cost (&#8364;)</td>
					</tr>
					<tr v-for="(vv, ind) in economy['Animal Housing Costs']">
						<td>
							{{vv.Name}}  
						</td>
						<td>
							<input type="value" v-model="vv.Cost" style="width:100px;"> 
							<button class="sub_button" v-if="ind > 0" v-on:click="removeHCostClass(vv, ind)">X</button>
						</td>
					</tr>
					<tr>
						<td>
							<input type="text" id="newHCostClass" placeholder="Type a HFC name">
						</td>
						<td colspan="2">
							<button class="sub_button" for="newHCostClass" v-on:click="createHCostClass()"> Creating new HFC Class </button>
						</td>
					</tr>					
				</table>
				<br>
				
				<div id="Variables_Info" onchange="checkEverything(this.id)">
					<h3> Declaring own variables</h3> 
						
					<button class="accordion"> Press here to get info on how this module works!</button>
					<div class="guideline">
						<p class="guidelines"> 
						In this module you can assign numeric values to variables  <br>
						Variables can as inputs for nodes and edges <br>
						The main intended use is when having a high number of nodes/edges with the same number (e.g. number of individuals) and simultaneously wanted to change values for all nodes/edges between simulation runs<br>
						</p>
					</div>
	
					<input id="newVariable" type="text" placeholder="Type variable name">
					<button class="sub_button" for="newVariable" v-on:click="createVariable()"> Creating new Variable </button>
					<br>
						<table class="var_table" v-if="individualsVar_options.length > 0">  
							<tr>
								<td>Variable Name</td>
								<td>Variable Value</td>
							</tr>							
							<tr v-for="vv in individualsVar_options">
								<td>
									{{vv.name}}  
								</td>
								<td>
									<input type="value" v-model="vv.value" v-on:input="updateNodesVar(vv); updateEdgesVar(vv);"> 
								</td>
								<td>
									<button class="sub_button" v-on:click="removeVariable(vv)">X</button>
								</td>
							</tr>
						</table>
					<br>
				</div>
			</div>
			
			<div id="MultipleChange" v-if="geninfo['advanced'] && geninfo['advanced_multi'] && nodes.length>1" onclick="checkEverything('edge-popUp')">
				<h3> Modify Multiple Nodes/Edges
					<span class="tooltip">&#9432 <span class="tooltiptext">
					Via this module multiple nodes/edges with similar attributed can be changed simultaneously. <br>
					Select which nodes/edges to change by setting requirements.	</span></span>
				</h3> 
				<br>
				<table class="var_table">  
					<tr>
						<td>Change </td>
						<td>Requirements</td>
						<td>Parameter to change             //          Change to </td>
						<td> </td>
					</tr>					
					<tr>
						<td>
							<select v-model="change_type">  <!-- onchange="updateSpeciesData(value)">  -->
								<option v-for="option in units_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>						
						<td v-if="change_type=='edges'">
						 Breeding Type  <br>
						 	<select v-model="change_type1">  <!-- onchange="updateSpeciesData(value)">  -->
								<option v-for="option in Breedingtype_options2" v-bind:value="option">
								{{ option }}
								</option>
							</select>							
							<br>
						Sex - Parental Node <br>
						 	<select v-model="change_type2">  <!-- onchange="updateSpeciesData(value)">  -->
								<option v-for="option in Sex_options3" v-bind:value="option">
								{{ option }}
								</option>
							</select>							
							<br>							
						Sex - Child Node <br>
						 	<select v-model="change_type3">  <!-- onchange="updateSpeciesData(value)">  -->
								<option v-for="option in Sex_options3" v-bind:value="option">
								{{ option }}
								</option>
							</select>							
							</td>						 
							<td v-if="change_type=='nodes'">
						Sex <br>
						 	<select v-model="change_type4">  <!-- onchange="updateSpeciesData(value)">  -->
								<option v-for="option in Sex_options3" v-bind:value="option">
								{{ option }}
								</option>
							</select>							
							<br>
						Number of Individuals <br> (leave empty to select all)<br>
							<input type="value" v-model="change_type5">
						<br>
						Phenotyping Class <br>
						<select v-model="change_type7">
							<option> All </option>
							<option v-for="option in phenotyping_class" v-bind:value="option.Name">
							{{ option.Name }}
							</option>
						</select>
						<br>	
						Housing Cost Class<br>
						<select v-model="change_type8">							
							<option> All </option>
							<option v-for="option in economy['Animal Housing Costs']" v-bind:value="option.Name">
							{{ option.Name }}
							</option>
						</select>
						</td>						 
						<td v-if="change_type=='edges'">						 
						<table style="margin:auto; width: 500px;">							
							<td> Breeding Type </td>
							<td> 
								<select v-model="change_type13['Breeding Type']">
								<option> Remain unchanged</option>										 
								<option v-for="option in Breedingtype_options" v-bind:value="option">
								{{ option }}
								</option>
								</select>
							</td>								
							<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production','Semen-collection', 'Aging', 'Selection', 'Split','Combine'].indexOf(change_type13['Breeding Type']) >= 0 ">
								<td>Time needed (leave empty to not change)
									<span class="tooltip">&#9432 <span class="tooltiptext">Provide the time this breeding action takes. Reproduction currently needs at least one time unit.</span></span>
								</td>
								<td><input style="width:40px;border:2px solid lightblue;font-size:12pt;height:22px;" id="time_n" list="indVar" onchange="updateVar('Time Needed', value)" placeholder="Type a number" min="0" max="10000" step="1" v-model="change_type13['Time Needed']"> {{geninfo['Time Unit']}}</td>
							</tr>	
							<tr v-show="geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_last']">
								<td>Use last available
								<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will use the last repeat available from that node - not necessary the node from the same repeat</span></span>
							</td>		
							<td>
								<select v-model="change_type13['last_available']">
								<option value=true> Yes </option>
								<option value=false> No </option>
								<option value='Remain unchanged'> Remain unchanged </option>
								</select>
							</td>
							</tr>			
							<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_ogc']">
								<td>OGC
									<span class="tooltip">&#9432 <span class="tooltiptext">Use optimum genetic contribution according to Wellmann et al 2019 to determine how often each individual is used for reproduction.</span></span>
								</td>
								<td>
									<select v-model="change_type13['OGC']">
										<option value='Yes'> Yes </option>
										<option value='No'> No </option>
										<option value='Remain unchanged'> Remain unchanged </option>				
									</select>
								</td>
							</tr>								
							<tr v-if="['Reproduction'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_mating']">
								<td>Avoid full-sib matings
									<span class="tooltip">&#9432 <span class="tooltiptext">Activate this to not use full-sibling pairs for reproduction.</span></span>
								</td>
								<td>
								<select v-model="change_type13['full_sib']">
									<option value=true> Yes </option>
									<option value=false> No </option>
									<option value='Remain unchanged'> Remain unchanged </option>				
								</select>
								</td>								
							</tr>
			
							<tr v-if="['Reproduction'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_mating']">
								<td>Avoid half-sib matings
									<span class="tooltip">&#9432 <span class="tooltiptext">Activate this to not use half-sibling pairs for reproduction.</span></span>
								</td>
								<td>
								<select v-model="change_type13['half_sib']">
									<option value=true> Yes </option>
									<option value=false> No </option>
									<option value='Remain unchanged'> Remain unchanged </option>
				
								</select>
								</td>
							</tr>
							
							<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_off']">
								<td>Maximal # offspring (leave empty to not change)
									<span class="tooltip">&#9432 <span class="tooltiptext">Maximum number of offspring each individual from the parent node is allowed to have.</span></span>
								</td>
								<td><input placeholder="Type a number or Inf" v-model="change_type13['Max Offspring']"></td>
							</tr>	
							<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_off']">
								<td>Maximal # offspring from one mating pair (leave empty to not change)
									<span class="tooltip">&#9432 <span class="tooltiptext">Maximum number of offspring from one mating pair.</span></span>
								</td>
								<td><input placeholder="Type a number or Inf" v-model="change_type13['Max Offspring Pair']"></td>
							</tr>	
							<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_ratio']">
								<td>Selection ratio (leave empty to not change)
									<span class="tooltip">&#9432 <span class="tooltiptext">Use this if you want to use some individuals more frequently for reproduction. A value of 2 will lead to the individual will the highest genomic value to be used twice as often as the worst individual.</span></span>
								</td>
								<td><input placeholder="Type a number (if nothing enter all individuals are picked equally)" v-model="change_type13['selection_ratio']"></td>
							</tr>
							<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_ratio']">
								<td>Selection ratio type (leave empty to not change)
									<span class="tooltip">&#9432 <span class="tooltiptext">Use this if you want to use some individuals more frequently for reproduction. A value of 2 will lead to the individual will the highest genomic value to be used twice as often as the worst individual.</span></span>
								</td>
								<td> 
									<select v-model="change_type13['selection_ratio_type']">
										<option v-for="option in selection_type" v-bind:value="option">
										{{ option}}
										</option>
									</select>
								</td>				
							</tr>	
							<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] && (geninfo['advanced_advanced_ogc'] ||  geninfo['advanced_advanced_ratio'])">
								<td>Index for Selection ratio / OGC
									<span class="tooltip">&#9432 <span class="tooltiptext">Select the index used for the selection ratio or OGC. </span></span>
								</td>
								<td> 
									<select v-model="change_type13['selection_ratio_index']">
										<option>
											Remain unchanged
										</option>
										<option v-for="option in selection_index" v-bind:value="option">
										{{ option.Name }}
										</option>
									</select>
								</td>
							</tr>							
							<tr v-show="change_type13['Breeding Type'] == 'Repeat'">
								<td>Number of Repeat (leave empty to not change)
									<span class="tooltip">&#9432 <span class="tooltiptext">Number of times the Repeat is executed. The parent node will be the new input of the child node and everything in between is rerun. Nodes will be named the same + _1, _2, _3 etc.</span></span>
								</td> 
								<td> 
									<input list="indVar" placeholder="Type a number" v-model="change_type13['Number of Repeat']" onchange="updateVar('Number of Repeat', value)">
								</td>
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 ">
								<td>Selection Type
									<span class="tooltip">&#9432 <span class="tooltiptext">Random: Each individual is taken with same probability, BVE/Pheno selected based on BVE/Phenotypes</span></span>
								</td>
								<td> 
									<select v-model="change_type13['Selection Type']">										
										<option>Remain unchanged</option>
										<option v-for="option in selectionType_options" v-bind:value="option">
										{{ option }}
										</option>
									</select>
								</td>
							</tr>			
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 && change_type13['Selection Type'] == 'BVE'">
								<td>BVE Method
									<span class="tooltip">&#9432 <span class="tooltiptext">Direct-Mixed-Model assumes known heritability and thereby is fastest. EMMREML/sommer use the respective R-package. Sommer also support multi-variable models but takes much longer! Bayesian models used are using BGLR. MAS is using lm() on randomly sampled effect markers.</span></span>
								</td>
								<td> 
									<select v-model="change_type13['BVE Method']">									
										<option>Remain unchanged</option>										
										<option v-for="option in BVEMethod_options" v-bind:value="option">
										{{ option }}
										</option>
									</select>
								</td>
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 && change_type13['Selection Type'] == 'BVE' & change_type13['BVE Method']=='Marker assisted selection (lm)'">
								<td>Number of markers used for MAS (leave empty to not change)
									<span class="tooltip">&#9432 <span class="tooltiptext">This is a very simple implementation of MAS using a linear model of randomly sampled effect markers</span></span>
								</td>
								<td><input placeholder="Type a number or Inf" v-model="change_type13['MAS_marker']"></td>
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 & change_type13['Selection Type']=='Pseudo-BVE'" v-for='(trait, index) in traitsinfo' >
								<td>Accuracy for Trait {{ trait[["Trait Name"]] }} (leave empty to not change)
									<span class="tooltip">&#9432 <span class="tooltiptext">If nothing is entered here the accuracy will be chosen according to the starting heritability</span></span>
								</td>
								<td><input v-model="change_type13[index]" style="width:80px;"></td>
							</tr>								
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 & traitsinfo.length > 1">
								<td>Selection Index
								<span class="tooltip">&#9432 <span class="tooltiptext">Select which index to use for BVE/Phenotypic selection</span></span>
								</td>
								<td> 
									<select v-model="change_type13['Selection Index']">
										<option>Remain unchanged</option>
										<option v-for="option in selection_index" v-bind:value="option.Name">
										{{ option.Name }}
										</option>
									</select>
								</td>
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_thres']">
								<td>Threshold for selection (leave empty to not change)
									<span class="tooltip">&#9432 <span class="tooltiptext">Calculate reliability of BVE according to vanRaden 2008</span></span>
								</td>		
								<td><input type="value" placeholder="Type a number if wanted" v-model="change_type13['threshold']"></td>
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_thres']">
								<td>Threshold sign
									<span class="tooltip">&#9432 <span class="tooltiptext"> Only keep individuals with higher/lower/eval estimated breeding value</span></span>
								</td>		
								<td> 
									<select v-model="change_type13['threshold_sign']">
										<option>
											Remain unchanged
										</option>
										<option v-for="option in threshold_options" v-bind:value="option">
										{{ option }}
										</option>
									</select>
								</td>
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 && change_type13['Selection Type']=='BVE' & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_reli']">
								<td>Calculate reliability
									<span class="tooltip">&#9432 <span class="tooltiptext">Calculate reliability of BVE according to vanRaden 2008</span></span>
								</td>
								<td>
								<select v-model="change_type13['reliability']">
									<option value="Yes"> Yes </option>
									<option value="No"> No </option>
									<option value='Remain unchanged'> Remain unchanged </option>				
								</select>
								</td>						
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 && change_type13['Selection Type']=='BVE' & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_skip']">
								<td> Skip traits with no index weight
									<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will lead to not performing BVE that are not assign with any weight (=0) in the selection index.</span></span>
								</td>	
								<td>
								<select v-model="change_type13['skip']">
									<option value="Yes"> Yes </option>
									<option value="No"> No </option>
									<option value='Remain unchanged'> Remain unchanged </option>
								</select>
								</td>							
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_pheno']">
								<td>Input Phenotype
									<span class="tooltip">&#9432 <span class="tooltiptext">Use the avg. phenotype of the offspring as the phenotype of the individuals in the parent node. Make sure there are offspring! (use of time needed!)</span></span>
								</td>
								<td>
								<select v-model="change_type13['phenotype_used']">								
									<option>Remain unchanged</option>
									<option v-for="option in phenotype_options" v-bind:value="option">
									{{ option }}
									</option>									
								</select>
								</td>
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 && change_type13['Selection Type'] == 'BVE'">
								<td>Relationship Matrix
									<span class="tooltip">&#9432 <span class="tooltiptext">According to vanRaden 2008, pedigree-based or singleStep (H based on Legarra 2014)</span></span>
								</td>
								<td> 
									<select v-model="change_type13['Relationship Matrix']">									
										<option>Remain unchanged</option>
										<option v-for="option in RelationshipMatrix_options" v-bind:value="option">
										{{ option }}
										</option>
									</select>
								</td>
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 && change_type13['Selection Type'] == 'BVE' && ( change_type13['Relationship Matrix']== 'Single Step'|change_type13['Relationship Matrix']== 'Pedigree')">
								<td>Depth of Pedigree (# generation back) (leave empty to not change)
									<span class="tooltip">&#9432 <span class="tooltiptext">Depth of the pedigree. All individuals before are assumed to be unrelated! If you need something else contact Torsten or use the R-package.</span></span>
								</td> 
								<td> <input type="value" onchange="updateVar('Depth of Pedigree', value)" placeholder="Type a number" v-model="change_type13['Depth of Pedigree']" list="indVar"></td>
							</tr>							
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 && change_type13['Selection Type'] == 'BVE'">
								<td>Cohorts used in BVE
									<span class="tooltip">&#9432 <span class="tooltiptext">Manual select provides you with maximum flexibility to select which individuals to use.</span></span>
								</td>
								<td> 
									<select v-model="change_type13['Cohorts used in BVE']" onchange="cohorts_Div(this);">
										<option>Remain unchanged</option>
										<option v-for="option in Cohorts_options" v-bind:value="option">
										{{ option }}
										</option>
									</select>
								</td>
							</tr>
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 && change_type13['Selection Type'] == 'BVE' && change_type13['Cohorts used in BVE'] == 'Manual select'">
								<td>Select cohorts for manuel select:
									<span class="tooltip">&#9432 <span class="tooltiptext">Cohorts available for selection are only refresh on reloading or when EDITING an edge. We are working on that.</span></span>
								</td>
							</tr>
						</table>
				
						<div id="edge_cohorts_Div" >				
						<table >					
							<tr v-show="['Selection', 'Split', 'Aging'].indexOf(change_type13['Breeding Type']) >= 0 && change_type13['Selection Type'] == 'BVE' && change_type13['Cohorts used in BVE'] == 'Manual select'">
								<td v-if="isCohortNameChanged ===''" >						
									<treeselect :value-consists-of="'LEAF_PRIORITY'" :multiple="true" :disabled="false" :flat="false" :always-open="true" :options="node_manualselect_options" v-model="change_type13['Manuel selected cohorts']" :searchable="true" />
								</td>
							</tr>
						</table>
						</div>
					</td>						 
					<td v-if="change_type=='nodes'">
						Sex <br>
						 	<select v-model="change_type9"> 
								<option>Remain unchanged</option>							
								<option v-for="option in Sex_options2" v-bind:value="option">
								{{ option }}
								</option>
							</select>							
							<br>							
						Number of Individuals <br>(leave empty to not change)
						<br>
							<input type="value" v-model="change_type10">
						<br>
						Phenotyping Class
						<br>
						<select v-model="change_type11">
							<option>Remain unchanged </option>
							<option v-for="option in phenotyping_class" v-bind:value="option.Name">
							{{ option.Name }}
							</option>
						</select>
						<br>	
						Housing Cost Class
						<br>
						<select v-model="change_type12">
							<option> Remain unchanged </option>
							<option v-for="option in economy['Animal Housing Costs']" v-bind:value="option.Name">
							{{ option.Name }}
							</option>
						</select>							
						 </td>						 
						 <td>
						 	<button class="sub_button" v-on:click="ExecuteChanges()"> Execute Changes </button>
						 </td>
					</tr>
				</table>
				<br>
			</div>					
			<div id="Breeding_Scheme">
				<br> <br>				
				<h3> Breeding Scheme </h3> 		
				<button class="accordion"> Press here to get info on how this module works!</button>
				<div class="guideline">
					<p class="guidelines"> 
						Draw your breeding scheme in the following interactive environment.  <br>
						
						&#8226 Nodes represent cohorts of individuals with similar/same characteristics (e.g. age / sex / genetic origin) <br>
						&#8226 Edges represent breeding actions that can be taken in a breeding program <br>
						
						<br>
						&#8226 For each breeding action and node you will be provide with a lot more options are generation <br>
						&#8226 On default only basic options are displayed. In case you are missing something you can most likely activate it <br>
						the General information > Advanced Settings > Advanced Edge/Node options <br>
						&#8226 If not and you think its useful - Contact us! <br>
						
						<br>						
						
						&#8226 Add a Node : Click Edit -> Add Node  -> Click in an empty space to place a new node.  <br>
						&#8226 Edit a Node : Select a Node from the diagram -> Click Edit Node <br>
						&#8226 Drag a Node : Add/Edit a Node or Edge -> Use grey area to drag the Node/Edge display box. <br>
						&#8226 Double-Click a Node/Edge : Display-only box appears for a node or edge. Need to click Edit Node to edit the node/edge.<br>
						&#8226 Copy Node : Select a Node -> Right Click.
					</p>									
				</div> 
				
				<div id="mynetwork_container" >
					<div id="mynetwork_label">
						<h4>Legends</h4><br>
						Nodes:
						<h5 v-for="key in Object.keys(node_colors)" v-bind:style="{color: node_colors[key], fontFamily: 'Calibri', fontSize: 18 + 'px', fontWeight: 'bold'}"> &#9645;  {{key}}</h5>
					<br>
						Edges:<br>
						<p v-for="key in Object.keys(edge_colors)" v-bind:style="{color: edge_colors[key], fontFamily: 'Calibri', fontSize: 18 + 'px', fontWeight: 'bold'}"> &#8599;  {{key}}</p>

					</div>
					<div id="mynetwork"></div>
				</div>
			</div>

			<div id="node-popUp" onclick="checkEverything(this.id)">
				<div id="node-popUpheader"> 
					<p class="operation"> {{node_operation}} </p> <p class="drag">Drag Node in this Area! </p> 
					<p class="title"> Node {{ active_node.id }}</p>
				</div> 
				<br>
				<table style="margin:auto; width:100%">
					<colgroup>
						<col width="40%">
						<col width="60%">
					</colgroup>
					
					<tr v-if="geninfo['advanced'] && geninfo['advanced_advanced'] && geninfo['advanced_advanced_copy']">
						<td> Copy Node-Options </td>
						<td>
							<select v-model="active_node['copy']">
								<option v-for="option in cohort_ids" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
						<td>
							<button onclick="ImportNode(data_Vue.active_node.copy)" >Import</button>
						</td>
					<tr>
					
					<tr>
						<td>Name
							<span class="tooltip">&#9432 <span class="tooltiptext">Name of the cohort. Please avoid names with : or _ . In particular avoid trailing numbers like ABC_1. Repeated nodes will use this syntax!</span></span>
						</td><td><input type="text" placeholder="Type a name" v-model="active_node.id" ></td>
					</tr>
					<tr>
						<td>Number of individuals
							<span class="tooltip">&#9432 <span class="tooltiptext">Number of individuals in this cohort</span></span>
						</td>
						<td><input type="text" placeholder="Type number/Select variable" id="individuals" list="indVar" v-model="active_node.individualsVar" onchange="updateIndividuals(value)">
							<datalist id="indVar">
								<option v-for="ivar in individualsVar_options" v-bind:value="ivar.name">
							</datalist>
						</td>	
					</tr>
					
					<tr v-show= "geninfo['advanced'] && geninfo['advanced_advanced'] && geninfo['advanced_advanced_scaling'] && (geninfo['individual-scaling'] != '' || geninfo['individual-scaling'] != 1)">
						<td>Ignore Size scaling
							<span class="tooltip">&#9432 <span class="tooltiptext">Are the individuals in this cohort founders or are they generated?</span></span>
						</td>
						<td> 
							<input type="checkbox" v-model="active_node.active_scaling">
						</td>
					</tr>
					
					<tr>
						<td>Founder
							<span class="tooltip">&#9432 <span class="tooltiptext">Are the individuals in this cohort founders or are they generated?</span></span>
						</td>
						<td> 
							<input type="checkbox" false-value="No" true-value="Yes" v-model="active_node.Founder">
						</td>
					</tr>
					<tr v-show="active_node.Founder == 'Yes'">
						<td> Genotype generation type 
							<span class="tooltip">&#9432 <span class="tooltiptext">How to generate genotypes/haplotypes for founder individuals</span></span>
						</td>
						<td>
							<select v-model="active_node['Genotype generation']">
								<option v-for="option in Genotype_generation_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-show="active_node.Founder == 'Yes' & active_node['Genotype generation'] == 'Upload Genotypes'">
						<td>Upload genotypes (plink/vcf)
							<span class="tooltip">&#9432 <span class="tooltiptext">You can provide a dataset in plink/vcf format. Only used markers that are included in the map. Make sure your dataset is phased or activate phasing via checking Phasing required! </span></span>
						</td>
						<td> 
							<form id='GenoUpload' target="print_popup" action='/GenoUpload' method='post' encType='multipart/form-data' onsubmit="window.open('about:blank','print_popup','width=1000,height=800'); setGenoPath();">
								<input type='file' name='genoFile' id='genoFile'>
								<input type='submit' value='Upload'>
							</form>
						</td>
					</tr>

					<tr v-show="active_node.Founder == 'Yes' & active_node['Genotype generation'] == 'Upload Genotypes' ">
						<td> Path genotypes </td><td>{{active_node.Path}}</td>
					</tr>	
					<tr v-show="active_node.Founder == 'Yes' & active_node['Genotype generation'] == 'Upload Genotypes' ">
						<td>Phasing required
							<span class="tooltip">&#9432 <span class="tooltiptext">If imputation via BEAGLE 5.0 default will be automatically processed. This is using the MoBPS function pedmap.to.phasebeaglevcf(). It is not supported when running MoBPS on your own system as paths will usally not match! Contact Torsten if interested to use this locally.</span></span>
						</td>
						<td>
							<input type="checkbox" false-value="No" true-value="Yes" v-model="active_node.phasing_required">
						</td>
					</tr>
					<tr v-show="active_node.Founder == 'Yes' & geninfo['advanced'] & geninfo['advanced_sub']">
						<td>Population allele frequencies
							<span class="tooltip">&#9432 <span class="tooltiptext">Pick the originating subpopulation. Allele frequencies for nodes with no genotype data from the same subpopulation will have similar allele frequencies.</span></span>
						</td>
						<td>
							<select v-model="active_node['Genotype generation subpopulation']">
								<option v-for="option in subpopulation['subpopulation_list']" v-bind:value="option.Name">
								{{ option.Name }}
								</option>
							</select>
						</td>
					</tr>					
					<tr>
						<td>Sex
							<span class="tooltip">&#9432 <span class="tooltiptext">Sex of the individuals in this cohort</span></span>
						</td>
						<td> 
							<select v-model="active_node.Sex">
								<option v-for="option in Sex_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr>
						<td>Phenotyping Class
							<span class="tooltip">&#9432 <span class="tooltiptext">Choose one of the phenotyping classes generated prior. Default is observation of all traits.</span></span>
						</td>
						<td> 
							<select v-model="active_node['Phenotyping Class']">
								<option v-for="option in phenotyping_class" v-bind:value="option.Name">
								{{ option.Name }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] && geninfo['advanced_eco']">
						<td>Housing Cost Class
							<span class="tooltip">&#9432 <span class="tooltiptext">Choose the housing class for the individuals in this cohort</span></span>
						</td>
						<td> 
							<select v-model="active_node['Housing Cost Class']">
								<option v-for="option in economy['Animal Housing Costs']" v-bind:value="option.Name">
								{{ option.Name }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-if="geninfo['advanced'] && geninfo['advanced_advanced'] && geninfo['advanced_advanced_geno']">
						<td>Proportion of genotyped individuals
							<span class="tooltip">&#9432 <span class="tooltiptext">Select the share of genotyped individuals. Mostly relevant for the use of single-step and economic calculations.</span></span>
						</td>
						<td> <input placeholder="Type a number" type="number" max=1 min=0 step=0.001 v-model="active_node['Proportion of genotyped individuals']"> </td>
					</tr>
					
					<tr v-if="geninfo['advanced'] && geninfo['advanced_advanced'] && geninfo['advanced_advanced_del']">
						<td> Delete cohort information
							<span class="tooltip">&#9432 <span class="tooltiptext">Setting this will automatically delete recombination history etc. of the cohort to reduce file size.</span></span>
						</td>
													
						<td> <input type="checkbox" v-model="active_node['delete_info']"> </td>
					</tr>
					<!--tr>
						<td>Include in Plot </td> 
						<td> <input type="checkbox" false-value="No" true-value="Yes" v-model="active_node['BV Plot']"> 
						</td>
					</tr-->	
					<tr v-show="active_node.Sex == 'Both'">
						<td>Proportion of Male</td>
						<td> <input placeholder="Type a number" type="number" max=1 min=0 step=0.001 v-model="active_node['Proportion of Male']"> </td>
					</tr>
				</table>
				<br>
				<input class='button_type' type="button" value="Save" id="node-saveButton" />
				&nbsp;&nbsp;&nbsp;<input class='button_type' type="button" value="Cancel" id="node-cancelButton" />
			</div>
			
			<div id="edge-popUp" onclick="checkEverything(this.id)">
				<div id="edge-popUpheader"> 
					<p class="operation"> {{edge_operation}} </p><p class="drag"> Drag Edge in this Area! </p>
					<p class="title">Edge {{ active_edge.id }}</p>
				</div>
				
				<table style="margin:auto; width: 100%;">
					<colgroup>
						<col width="45%">
						<col width="55%">
					</colgroup>
					<br>
					
					<tr v-if="geninfo['advanced'] && geninfo['advanced_advanced'] && geninfo['advanced_advanced_copy']">
						<td> Copy Edge-Options </td>
						<td>
							<select v-model="active_edge['copy']">
								<option v-for="option in edge_ids" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
						<td>
							<button onclick="ImportEdge(data_Vue.active_edge.copy)" >Import</button>
						</td>
					<tr>
						<td>Breeding Type
							<span class="tooltip">&#9432 <span class="tooltiptext">Selected the breeding action you want to use to generate new cohorts and/or link them. </span></span>
						</td>
						<td> 
							<select v-model="active_edge['Breeding Type']">
								<option v-for="option in Breedingtype_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-if="active_edge['Breeding Type'] == 'Split' && edge_operation == 'Add Edge'">
						<td>ID of 2.Sub-Group
							<span class="tooltip">&#9432 <span class="tooltiptext">Some breeding action require multiple cohorts</span></span>
						</td>
						<td> 
							<input id="split_input" list="splits" placeholder="Create/Choose ID">
							<datalist id="splits">
								<option v-for="par in nodes.get()" v-if="par.Sex == nodes.get(active_edge.from).Sex & par.id != nodes.get(active_edge.to).id & par.id != nodes.get(active_edge.from).id" v-bind:value="par.id">
							</datalist>
						</td>
					</tr>
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production','Semen-collection', 'Aging', 'Selection', 'Split','Combine'].indexOf(active_edge['Breeding Type']) >= 0 ">
						<td>Time needed
							<span class="tooltip">&#9432 <span class="tooltiptext">Provide the time this breeding action takes. Reproduction currently needs at least one time unit.</span></span>
						</td>
						<td><input style="width:40px;border:2px solid lightblue;font-size:12pt;height:22px;" id="time_n" list="indVar" onchange="updateVar('Time Needed', value)" placeholder="Type a number" min="0" max="10000" step="1" v-model="active_edge['Time Needed']"> {{geninfo['Time Unit']}}</td>
					</tr>	
					<tr v-show="geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_last']">
						<td>Use last available
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will use the last repeat available from that node - not necessary the node from the same repeat</span></span>
						</td>		
						<td><input type="checkbox" v-model="active_edge['last_available']"> </td>
					</tr>
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_ogc']">
						<td>OGC
							<span class="tooltip">&#9432 <span class="tooltiptext">Use optimum genetic contribution according to Wellmann et al. 2019 to determine how often each individual is used for reproduction.</span></span>
						</td>
						<td><input type="checkbox" false-value="No" true-value="Yes"  v-model="active_edge['OGC']"></td>
					</tr>	
					
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['OGC']=='Yes'">
						<td>OGC Target 
							<span class="tooltip">&#9432 <span class="tooltiptext">Variable names according to Wellmann et al. 2019: min.sKin (minimize kinship), max.BV (maximize genomic value), min.BV (minimize genomic value)</span></span>
						</td>
						<td>
							<select v-model="active_edge['ogc_target']">
								<option v-for="option in ogc_targets" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>	
					
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['OGC']=='Yes'">
						<td>OGC Relationship matrix 
							<span class="tooltip">&#9432 <span class="tooltiptext">Relationship matrix used in OGC</span></span>
						</td>
						<td>
							<select v-model="active_edge['ogc_relation']">
								<option v-for="option in RelationshipMatrixOGC_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['OGC']=='Yes'">
						<td>OGC constrain 1 </td>
							<span class="tooltip">&#9432 <span class="tooltiptext">Variable names according to Wellmann et al. 2019: ub.bv (upper bound on genomic value), eq.bv (target genomic value), lb.bv (lower bound genomic value), ub.sKin (upper bound for avg. kinship), uniform (equal contributes of male/females), lb.BV.increase (minimum increase in genomic value), ub.sKin.increase (maximum increase in avg. relationship)</span></span>
						<td>
							<select v-model="active_edge['ogc_constrain1']">
								<option v-for="option in ogc_constrain" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
						<td v-if="active_edge['ogc_constrain1'] != 'inactive' && active_edge['ogc_constrain1'] != 'uniform'"><input placeholder="Type in the constrain value." v-model="active_edge['ogc_constrain1_value']"></td>
						<td v-if="active_edge['ogc_constrain1'] == 'uniform'">
							<select v-model="active_edge['ogc_constrain1_value']">
								<option v-for="option in ogc_uniform" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['OGC']=='Yes' && active_edge['ogc_constrain1'] != 'inactive'">
						<td>OGC constrain 2 </td>
						<td>
							<select v-model="active_edge['ogc_constrain2']">
								<option v-for="option in ogc_constrain" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
						<td v-if="active_edge['ogc_constrain2'] != 'inactive' && active_edge['ogc_constrain2'] != 'uniform'"><input placeholder="Type in the constrain value." v-model="active_edge['ogc_constrain2_value']"></td>
						<td v-if="active_edge['ogc_constrain2'] == 'uniform'">
							<select v-model="active_edge['ogc_constrain2_value']">
								<option v-for="option in ogc_uniform" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>

					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['OGC']=='Yes' && active_edge['ogc_constrain1'] != 'inactive' && active_edge['ogc_constrain2'] != 'inactive'">
						<td>OGC constrain 3 </td>
						<td>
							<select v-model="active_edge['ogc_constrain3']">
								<option v-for="option in ogc_constrain" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
						<td v-if="active_edge['ogc_constrain3'] != 'inactive' && active_edge['ogc_constrain3'] != 'uniform'"><input placeholder="Type in the constrain value." v-model="active_edge['ogc_constrain3_value']"></td>
						<td v-if="active_edge['ogc_constrain3'] == 'uniform'">
							<select v-model="active_edge['ogc_constrain3_value']">
								<option v-for="option in ogc_uniform" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>					
					
					<tr v-if="['Reproduction'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_mating']">
						<td>Avoid full-sib matings
							<span class="tooltip">&#9432 <span class="tooltiptext">Activate this to not use full-sibling pairs for reproduction.</span></span>
						</td>
						<td><input type="checkbox" v-model="active_edge['full_sib']"></td>
					</tr>
					
					<tr v-if="['Reproduction'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_mating']">
						<td>Avoid half-sib matings
							<span class="tooltip">&#9432 <span class="tooltiptext">Activate this to not use half-sibling pairs for reproduction.</span></span>
						</td>
						<td><input type="checkbox" v-model="active_edge['half_sib']"></td>
					</tr>
					
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_off']">
						<td>Maximal # offspring
							<span class="tooltip">&#9432 <span class="tooltiptext">Maximum number of offspring each individual from the parent node is allowed to have.</span></span>
						</td>
						<td><input placeholder="Type a number or Inf" v-model="active_edge['Max Offspring']"></td>
					</tr>	
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_off']">
						<td>Maximal # offspring from one mating pair
							<span class="tooltip">&#9432 <span class="tooltiptext">Maximum number of offspring from one mating pair.</span></span>
						</td>
						<td><input placeholder="Type a number or Inf" v-model="active_edge['Max Offspring Pair']"></td>
					</tr>	
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_ratio']">
						<td>Selection ratio
							<span class="tooltip">&#9432 <span class="tooltiptext">Use this if you want to use some individuals more frequently for reproduction. A value of 2 will lead to the individual will the highest genomic value to be used twice as often as the worst individual.</span></span>
						</td>
						<td><input placeholder="Type a number (if nothing enter all individuals are picked equally)" v-model="active_edge['selection_ratio']"></td>
					</tr>	
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_ratio']">
						<td>Selection ratio type
							<span class="tooltip">&#9432 <span class="tooltiptext">Use this if you want to use some individuals more frequently for reproduction. A value of 2 will lead to the individual will the highest genomic value to be used twice as often as the worst individual.</span></span>
						</td>
						<td> 
							<select v-model="active_edge['selection_ratio_type']">
								<option v-for="option in selection_type" v-bind:value="option">
								{{ option}}
								</option>
							</select>
						</td>					</tr>	
					<tr v-if="['Reproduction', 'Cloning', 'Selfing', 'DH-Production', 'Semen-collection'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] && (geninfo['advanced_advanced_ogc'] ||  geninfo['advanced_advanced_ratio'])">
						<td>Index for Selection ratio / OGC
							<span class="tooltip">&#9432 <span class="tooltiptext">Select the index used for the selection ratio or OGC. </span></span>
						</td>
						<td> 
							<select v-model="active_edge['selection_ratio_index']">
								<option v-for="option in selection_index" v-bind:value="option">
								{{ option.Name }}
								</option>
							</select>
						</td>
					</tr>	
					<tr v-if="active_edge['Breeding Type'] == 'Reproduction' && edge_operation == 'Add Edge'">
						<td>ID of 2.parent </td> 
						<td> 
							<input id="parents_input" list="parents" placeholder="Create/Choose ID">
							<datalist id="parents">
								<option v-for="par in nodes.get()" v-if="par.Sex != nodes.get(active_edge.from).Sex & par.id != nodes.get(active_edge.to).id" v-bind:value="par.id">
							</datalist>
						</td>
					</tr>
					<tr v-show="active_edge['Breeding Type'] == 'Repeat'">
						<td>Number of Repeat 
							<span class="tooltip">&#9432 <span class="tooltiptext">Number of times the Repeat is executed. The parent node will be the new input of the child node and everything in between is rerun. Nodes will be named the same + _1, _2, _3 etc.</span></span>
						</td> 
						<td> 
							<input list="indVar" placeholder="Type a number" v-model="active_edge['Number of Repeat']" onchange="updateVar('Number of Repeat', value)">
						</td>
					</tr>
					<!--tr v-show="active_edge['Breeding Type'] == 'Recombination'">
						<td>New Mutation Rate </td> 
						<td> <input onchange="updateVar('New Mutation Rate', value)" placeholder="Type a number" v-model="active_edge['New Mutation Rate']" list="indVar"></td>
					</tr>
					<tr v-show="active_edge['Breeding Type'] == 'Recombination'">
						<td>New Remutation Rate </td> 
						<td> <input onchange="updateVar('New Remutation Rate', value)" placeholder="Type a number" v-model="active_edge['New Remutation Rate']" list="indVar"></td>
					</tr>
					<tr v-show="active_edge['Breeding Type'] == 'Recombination'">
						<td>Number of Recombination per M </td> 
						<td> <input onchange="updateVar('Number of Rec per M', value)" placeholder="Type a number" v-model="active_edge['Number of Rec per M']" list="indVar"></td>
					</tr -->
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 ">
						<td>Selection Type
							<span class="tooltip">&#9432 <span class="tooltiptext">Random: Each individual is taken with same probability, BVE/Pheno selected based on BVE/Phenotypes</span></span>
						</td>
						<td> 
							<select v-model="active_edge['Selection Type']">
								<option v-for="option in selectionType_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE'">
						<td>BVE Method
							<span class="tooltip">&#9432 <span class="tooltiptext">Direct-Mixed-Model assumes known heritability and thereby is fastest. EMMREML/sommer use the respective R-package. Sommer also support multi-variable models but takes much longer! Bayesian models used are using BGLR. MAS is using lm() on randomly sampled effect markers.</span></span>
						</td>
						<td> 
							<select v-model="active_edge['BVE Method']">
								<option v-for="option in BVEMethod_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE' & active_edge['BVE Method']=='Marker assisted selection (lm)'">
						<td>Number of markers used for MAS
							<span class="tooltip">&#9432 <span class="tooltiptext">This is a very simple implementation of MAS using a linear model of randomly sampled effect markers</span></span>
						</td>
						<td><input placeholder="Type a number or Inf" v-model="active_edge['MAS_marker']"></td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 & active_edge['Selection Type']=='Pseudo-BVE'" v-for='(trait, index) in traitsinfo' >
						<td>Accuracy for Trait {{ trait[["Trait Name"]] }} 
							<span class="tooltip">&#9432 <span class="tooltiptext">If nothing is entered here the accuracy will be chosen according to the starting heritability</span></span>
						</td>
						<td>
							<input v-model="active_edge[index]" style="width:80px;"> 
						</td>
					</tr>					
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 & traitsinfo.length > 1">
						<td>Selection Index
						<span class="tooltip">&#9432 <span class="tooltiptext">Select which index to use for BVE/Phenotypic selection</span></span>
						</td>
						<td> 
							<select v-model="active_edge['Selection Index']">
								<option v-for="option in selection_index" v-bind:value="option.Name">
								{{ option.Name }}
								</option>
							</select>
						</td>
					</tr>						
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0">
						<td>Selection Proportion
							<span class="tooltip">&#9432 <span class="tooltiptext">Selection intensity - the tool will automatically calculate this for you based on the size of the parent/child node.</span></span>
						</td>
						<td>
							<!-- <input list="indVar" onchange="updateVar('Selection Proportion', value)" placeholder="Type a number" min="0" max="1" step="0.001" v-model="selection_proportion"> -->
							{{ selection_proportion }} (calculated based on # Individuals)
						</td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_thres']">
						<td>Threshold for selection
							<span class="tooltip">&#9432 <span class="tooltiptext">Calculate reliability of BVE according to vanRaden 2008</span></span>
						</td>		
						<td><input type="value" placeholder="Type a number if wanted" v-model="active_edge['threshold']"></td>
					</tr>
					
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_thres']">
						<td>Threshold sign
							<span class="tooltip">&#9432 <span class="tooltiptext"> Only keep individuals with higher/lower/eval estimated breeding value</span></span>
						</td>		
						<td> 
							<select v-model="active_edge['threshold_sign']">
								<option v-for="option in threshold_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type']=='BVE' & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_reli']">
						<td>Calculate reliability
							<span class="tooltip">&#9432 <span class="tooltiptext">Calculate reliability of BVE according to vanRaden 2008</span></span>
						</td>		
						<td><input type="checkbox" false-value="No" true-value="Yes" v-model="active_edge['reliability']"> </td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type']=='BVE' & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_skip']">
						<td> Skip traits with no index weight
							<span class="tooltip">&#9432 <span class="tooltiptext">Activating this will lead to not performing BVE that are not assign with any weight (=0) in the selection index.</span></span>
						</td>		
						<td><input type="checkbox" false-value="No" true-value="Yes" v-model="active_edge['skip']"> </td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 & geninfo['advanced'] && geninfo['advanced_advanced'] & geninfo['advanced_advanced_pheno']">
						<td>Input Phenotype
							<span class="tooltip">&#9432 <span class="tooltiptext">Use the avg. phenotype of the offspring as the phenotype of the individuals in the parent node. Make sure there are offspring! (use of time needed!)</span></span>
						</td>
						<td>
						<select v-model="active_edge['phenotype_used']">						
								<option v-for="option in phenotype_options" v-bind:value="option">
								{{ option }}
								</option>								
							</select>
						</td>
					</tr>
					
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE'">
						<td>Relationship Matrix
							<span class="tooltip">&#9432 <span class="tooltiptext">According to vanRaden 2008, pedigree-based or singleStep (H based on Legarra 2014)</span></span>
						</td>
						<td> 
							<select v-model="active_edge['Relationship Matrix']">
								<option v-for="option in RelationshipMatrix_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE' && ( active_edge['Relationship Matrix']== 'Single Step'|active_edge['Relationship Matrix']== 'Pedigree')">
						<td>Depth of Pedigree (# generation back)
							<span class="tooltip">&#9432 <span class="tooltiptext">Depth of the pedigree. All individuals before are assumed to be unrelated! If you need something else contact Torsten or use the R-package.</span></span>
						</td> 
						<td> <input type="value" onchange="updateVar('Depth of Pedigree', value)" placeholder="Type a number" v-model="active_edge['Depth of Pedigree']" list="indVar"></td>
					</tr>

					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE'">
						<td>Cohorts used in BVE
							<span class="tooltip">&#9432 <span class="tooltiptext">Manual select provides you with maximum flexibility to select which individuals to use.</span></span>
						</td>
						<td> 
							<select v-model="active_edge['Cohorts used in BVE']" onchange="cohorts_Div(this);">
								<option v-for="option in Cohorts_options" v-bind:value="option">
								{{ option }}
								</option>
							</select>
						</td>
					</tr>
					<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE' && active_edge['Cohorts used in BVE'] == 'Manual select'">
						<td>Select cohorts for manuel select:
							<span class="tooltip">&#9432 <span class="tooltiptext">Cohorts available for selection are only refresh on reloading or when EDITING an edge. We are working on that.</span></span>
						</td>
					</tr>
					</table>
					<br>
					<input class='button_type' type="button" value="Save" id="edge-saveButton" />
					&nbsp;&nbsp;&nbsp;<input class='button_type' type="button" value="Cancel" id="edge-cancelButton" />
				</div>
				<br>

				<div id="edge_cohorts_Div" >				
					<table >					
						<tr v-show="['Selection', 'Split', 'Aging'].indexOf(active_edge['Breeding Type']) >= 0 && active_edge['Selection Type'] == 'BVE' && active_edge['Cohorts used in BVE'] == 'Manual select'">

						<td v-if="isCohortNameChanged ==='' && edge_operation === ('Add Edge')" >						
							<treeselect :value-consists-of="'LEAF_PRIORITY'" :multiple="true" :disabled="false" :flat="false" :always-open="true" :options="allNode_manualselect_options" v-model="active_edge['Manuel selected cohorts']" :searchable="true"/>
						</td>
						<td v-if="isCohortNameChanged ==='' && edge_operation === ('Edit Edge')" >						
							<treeselect :value-consists-of="'LEAF_PRIORITY'" :multiple="true" :disabled="false" :flat="false" :always-open="true" :options="node_manualselect_options" v-model="active_edge['Manuel selected cohorts']" :searchable="true" />
						</td>
						<td v-if="isCohortNameChanged ==='Yes'" >	
							<treeselect :value-consists-of="'LEAF_PRIORITY'" :multiple="true" :disabled="false" :flat="false" :always-open="true" :options="updated_manualselect_options" v-model="active_edge['Manuel selected cohorts']" :searchable="true"/>
						</td>
						</tr>
					</table>
				</div>
				<br>
        
				<button class="accordion"> Press here to get info on how this module works!</button>
				<div class="guideline">
					<p class="guidelines"> 
						Intension of this module is to aline nodes in the flash application to get nice graphs <br>			
						&#8226 All nodes with similar x or y coordinate will be put to the same value in the respective axis <br>
						&#8226 Increase values to align more nodes / Reduce values to align less nodes <br>
						&#8226 Expand flash environment will increase the size of the flash environment <br>
					</p>									
				</div> 				
					
				<table style="margin:left; width: 85%;">
				<tr>
					<td>Max X-Axis difference</td>
					<td> <input placeholder="Max x-axis difference" type="text" v-model="geninfo['max_d']">	</td>
					<td>Max Y-Axis difference</td>
					<td><input placeholder="Max y-axis difference" type="text" v-model="geninfo['max_d2']">	</td>
					<td><button class="sub_button" onclick="alignNodes()"> Align nodes </button></td>
				</tr>
				</table>
	
				<p> Expand flash enviroment1 : <span class="tooltip">&#9432 <span class="tooltiptext"> Please click this button to have bigger breeding scheme diagram and save the project to rearrange nodes and edges.</span></span>
					 <input id="heightDefault" v-model = "geninfo['slider_value']" type="range" min="0" max="6" value="0" onchange="heightNetwork.value=value;networkBoxChange(this.value);"></input> 
					<output id="heightNetwork"></output>
				</p>
	
				<br>

				<div v-if = "geninfo['Project Name']" id="Results_Div">
				<h3 id="RSim"></h3>
				<br>
				<br>
				<h3>Analyze Breeding Program with R</h3>
				<p>Click on button 'Start R Simulation' to run R.</p>
				
				<h4 style="color: red" v-if="warnings.length > 0"> Please check warnings before running the simulation!</h4>
				<button v-if="warnings.length > 0" v-on:click='show_warnings = !show_warnings; show_geninfo=false;' >Show/Hide Warnings</button>
				<button style="font-size:18px;" onclick="runningR1(); " v-if="warnings.length > 0">Start R Simulation (despite warnings!)</button>
				<button style="font-size:18px;" onclick="runningR1(); " v-if="warnings.length == 0">Start R Simulation</button>
				<br>
				<br>
				<p> Click here to estimate computing time / costs of your breeding program </p>
				<button onclick="runningR2(); " v-if="warnings.length == 0">Estimate time/cost</button>

				<br>
				<p>Or click here to restore the latest R result that you have already generated for the breeding program {{geninfo['Project Name']}}.</p>
				<button onclick="ReloadSim(); " >Reload Old Results</button>	
				<br>
				<!--
				<p> <a v-if="filename" href="/Rdownload"> Download {{geninfo['Project Name']}}.RData </a> </p>
				<p> <a v-if="filename" href="/Exceldownload"> Download {{geninfo['Project Name']}}.csv </a> </p> 

				<table>
				<tr>
					<td> Cohorts Information :
						<span class="tooltip">&#9432 <span class="tooltiptext">List of all cohorts and some basic information (time of generation, economic stuff etc.)</span></span>
					</td>
					<td>
						<textarea v-model="geninfo['listOfCohorts_withInfo']" placeholder="add multiple lines" id="listOfCohorts" style="width:450px;height:50px;"></textarea>						
					</td>
				</tr>
				</table>
				-->
				
				<br>
					<div v-if="warningsLog.length >2" style="color:red;" v-for="(number, index) in warningsLog" :key="index">
						{{warningsLog[index]}}
					</div>
				<br>
				
				<div v-if="warningsLog.length >2" style="color:red;" v-for="(number, index) in warningsLog" :key="index">
						{{warningsLog[index]}}
				</div>
				<br>
				
				<p v-if="cohortsList.length >0" class="boldText">  Download Rdata : 
					<span class="tooltip">&#9432 <span class="tooltiptext"> Download the output of the simulation as Rdata file to save it on your own machine or use it for further analysis in R. </span></span> &nbsp;&nbsp;
					<a v-if="filename" href="/Rdownload" onclick="alert('Genotype coding can vary between miraculix-versions / compilers! In case you plan to export genotype from the RData-object please deactivate miraculix for now (Advanced settings)!!!')"> {{geninfo['Project Name']}}.RData </a> 
				</p>
				<br>
				<p>Attention: The actual breeding program must match the uploaded R results, otherwise plotting functions will not work.</p>
				<form id='ResultUpload' target="print_popup" action='/ResultUpload' method='post' encType='multipart/form-data' onsubmit="window.open('about:blank','print_popup','width=1000,height=800')">
					<input type='file' name='ResultFile' id='ResultFile'>
					<!--input type='submit' value='Load Results'-->
					<button onclick="UploadSim();">Load Results</button>
				</form>
			</div>

			<br>
			
			<p v-show="runned">Click here to clear all results and plots.</p>
			<button v-show="runned" onclick="clearResult()" >Clear Results</button>
			<button v-show="runned" v-on:click="rlog = !rlog" >Show/Hide R Log</button>
			<br>
			<br>
			<div v-show="runned & rlog" id="Rout_Div">

			</div>
			<br>
			<br>
			<div v-show="runned">
				<h4 style="color: 'red'">Results: Summary</h4>
				<div id="Result_Summary">
				</div>

				<br>
				<p v-if="cohortsList.length >0" class="boldText"> Download Cohorts List: 
				<span class="tooltip">&#9432 <span class="tooltiptext"> Download CSV file containing cohort names, number of individuals, time-point of generation, total costs, genotyping costs, cost phenotyping and  housing costs.</span></span>&nbsp; 
				<a v-if="filename" href="/Exceldownload"> Download {{geninfo['Project Name']}}.csv </a> 
				</p>
				<br>	
				<div v-if="cohortsList.length >0">
					<table  class="scrollableTable1">
						<thead>
							<tr>
								<th style="width:15%; text-align:left;"> Cohort Name </th>
								<th>Nr. of Individuals </th>
								<th>Time point </th>
								<th>Total costs </th>
								<th>Cost genotyping </th>
								<th>Cost phenotyping </th>
								<th>Cost housing </th>
							</tr>
						</thead>
						<tbody>							
							<tr v-for='(cohorts, index) in cohortsList' >
								<td style="width:15%; text-align:left;"> {{cohortsList[index]['Cohort name']}} </td>
								<td> {{cohortsList[index]['Nr. of individuals']}} </td>
								<td> {{cohortsList[index]['Time-point']}} </td>
								<td> {{cohortsList[index]['Total costs']}} </td>
								<td> {{cohortsList[index]['Cost genotyping']}} </td>
								<td> {{cohortsList[index]['Cost phenotyping']}} </td>
								<td> {{cohortsList[index]['Cost housing']}} </td>			
							</tr>
						</tbody>
					</table>
				</div>

				<p v-if="cohortsList.length >0" class="boldText"> Download Cohorts List: 
					<span class="tooltip">&#9432 <span class="tooltiptext"> Download CSV file containing cohort names, number of individuals, time-point of generation, total costs, genotyping costs, cost phenotyping and  housing costs.</span></span>&nbsp; 
					<a v-if="filename" href="/Exceldownload"> Download {{geninfo['Project Name']}}.csv </a> 
				</p>
				<br>	

				<p v-if="cohortsTimeList.length >0" class="boldText">Computing time estimates : </p>
				<div v-if="cohortsTimeList.length >0">
					<table class="scrollableTable2">
						<thead>
							<tr>
								<th style="width:20%; text-align:left;">Cohort Name </th>
								<th>BVE Time </th>
								<th>Generation Time </th>
								<th>TOTAL Time </th>
							</tr>
							</thead>
						<tbody>
							<tr v-for='(cohorts, index) in cohortsTimeList' >
								<td style="width:20%; text-align:left;"> {{cohortsTimeList[index]['Cohort name']}} </td>
								<td> {{cohortsTimeList[index]['BVEtime']}} </td>
								<td> {{cohortsTimeList[index]['Gentime']}} </td>
								<td> {{cohortsTimeList[index]['Totaltime']}} </td>			
							</tr>
						</tbody>
					</table>
				</div>
				
				<button class="accordion"> Press here to get info on how this module works!</button>
				<div class="guideline">
					<p class="guidelines"> 
						The following modules allow you to download information on the simulated data<br>			
						&#8226 The population list itself is stored as an RData-object <br>
						&#8226 In case you are not using miraculix you will not be able to decode underlying genotypes. If that is needed please deactive miraculix in the Advanced settings > miraculix-active <br>
						
						<br>
						&#8226 Before downloading specific data files (VCF / phenotypes etc) press the Prepare data. <br>
						&#8226 selecting repeat = 0 means the node itself. The first repeat (1) is the second cycle of the breeding program!
					</p>									
				</div> 
											
				<h4> Download population list: </h4>
				<div class="populationList">	
					<a href="/Rdownload" style="background-color:#eee;" onclick="alert('Genotype coding can vary between miraculix-versions / compilers! In case you plan to export genotype from the RData-object please deactivate miraculix for now (Advanced settings)!!!')"> Download RData</a>
					<br>
					<br>
					<h4>Download: Population data</h4>
					<table>
					<tr>
					<td> Selected datatype: </td>
					<select v-model="plottingPar.download1" >
						<option v-for="option in download_data" v-bind:value="option">
						{{ option }}
						</option>
					</select>
					</tr>
					<tr>
					<td> Selected cohort: </td>
					<select v-model="plottingPar.download2" >
						<option>
						ALL
						</option>
						<option v-for="option in nodes.get()" v-bind:value="option">
						{{ option.id }}
						</option>
					</select>
					</tr>
					<tr>
					<td v-if="plottingPar.download2!='ALL'"> Selected repeat: </td>
					<td><input placeholder="Type a number (Default: 0)" v-model="plottingPar.download3">
					</td>
					</tr>
					<tr>
					<td>
					<button onclick="RunDataPreparation()">Prepare data</button>
					<a v-if="plottingPar.download1=='Ped'" href="/peddownload"> Download Ped</a>
					<a v-if="plottingPar.download1=='Map'" href="/mapdownload"> Download Map</a>
					<a v-if="plottingPar.download1=='VCF'" href="/vcfdownload"> Download VCF</a>
					<a v-if="plottingPar.download1!='Ped' && plottingPar.download1!='Map' && plottingPar.download1!='VCF'" href="/txtdownload"> Download txt</a>					
					</td>
					</tr>
					</table>
				</div>					
				<br>
				
				<table>
				<tr>					
				<td>Display 95% Confidence Intervals </td> 
				<td>
					<input type="checkbox" v-model="plottingData.confidence">
				</td>
				</tr>
				<tr>
				<td>Display Legend </td> 
				<td>
					<input type="checkbox" v-model="plottingData.legend">
				</td>
				</tr>
				</table>
				
				<h4>Results: Observed Phenotypes</h4>
				<button v-if="plottingData.RespMean == ''" onclick="RunResultpMean()" >Run Analysis</button>
				<template v-if="plottingData.RespMean != ''">
					Select plotting type:
					<select v-model="plottingPar.RespMean_pType" >
						<option v-for="option in plottingType" v-bind:value="option">
						{{ option }}
						</option>
					</select>
					<br>
					Select cohorts (multiple selection possible): 
					<button onclick="plottingResultpMean()" >Plot Results</button>
					<treeselect :multiple="true" :options="Summary" v-model="plottingPar.RespMean_cohorts" :searchable="true" style="width:800px"/>
					<!--select v-model="plottingPar.RespMean_cohorts" size="6" multiple>
						<option v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
						{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
						</option>
					</select-->
				</template>
				<br>
				<br>
				<div v-show="plottingData.RespMean != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
					<div v-for="(trait, index) in traitsinfo" v-bind:id="'RespMean_Div'+(index+1)">
					</div>
				</div>
				<br>
				<br>
				<h4>Results: True Breeding Values</h4>
				<button v-if="plottingData.ResgMean == ''" onclick="RunResultgMean()" >Run Analysis</button>
				<template v-if="plottingData.ResgMean != ''">
					Select plotting type:
					<select v-model="plottingPar.ResgMean_pType" >
						<option v-for="option in plottingType" v-bind:value="option">
						{{ option }}
						</option>
					</select>
					<br>
					Select cohorts (multiple selection possible): 
					<!--select v-model="plottingPar.ResgMean_cohorts" size="6" multiple>
						<option v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
						{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
						</option>
					</select-->
					<button onclick="plottingResultgMean()" >Plot Results</button>
					<treeselect :multiple="true" :options="Summary" v-model="plottingPar.ResgMean_cohorts" :searchable="true" style="width:800px"/>
				</template>
				<br>
				<br>
				<div v-show="plottingData.ResgMean != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
					<div v-for="(trait, index) in traitsinfo" v-bind:id="'ResgMean_Div'+(index+1)">
					</div>
				</div>
				<br>
				<br>

				<h4>Results: Accuracy of Breeding Value Estimation</h4>
				<button v-if="plottingData.ResAccBVE == ''" onclick="RunResultAccBVE()" >Run Analysis</button>

				<template v-if="plottingData.ResAccBVE != ''">
					Select plotting type:
					<select v-model="plottingPar.ResAccBVE_pType" >
						<option v-for="option in plottingType" v-bind:value="option">
						{{ option }}
						</option>
					</select>
					<br>
					Select cohorts (multiple selection possible):
					<button onclick="plottingResultAccBVE()" >Plot Results</button>
					<treeselect :multiple="true" :options="Summary_AccBVE" v-model="plottingPar.ResAccBVE_cohorts" :searchable="true" style="width:800px"/>
				</template>

				<br>
				<br>
					
				<div v-show="plottingData.ResAccBVE != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
					<div v-for="(trait, index) in selection_index" v-bind:id="'ResAccBVE_Div'+(index+1)">
					</div>
				</div>
				<br>
				<br>

				<h4> Results: Relationship and Inbreeding within Cohorts </h4>
				<table>
					<tr>
					<td>Only consider cohorts named:
						<span class="tooltip">&#9432 <span class="tooltiptext">Test</span></span>
					</td>
					<td>
						<input placeholder="Type a name" v-model="plottingData.consider_cohort">
					</td>
					</tr>
				</table>
				<button onclick="RunResultRel()" >Run Analysis</button>
				<template v-if="plottingData.ResRel != ''">
					Select plotting type:
					<select v-model="plottingPar.ResRel_pType" >
						<option v-for="option in plottingType" v-bind:value="option">
						{{ option }} 
						</option>
					</select>
					<br>
					Select cohorts (multiple selection possible):
					<!--select v-model="plottingPar.ResRel_cohorts" size="6" multiple>
						<option v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
						{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
						</option>
					</select-->
					<button onclick="plottingResultRel()" >Plot Results</button>
					<treeselect :multiple="true" :options="Summary" v-model="plottingPar.ResRel_cohorts" :searchable="true" style="width:800px"/>
				</template>
				<br>
				<br>
				<div v-show="plottingData.ResRel != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
					<div id="ResRel_Div1">
					</div>
					<div id="ResRel_Div2">
					</div>
				</div>
				<br>
				<br>
<!--
				<h4> CURRENTLY INACTIV - Results: Relationship between Cohorts </h4>
				<button v-if="plottingData.ResRelbetweenC == ''" onclick="RunResultRelbetweenC()" >Run Analysis</button>
-->
				<template v-if="plottingData.ResRelbetweenC != ''">
					Select cohort 1 :
					<select v-model="plottingPar.ResRelbetweenC_cohort1">
						<option v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
						{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
						</option>
					</select>
					Select cohort 2 :
					<select v-model="plottingPar.ResRelbetweenC_cohort2">
						<option v-if="option != plottingPar.ResRelbetweenC_cohort1" v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
						{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
						</option>
					</select>
					<button onclick="plottingResultRelbetweenC()" >Plot Results</button>
				</template>

				<div v-show="plottingData.ResRelbetweenC != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
					<div id="ResRelbetweenC_Div1">
					</div>
				</div>
				<br>
				<br>

				<h4> Results: Major QTLs (Allele Frequency, exp./obs. Heterozygosity)</h4>
				<button v-if="plottingData.ResQTL == ''" onclick="RunResultQTL()" >Run Analysis</button>
				<template v-if="plottingData.ResQTL != ''">
				
					Select plotting type:
					<select v-model="plottingPar.ResQTL_pType" >
						<option v-for="option in plottingType2" v-bind:value="option">
						{{ option }} 
						</option>
					</select>
					
					Select Trait :
					<select v-model="plottingPar.ResQTL_trait">
						<option v-if="option['Trait Major QTL'] > 0" v-for="(option, index) in traitsinfo" v-bind:value="index+1">
						{{ option["Trait Name"] }} 
						</option>
					</select>
					Select QTL :
					<select v-if="plottingPar.ResQTL_trait != ''" v-model="plottingPar.ResQTL_qtl">
						<option v-for="(option, index) in traitsinfo[plottingPar.ResQTL_trait -1]['Trait QTL Info']" v-bind:value="index">
						{{ option['QTL ID'] }}  
						</option>
					</select>
					<br>
					Select Cohorts (multiple selection possible):
					<!--select v-model="plottingPar.ResQTL_cohorts" size="6" multiple>
						<option v-for="option in Object.keys(plottingData.Summary)" v-bind:value="option">
						{{ option }}   ({{plottingData.Summary[option].trep - 1}} Repeats)
						</option>
					</select-->
					<button onclick="plottingResultQTL()" >Plot Results</button>
					<treeselect :multiple="true" :options="Summary" v-model="plottingPar.ResQTL_cohorts" :searchable="true" style="width:800px"/>
				</template>
				<br>
				<br>
				<div v-show="plottingData.ResQTL != ''" style="resize: both; overflow: scroll;" class="RSim_Div">
					<div id="ResQTL_Div1">
					</div>
					<div id="ResQTL_Div2">
					</div>				
					<div id="ResQTL_Div3">
					</div>
				</div>
									
				<br>
				<p v-if="cohortsList.length >0" class="boldText">Cohorts List : </p>
				
				<div v-if="cohortsList.length >0">
					<table  class="scrollableTable1">
						<thead>
							<tr>
								<th style="width:15%; text-align:left;"> Cohort Name </th>
								<th>Nr. of Individuals </th>
								<th>Time point </th>
								<th>Total costs </th>
								<th>Cost genotyping </th>
								<th>Cost phenotyping </th>
								<th>Cost housing </th>
							</tr>
						</thead>
						
						<tbody>	
							<tr v-for='(cohorts, index) in cohortsList' >
								<td style="width:15%; text-align:left;"> {{cohortsList[index]['Cohort name']}} </td>
								<td> {{cohortsList[index]['Nr. of individuals']}} </td>
								<td> {{cohortsList[index]['Time-point']}} </td>
								<td> {{cohortsList[index]['Total costs']}} </td>
								<td> {{cohortsList[index]['Cost genotyping']}} </td>
								<td> {{cohortsList[index]['Cost phenotyping']}} </td>
								<td> {{cohortsList[index]['Cost housing']}} </td>			
							</tr>
						</tbody>
					</table>
				</div>
				
				<p v-if="cohortsList.length >0" class="boldText"> Download Cohorts List: 
					<span class="tooltip">&#9432 <span class="tooltiptext"> Download CSV file containing cohort names, number of individuals, time-point of generation, total costs, genotyping costs, cost phenotyping and  housing costs.</span></span>&nbsp; 
					<a v-if="filename" href="/Exceldownload"> Download {{geninfo['Project Name']}}.csv </a> 
				</p>
				<br>					
				
				<p v-if="cohortsTimeList.length >0" class="boldText">Computing time estimates: </p>
				<div v-if="cohortsTimeList.length >0">
					<table class="scrollableTable2">
						<thead>
							<tr>
								<th style="width:20%; text-align:left;">Cohort Name </th>
								<th>BVE Time </th>
								<th>Generation Time </th>
								<th>TOTAL Time </th>

							</tr>
						</thead>
						<tbody>
							<tr v-for='(cohorts, index) in cohortsTimeList' >
								<td style="width:20%; text-align:left;"> {{cohortsTimeList[index]['Cohort name']}} </td>
								<td> {{cohortsTimeList[index]['BVEtime']}} </td>
								<td> {{cohortsTimeList[index]['Gentime']}} </td>
								<td> {{cohortsTimeList[index]['Totaltime']}} </td>			
							</tr>
						</tbody>						
					</table>
				</div>
				</div>
				<br>
				<br>
				
				<div id="Export_Data_H3">
				<h4> JSON-File Viewer: </h4>
				
					<br>
					<button class="sub_button" id="export_button"  onclick="downloadNetwork()" >Display JSON</button>
					<span class="tooltip">&#9432 <span class="tooltiptext">
						<p> The current state of the project will be displayed in the Output Area for further inspection and/or editing
					</p> </span></span>
					<button class="sub_button" id="import_button" onclick="importNetwork_fromBox()" >Import JSON data</button>
					<span class="tooltip">&#9432 <span class="tooltiptext">
						<p> Import JSON-project from the Output Area below
					</p> </span></span>
					<br>
					<br>
					
					<div>
						<textarea id='OutputArea' style="margin-left:177px; width:750px; height:200px; max-height: 750px;"></textarea>
					</div>
					<br>
				</div>			
			</div>
		</div>
	</div>
	</body>
		
	<!-- Eigene Javascripte Species Data -->
	<script type="text/javascript" src="javascript/GenomeData.js"></script>
	
	<!-- Eigene Javascripte Main Code -->
		<script src="javascript/socket.io.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/2.2.0/socket.io.js"></script>
	<script type="text/javascript" src="javascript/home.js"></script>
	<script type="text/javascript" src="javascript/Rmodule.js"></script>	
	
	<!-- Eigene Javascripte Checking and additional Services -->
	<script type="text/javascript" src="javascript/Check_Services.js"></script>

	<!-- Node and Edge PopUp draggable -->
	<script type="text/javascript" src="javascript/node_edge_draggable.js"></script>
	
	<!-- Navigation Menu related -->
	<script type="text/javascript">
		function openNav() {
		  document.getElementById("mySidebar").style.width = "225px";
		  document.getElementById("main").style.marginLeft = "225px";
		 // document.getElementById("openDiv").style.display = "none";
		  document.getElementById('openB').style.visibility = 'hidden';
		  document.getElementById('closeB').style.visibility = 'visible';

		}
		
		function closeNav() {
		  document.getElementById("mySidebar").style.width = "0";
		  document.getElementById("main").style.marginLeft= "0";
		 // document.getElementById("openDiv").style.display = "block";
		  document.getElementById('openB').style.visibility = 'visible';
		  document.getElementById('closeB').style.visibility = 'hidden';
		}
		
				
	function action(hidden) {
	        if(hidden==true) {
	            document.getElementById('node-saveButton').style.visibility = 'hidden';
				document.getElementById('edge-saveButton').style.visibility = 'hidden';
	        } else {
	            document.getElementById('node-saveButton').style.visibility = 'visible';
				document.getElementById('edge-saveButton').style.visibility = 'visible';
	        }
	    }

		var acc = document.getElementsByClassName("accordion");
		var z;
		
		for (z = 0; z < acc.length; z++) {
		  acc[z].addEventListener("click", function() {
		    this.classList.toggle("active");
		    var panel = this.nextElementSibling;
		    if (panel.style.maxHeight) {
		      panel.style.maxHeight = null;
		    } else {
		      panel.style.maxHeight = panel.scrollHeight + "px";
		    } 
		  });
		}

		function isTrait_dragdrop(isDragDrop) {
			if(isDragDrop==true) {
				document.getElementById("save_matrix").style.display = "none";
		        } else {
				document.getElementById("save_matrix").style.display = "block";
		        }
		    }


		function showCorrDiv(val) {
		if (val==true) {
		    document.getElementById("corrDiv").style.display = "none";
		  } else {
		    document.getElementById("corrDiv").style.display = "block";
		  }
		}
		
		function textAreaAdjust(o) 
		    {
		        o.style.height = "1px";
		        o.style.height = (25+o.scrollHeight)+"px";
		    }
			

		function networkBoxChange(val) {    
			document.getElementById('heightNetwork').innerHTML = val; 
			if (val == 1) {
				document.getElementById("mynetwork_container").style.height = "930px";
				document.getElementById("mynetwork").style.height = "926px";
				network.setOptions({height: "926px" });
				network.moveTo({scale: 0.98}); 
			} else if (val == 2) {
				document.getElementById("mynetwork_container").style.height = "1000px";
				document.getElementById("mynetwork").style.height = "996px";
				network.setOptions({height: "996px" });
				network.moveTo({scale: 0.98}); 
			} else if (val == 3) {
				document.getElementById("mynetwork_container").style.height = "1100px";
				document.getElementById("mynetwork").style.height = "1096px";
				network.setOptions({height: "1096px" });
				network.moveTo({scale: 0.98}); 
			} else if (val == 4) {
				document.getElementById("mynetwork_container").style.height = "1200px";
				document.getElementById("mynetwork").style.height = "1196px";
				network.setOptions({height: "1196px" });
				network.moveTo({scale: 0.98}); 
			} else if (val == 5) {
				document.getElementById("mynetwork_container").style.height = "1300px";
				document.getElementById("mynetwork").style.height = "1296px";
				network.setOptions({height: "1296px" });
				network.moveTo({scale: 0.98}); 
			} else if (val == 6) {
				document.getElementById("mynetwork_container").style.height = "1400px";
				document.getElementById("mynetwork").style.height = "1396px";
				network.setOptions({height: "1396px" });
				network.moveTo({scale: 0.98}); 
			} else {
				document.getElementById("mynetwork_container").style.height = "835px";
				document.getElementById("mynetwork").style.height = "830px";
				network.setOptions({height: "827px" });
				network.moveTo({scale: 0.98});
			}
		}


			// Manual select cohorts for edge:
		function cohorts_Div(select) {
			if (select.value =="Manual select") {
				document.getElementById("edge_cohorts_Div").style.display = "block";
			}
			else {		
			  document.getElementById("edge_cohorts_Div").style.display = "none";
			}
		}


		// share Project with users related code:	
		const arrHead = ['', ''];

		function shareUserTable() {
			var shareUser = document.createElement('table');
			shareUser.setAttribute('id', 'shareUser');
			var tr = shareUser.insertRow(-1);
			for (var h = 0; h < arrHead.length; h++) {
				var th = document.createElement('th');        
				th.innerHTML = arrHead[h];
				tr.appendChild(th);
			}
			var div = document.getElementById('cont');
			div.appendChild(shareUser);    
			addRow();
		}


		function addRow() {
			var shareTab = document.getElementById('shareUser');
			var rowCnt = shareTab.rows.length;        
			var tr = shareTab.insertRow(rowCnt);      
			tr = shareTab.insertRow(rowCnt);

			for (var c = 0; c < arrHead.length; c++) {
				var td = document.createElement('td');       
				td = tr.insertCell(c);
				if (c == 0) {   
					var ele = document.createElement('input');
					ele.setAttribute('type', 'text');
					ele.setAttribute('value', '');
					td.appendChild(ele);
				}
				else {
					var button = document.createElement('input');
					button.setAttribute('type', 'button');
					button.setAttribute('value', 'X');
					button.setAttribute('class', 'button_type');
					button.setAttribute('id', 'rm');
					button.setAttribute('onclick', 'removeRow(this)');
					td.appendChild(button);
				}
			}
		}

		function removeRow(oButton) {
			var shareTab = document.getElementById('shareUser');
			shareTab.deleteRow(oButton.parentNode.parentNode.rowIndex);       // BUTTON -> TD -> TR.
		}

		$(window).off('beforeunload');
		window.onbeforeunload = null;

		function shareUsersubmit() {
			var myTab = document.getElementById('shareUser');
			var values = new Array();

			for (row = 1; row < myTab.rows.length - 1; row++) {
				for (c = 0; c < myTab.rows[row].cells.length; c++) {                  // EACH CELL IN A ROW.
					var element = myTab.rows.item(row).cells[c];
					if (element.childNodes[0].getAttribute('type') == 'text') {
						values.push(element.childNodes[0].value);                    
					}
				}
			}

			values = values.filter(uniqueUsers);
			var shareValues = removeFrmArr(values, data_Vue.geninfo['user']);
			var shvalues = shareValues.filter(function(el) { return el; });

			shareUser(shvalues);
		}

		function uniqueUsers(value, index, self) {
			return self.indexOf(value) === index;
		}


		function removeFrmArr(array, element) {
			return array.filter(e => e !== element);
		};
	
	
//	$(document).on("mousewheel.noPreventDefault", handler);
//	$(document).on("mousewheel", handler);
	</script>		

</html>

